--- before/net/minecraft/entity/projectile/ProjectileHelper.java
+++ after/net/minecraft/entity/projectile/ProjectileHelper.java
@@ -36,7 +36,7 @@
             List<Entity> list = world.func_72839_b(p_188802_0_, p_188802_0_.func_174813_aQ().func_72321_a(d3, d4, d5).func_186662_g(1.0));
             double d6 = 0.0;
 
-            for (int i = 0; i < list.size(); i++)
+            for (int i = 0; i < list.size(); ++i)
             {
                 Entity entity1 = list.get(i);
 
