--- before/net/minecraft/realms/RealmsScreen.java
+++ after/net/minecraft/realms/RealmsScreen.java
@@ -109,7 +109,7 @@
 
     public void render(int p_render_1_, int p_render_2_, float p_render_3_)
     {
-        for (int i = 0; i < this.proxy.buttons().size(); i++)
+        for (int i = 0; i < this.proxy.buttons().size(); ++i)
         {
             this.proxy.buttons().get(i).render(p_render_1_, p_render_2_, p_render_3_);
         }
