--- before/net/minecraft/potion/PotionUtils.java
+++ after/net/minecraft/potion/PotionUtils.java
@@ -61,7 +61,7 @@
         {
             NBTTagList nbttaglist = tag.getTagList("CustomPotionEffects", 10);
 
-            for (int i = 0; i < nbttaglist.tagCount(); i++)
+            for (int i = 0; i < nbttaglist.tagCount(); ++i)
             {
                 NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
                 PotionEffect potioneffect = PotionEffect.readCustomPotionEffectFromNBT(nbttagcompound);
@@ -273,7 +273,7 @@
                         + I18n.translateToLocalFormatted(
                             "attribute.modifier.plus." + attributemodifier2.getOperation(),
                             ItemStack.DECIMALFORMAT.format(d1),
-                            I18n.translateToLocal("attribute.name." + tuple.getFirst())
+                            I18n.translateToLocal("attribute.name." + (String)tuple.getFirst())
                         )
                     );
                 }
@@ -285,7 +285,7 @@
                         + I18n.translateToLocalFormatted(
                             "attribute.modifier.take." + attributemodifier2.getOperation(),
                             ItemStack.DECIMALFORMAT.format(d1),
-                            I18n.translateToLocal("attribute.name." + tuple.getFirst())
+                            I18n.translateToLocal("attribute.name." + (String)tuple.getFirst())
                         )
                     );
                 }
