--- before/net/minecraft/advancements/critereon/BredAnimalsTrigger.java
+++ after/net/minecraft/advancements/critereon/BredAnimalsTrigger.java
@@ -96,10 +96,15 @@
 
             public boolean test(EntityPlayerMP player, EntityAnimal parent1In, EntityAnimal parent2In, EntityAgeable childIn)
             {
-                return !this.child.test(player, childIn)
-                       ? false
-                       : this.parent.test(player, parent1In) && this.partner.test(player, parent2In)
-                       || this.parent.test(player, parent2In) && this.partner.test(player, parent1In);
+                if (!this.child.test(player, childIn))
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.parent.test(player, parent1In) && this.partner.test(player, parent2In)
+                           || this.parent.test(player, parent2In) && this.partner.test(player, parent1In);
+                }
             }
         }
 
