--- before/net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
+++ after/net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
@@ -96,7 +96,14 @@
 
             public boolean func_193190_a(DimensionType p_193190_1_, DimensionType p_193190_2_)
             {
-                return this.field_193191_a != null && this.field_193191_a != p_193190_1_ ? false : this.field_193192_b == null || this.field_193192_b == p_193190_2_;
+                if (this.field_193191_a != null && this.field_193191_a != p_193190_1_)
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.field_193192_b == null || this.field_193192_b == p_193190_2_;
+                }
             }
         }
 
