--- before/net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
+++ after/net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
@@ -96,7 +96,14 @@
 
             public boolean test(DimensionType from, DimensionType to)
             {
-                return this.from != null && this.from != from ? false : this.to == null || this.to == to;
+                if (this.from != null && this.from != from)
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.to == null || this.to == to;
+                }
             }
         }
 
