--- before/net/minecraft/advancements/critereon/CuredZombieVillagerTrigger.java
+++ after/net/minecraft/advancements/critereon/CuredZombieVillagerTrigger.java
@@ -93,7 +93,14 @@
 
             public boolean test(EntityPlayerMP player, EntityZombie zombie, EntityVillager villager)
             {
-                return !this.zombie.test(player, zombie) ? false : this.villager.test(player, villager);
+                if (!this.zombie.test(player, zombie))
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.villager.test(player, villager);
+                }
             }
         }
 
