--- before/net/minecraft/advancements/critereon/DamageSourcePredicate.java
+++ after/net/minecraft/advancements/critereon/DamageSourcePredicate.java
@@ -90,11 +90,13 @@
         {
             return false;
         }
+        else if (!this.directEntity.test(player, source.getImmediateSource()))
+        {
+            return false;
+        }
         else
         {
-            return !this.directEntity.test(player, source.getImmediateSource())
-                   ? false
-                   : this.sourceEntity.test(player, source.getTrueSource());
+            return this.sourceEntity.test(player, source.getTrueSource());
         }
     }
 
