--- before/net/minecraft/advancements/critereon/DamagePredicate.java
+++ after/net/minecraft/advancements/critereon/DamagePredicate.java
@@ -54,9 +54,13 @@
         {
             return false;
         }
+        else if (this.blocked != null && this.blocked != blocked)
+        {
+            return false;
+        }
         else
         {
-            return this.blocked != null && this.blocked != blocked ? false : this.type.test(player, source);
+            return this.type.test(player, source);
         }
     }
 
