--- before/net/minecraft/advancements/critereon/EnchantedItemTrigger.java
+++ after/net/minecraft/advancements/critereon/EnchantedItemTrigger.java
@@ -92,7 +92,14 @@
 
             public boolean test(ItemStack item, int levelsIn)
             {
-                return !this.item.test(item) ? false : this.levels.test((float)levelsIn);
+                if (!this.item.test(item))
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.levels.test((float)levelsIn);
+                }
             }
         }
 
