--- before/net/minecraft/advancements/critereon/EnterBlockTrigger.java
+++ after/net/minecraft/advancements/critereon/EnterBlockTrigger.java
@@ -104,7 +104,7 @@
                 if (iproperty == null)
                 {
                     throw new JsonSyntaxException(
-                        "Unknown block state property '" + entry.getKey() + "' for block '" + Block.REGISTRY.getNameForObject(block) + "'"
+                        "Unknown block state property '" + (String)entry.getKey() + "' for block '" + Block.REGISTRY.getNameForObject(block) + "'"
                     );
                 }
 
@@ -114,7 +114,13 @@
                 if (!optional.isPresent())
                 {
                     throw new JsonSyntaxException(
-                        "Invalid block state value '" + s + "' for property '" + entry.getKey() + "' on block '" + Block.REGISTRY.getNameForObject(block) + "'"
+                        "Invalid block state value '"
+                        + s
+                        + "' for property '"
+                        + (String)entry.getKey()
+                        + "' on block '"
+                        + Block.REGISTRY.getNameForObject(block)
+                        + "'"
                     );
                 }
 
