--- before/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
+++ after/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
@@ -107,21 +107,21 @@
                 int k = 0;
                 List<ItemPredicate> list = Lists.newArrayList(this.field_192269_d);
 
-                for (int l = 0; l < p_192265_1_.func_70302_i_(); l++)
+                for (int l = 0; l < p_192265_1_.func_70302_i_(); ++l)
                 {
                     ItemStack itemstack = p_192265_1_.func_70301_a(l);
 
                     if (itemstack.func_190926_b())
                     {
-                        j++;
+                        ++j;
                     }
                     else
                     {
-                        k++;
+                        ++k;
 
                         if (itemstack.func_190916_E() >= itemstack.func_77976_d())
                         {
-                            i++;
+                            ++i;
                         }
 
                         Iterator<ItemPredicate> iterator = list.iterator();
@@ -146,9 +146,13 @@
                 {
                     return false;
                 }
+                else if (!this.field_192266_a.func_192514_a((float)k))
+                {
+                    return false;
+                }
                 else
                 {
-                    return !this.field_192266_a.func_192514_a((float)k) ? false : list.isEmpty();
+                    return list.isEmpty();
                 }
             }
         }
