--- before/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
+++ after/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
@@ -107,21 +107,21 @@
                 int k = 0;
                 List<ItemPredicate> list = Lists.newArrayList(this.items);
 
-                for (int l = 0; l < inventory.getSizeInventory(); l++)
+                for (int l = 0; l < inventory.getSizeInventory(); ++l)
                 {
                     ItemStack itemstack = inventory.getStackInSlot(l);
 
                     if (itemstack.isEmpty())
                     {
-                        j++;
+                        ++j;
                     }
                     else
                     {
-                        k++;
+                        ++k;
 
                         if (itemstack.getCount() >= itemstack.getMaxStackSize())
                         {
-                            i++;
+                            ++i;
                         }
 
                         Iterator<ItemPredicate> iterator = list.iterator();
@@ -146,9 +146,13 @@
                 {
                     return false;
                 }
+                else if (!this.occupied.test((float)k))
+                {
+                    return false;
+                }
                 else
                 {
-                    return !this.occupied.test((float)k) ? false : list.isEmpty();
+                    return list.isEmpty();
                 }
             }
         }
