--- before/net/minecraft/advancements/critereon/ItemDurabilityTrigger.java
+++ after/net/minecraft/advancements/critereon/ItemDurabilityTrigger.java
@@ -99,11 +99,13 @@
                 {
                     return false;
                 }
+                else if (!this.durability.test((float)(item.getMaxDamage() - p_193197_2_)))
+                {
+                    return false;
+                }
                 else
                 {
-                    return !this.durability.test((float)(item.getMaxDamage() - p_193197_2_))
-                           ? false
-                           : this.delta.test((float)(item.getItemDamage() - p_193197_2_));
+                    return this.delta.test((float)(item.getItemDamage() - p_193197_2_));
                 }
             }
         }
