--- before/net/minecraft/advancements/critereon/ItemPredicate.java
+++ after/net/minecraft/advancements/critereon/ItemPredicate.java
@@ -37,15 +37,7 @@
         this.nbt = NBTPredicate.ANY;
     }
 
-    public ItemPredicate(
-        @Nullable Item item,
-        @Nullable Integer data,
-        MinMaxBounds count,
-        MinMaxBounds durability,
-        EnchantmentPredicate[] enchantments,
-        @Nullable PotionType potion,
-        NBTPredicate nbt
-    )
+    public ItemPredicate(@Nullable Item item, @Nullable Integer data, MinMaxBounds count, MinMaxBounds durability, EnchantmentPredicate[] enchantments, @Nullable PotionType potion, NBTPredicate nbt)
     {
         this.item = item;
         this.data = data;
@@ -86,7 +78,7 @@
         {
             Map<Enchantment, Integer> map = EnchantmentHelper.getEnchantments(item);
 
-            for (int i = 0; i < this.enchantments.length; i++)
+            for (int i = 0; i < this.enchantments.length; ++i)
             {
                 if (!this.enchantments[i].test(map))
                 {
@@ -95,7 +87,15 @@
             }
 
             PotionType potiontype = PotionUtils.getPotionFromItem(item);
-            return this.potion == null || this.potion == potiontype;
+
+            if (this.potion != null && this.potion != potiontype)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
         }
     }
 
@@ -104,6 +104,13 @@
         if (element != null && !element.isJsonNull())
         {
             JsonObject jsonobject = JsonUtils.getJsonObject(element, "item");
+            if (jsonobject.has("type"))
+            {
+                 final ResourceLocation rl = new ResourceLocation(JsonUtils.getString(jsonobject, "type"));
+                 final Map<ResourceLocation, java.util.function.Function<JsonObject, ItemPredicate>> map = net.minecraftforge.advancements.critereon.ItemPredicates.getPredicates();
+                 if (map.containsKey(rl)) return map.get(rl).apply(jsonobject);
+                 else throw new JsonSyntaxException("There is no ItemPredicate of type "+rl);
+            }
             MinMaxBounds minmaxbounds = MinMaxBounds.deserialize(jsonobject.get("count"));
             MinMaxBounds minmaxbounds1 = MinMaxBounds.deserialize(jsonobject.get("durability"));
             Integer integer = jsonobject.has("data") ? JsonUtils.getInt(jsonobject, "data") : null;
