--- before/net/minecraft/advancements/critereon/MinMaxBounds.java
+++ after/net/minecraft/advancements/critereon/MinMaxBounds.java
@@ -19,14 +19,26 @@
 
     public boolean test(float value)
     {
-        return this.min != null && this.min > value ? false : this.max == null || !(this.max < value);
+        if (this.min != null && this.min > value)
+        {
+            return false;
+        }
+        else
+        {
+            return this.max == null || !(this.max < value);
+        }
     }
 
     public boolean testSquare(double value)
     {
-        return this.min != null && (double)(this.min * this.min) > value
-               ? false
-               : this.max == null || !((double)(this.max * this.max) < value);
+        if (this.min != null && (double)(this.min * this.min) > value)
+        {
+            return false;
+        }
+        else
+        {
+            return this.max == null || !((double)(this.max * this.max) < value);
+        }
     }
 
     public static MinMaxBounds deserialize(@Nullable JsonElement element)
