--- before/net/minecraft/advancements/critereon/MobEffectsPredicate.java
+++ after/net/minecraft/advancements/critereon/MobEffectsPredicate.java
@@ -126,11 +126,13 @@
                 {
                     return false;
                 }
+                else if (this.ambient != null && this.ambient != effect.getIsAmbient())
+                {
+                    return false;
+                }
                 else
                 {
-                    return this.ambient != null && this.ambient != effect.getIsAmbient()
-                           ? false
-                           : this.visible == null || this.visible == effect.doesShowParticles();
+                    return this.visible == null || this.visible == effect.doesShowParticles();
                 }
             }
 
