--- before/net/minecraft/advancements/critereon/NBTPredicate.java
+++ after/net/minecraft/advancements/critereon/NBTPredicate.java
@@ -36,7 +36,14 @@
 
     public boolean test(@Nullable NBTBase nbt)
     {
-        return nbt == null ? this == ANY : this.tag == null || NBTUtil.areNBTEquals(this.tag, nbt, true);
+        if (nbt == null)
+        {
+            return this == ANY;
+        }
+        else
+        {
+            return this.tag == null || NBTUtil.areNBTEquals(this.tag, nbt, true);
+        }
     }
 
     public static NBTPredicate deserialize(@Nullable JsonElement json)
