--- before/net/minecraft/advancements/critereon/NetherTravelTrigger.java
+++ after/net/minecraft/advancements/critereon/NetherTravelTrigger.java
@@ -102,11 +102,13 @@
                 {
                     return false;
                 }
+                else if (!this.exited.test(world, x, y, z))
+                {
+                    return false;
+                }
                 else
                 {
-                    return !this.exited.test(world, x, y, z)
-                           ? false
-                           : this.distance
+                    return this.distance
                            .test(enteredNetherPosition.x, enteredNetherPosition.y, enteredNetherPosition.z, x, y, z);
                 }
             }
