--- before/net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
+++ after/net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
@@ -97,9 +97,14 @@
                 EntityPlayerMP player, Entity entity, DamageSource source, float dealt, float taken, boolean blocked
             )
             {
-                return !this.damage.test(player, source, dealt, taken, blocked)
-                       ? false
-                       : this.entity.test(player, entity);
+                if (!this.damage.test(player, source, dealt, taken, blocked))
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.entity.test(player, entity);
+                }
             }
         }
 
