--- before/net/minecraft/advancements/critereon/VillagerTradeTrigger.java
+++ after/net/minecraft/advancements/critereon/VillagerTradeTrigger.java
@@ -93,7 +93,14 @@
 
             public boolean test(EntityPlayerMP player, EntityVillager villager, ItemStack item)
             {
-                return !this.villager.test(player, villager) ? false : this.item.test(item);
+                if (!this.villager.test(player, villager))
+                {
+                    return false;
+                }
+                else
+                {
+                    return this.item.test(item);
+                }
             }
         }
 
