--- before/net/minecraft/item/crafting/RecipesMapExtending.java
+++ after/net/minecraft/item/crafting/RecipesMapExtending.java
@@ -44,7 +44,7 @@
         {
             ItemStack itemstack = ItemStack.EMPTY;
 
-            for (int i = 0; i < inv.getSizeInventory() && itemstack.isEmpty(); i++)
+            for (int i = 0; i < inv.getSizeInventory() && itemstack.isEmpty(); ++i)
             {
                 ItemStack itemstack1 = inv.getStackInSlot(i);
 
@@ -66,9 +66,13 @@
                 {
                     return false;
                 }
+                else if (this.isExplorationMap(mapdata))
+                {
+                    return false;
+                }
                 else
                 {
-                    return this.isExplorationMap(mapdata) ? false : mapdata.scale < 4;
+                    return mapdata.scale < 4;
                 }
             }
         }
@@ -95,7 +99,7 @@
     {
         ItemStack itemstack = ItemStack.EMPTY;
 
-        for (int i = 0; i < inv.getSizeInventory() && itemstack.isEmpty(); i++)
+        for (int i = 0; i < inv.getSizeInventory() && itemstack.isEmpty(); ++i)
         {
             ItemStack itemstack1 = inv.getStackInSlot(i);
 
