--- before/net/minecraft/client/renderer/debug/DebugRendererPathfinding.java
+++ after/net/minecraft/client/renderer/debug/DebugRendererPathfinding.java
@@ -90,7 +90,7 @@
                         0.5F
                     );
 
-                    for (int j = 0; j < path.func_75874_d(); j++)
+                    for (int j = 0; j < path.func_75874_d(); ++j)
                     {
                         PathPoint pathpoint1 = path.func_75877_a(j);
 
@@ -168,7 +168,7 @@
                     }
                 }
 
-                for (int k = 0; k < path1.func_75874_d(); k++)
+                for (int k = 0; k < path1.func_75874_d(); ++k)
                 {
                     PathPoint pathpoint2 = path1.func_75877_a(k);
 
@@ -215,7 +215,7 @@
         BufferBuilder bufferbuilder = tessellator.func_178180_c();
         bufferbuilder.func_181668_a(3, DefaultVertexFormats.field_181706_f);
 
-        for (int i = 0; i < p_190067_2_.func_75874_d(); i++)
+        for (int i = 0; i < p_190067_2_.func_75874_d(); ++i)
         {
             PathPoint pathpoint = p_190067_2_.func_75877_a(i);
 
