--- before/net/minecraft/client/renderer/debug/DebugRendererPathfinding.java
+++ after/net/minecraft/client/renderer/debug/DebugRendererPathfinding.java
@@ -90,7 +90,7 @@
                         0.5F
                     );
 
-                    for (int j = 0; j < path.getCurrentPathLength(); j++)
+                    for (int j = 0; j < path.getCurrentPathLength(); ++j)
                     {
                         PathPoint pathpoint1 = path.getPathPointFromIndex(j);
 
@@ -168,7 +168,7 @@
                     }
                 }
 
-                for (int k = 0; k < path1.getCurrentPathLength(); k++)
+                for (int k = 0; k < path1.getCurrentPathLength(); ++k)
                 {
                     PathPoint pathpoint2 = path1.getPathPointFromIndex(k);
 
@@ -215,7 +215,7 @@
         BufferBuilder bufferbuilder = tessellator.getBuffer();
         bufferbuilder.begin(3, DefaultVertexFormats.POSITION_COLOR);
 
-        for (int i = 0; i < pathIn.getCurrentPathLength(); i++)
+        for (int i = 0; i < pathIn.getCurrentPathLength(); ++i)
         {
             PathPoint pathpoint = pathIn.getPathPointFromIndex(i);
 
