--- before/net/minecraft/util/CooldownTracker.java
+++ after/net/minecraft/util/CooldownTracker.java
@@ -11,8 +11,8 @@
 
 public class CooldownTracker
 {
-    private final Map<Item, CooldownTracker.Cooldown> cooldowns = Maps.newHashMap();
-    private int ticks;
+    public final Map<Item, Cooldown> cooldowns = Maps.<Item, Cooldown>newHashMap();
+    public int ticks;
 
     public boolean hasCooldown(Item itemIn)
     {
@@ -37,7 +37,7 @@
 
     public void tick()
     {
-        this.ticks++;
+        ++this.ticks;
 
         if (!this.cooldowns.isEmpty())
         {
@@ -77,10 +77,10 @@
     {
     }
 
-    class Cooldown
+    public class Cooldown
     {
-        final int createTicks;
-        final int expireTicks;
+        public final int createTicks;
+        public final int expireTicks;
 
         private Cooldown(int createTicksIn, int expireTicksIn)
         {
