--- before/net/minecraft/util/EntityDamageSourceIndirect.java
+++ after/net/minecraft/util/EntityDamageSourceIndirect.java
@@ -39,8 +39,11 @@
         ItemStack itemstack = this.indirectEntity instanceof EntityLivingBase ? ((EntityLivingBase)this.indirectEntity).getHeldItemMainhand() : ItemStack.EMPTY;
         String s = "death.attack." + this.damageType;
         String s1 = s + ".item";
-        return !itemstack.isEmpty() && itemstack.hasDisplayName() && I18n.canTranslate(s1)
-               ? new TextComponentTranslation(s1, entityLivingBaseIn.getDisplayName(), itextcomponent, itemstack.getTextComponent())
-               : new TextComponentTranslation(s, entityLivingBaseIn.getDisplayName(), itextcomponent);
+        return !itemstack.isEmpty() && itemstack.hasDisplayName() && I18n.canTranslate(s1) ? new TextComponentTranslation(s1, new Object[] {entityLivingBaseIn.getDisplayName(), itextcomponent, itemstack.getTextComponent()}) : new TextComponentTranslation(s, new Object[] {entityLivingBaseIn.getDisplayName(), itextcomponent});
+    }
+
+    @Nullable
+    public Entity getProximateDamageSource() {
+        return super.getTrueSource();
     }
 }
