--- before/net/minecraft/util/FrameTimer.java
+++ after/net/minecraft/util/FrameTimer.java
@@ -14,7 +14,7 @@
     public void func_181747_a(long p_181747_1_)
     {
         this.field_181752_a[this.field_181755_d] = p_181747_1_;
-        this.field_181755_d++;
+        ++this.field_181755_d;
 
         if (this.field_181755_d == 240)
         {
@@ -24,7 +24,7 @@
         if (this.field_181754_c < 240)
         {
             this.field_181753_b = 0;
-            this.field_181754_c++;
+            ++this.field_181754_c;
         }
         else
         {
