--- before/net/minecraft/util/FrameTimer.java
+++ after/net/minecraft/util/FrameTimer.java
@@ -14,7 +14,7 @@
     public void addFrame(long runningTime)
     {
         this.frames[this.index] = runningTime;
-        this.index++;
+        ++this.index;
 
         if (this.index == 240)
         {
@@ -24,7 +24,7 @@
         if (this.counter < 240)
         {
             this.lastIndex = 0;
-            this.counter++;
+            ++this.counter;
         }
         else
         {
