--- before/net/minecraft/util/JsonUtils.java
+++ after/net/minecraft/util/JsonUtils.java
@@ -55,7 +55,14 @@
 
     public static boolean hasField(JsonObject json, String memberName)
     {
-        return json == null ? false : json.get(memberName) != null;
+        if (json == null)
+        {
+            return false;
+        }
+        else
+        {
+            return json.get(memberName) != null;
+        }
     }
 
     public static String getString(JsonElement json, String memberName)
@@ -296,7 +303,7 @@
             JsonObject json, String memberName, T fallback, JsonDeserializationContext context, Class <? extends T > adapter
         )
     {
-        return json.has(memberName) ? deserializeClass(json.get(memberName), memberName, context, adapter) : fallback;
+        return (T)(json.has(memberName) ? deserializeClass(json.get(memberName), memberName, context, adapter) : fallback);
     }
 
     public static String toString(JsonElement json)
@@ -362,7 +369,7 @@
         {
             JsonReader jsonreader = new JsonReader(p_193838_1_);
             jsonreader.setLenient(p_193838_3_);
-            return (T)gson.getAdapter(TypeToken.get(p_193838_2_)).read(jsonreader);
+            return (T) gson.getAdapter(TypeToken.get(p_193838_2_)).read(jsonreader);
         }
         catch (IOException ioexception)
         {
