--- before/net/minecraft/util/MovementInputFromOptions.java
+++ after/net/minecraft/util/MovementInputFromOptions.java
@@ -22,7 +22,7 @@
 
         if (this.gameSettings.keyBindForward.isKeyDown())
         {
-            this.moveForward++;
+            ++this.moveForward;
             this.forwardKeyDown = true;
         }
         else
@@ -32,7 +32,7 @@
 
         if (this.gameSettings.keyBindBack.isKeyDown())
         {
-            this.moveForward--;
+            --this.moveForward;
             this.backKeyDown = true;
         }
         else
@@ -42,7 +42,7 @@
 
         if (this.gameSettings.keyBindLeft.isKeyDown())
         {
-            this.moveStrafe++;
+            ++this.moveStrafe;
             this.leftKeyDown = true;
         }
         else
@@ -52,7 +52,7 @@
 
         if (this.gameSettings.keyBindRight.isKeyDown())
         {
-            this.moveStrafe--;
+            --this.moveStrafe;
             this.rightKeyDown = true;
         }
         else
