--- before/net/minecraft/util/Timer.java
+++ after/net/minecraft/util/Timer.java
@@ -24,8 +24,8 @@
         long i = Minecraft.getSystemTime();
         this.elapsedPartialTicks = (float)(i - this.lastSyncSysClock) / this.tickLength;
         this.lastSyncSysClock = i;
-        this.renderPartialTicks = this.renderPartialTicks + this.elapsedPartialTicks;
+        this.renderPartialTicks += this.elapsedPartialTicks;
         this.elapsedTicks = (int)this.renderPartialTicks;
-        this.renderPartialTicks = this.renderPartialTicks - (float)this.elapsedTicks;
+        this.renderPartialTicks -= (float)this.elapsedTicks;
     }
 }
