--- before/net/minecraft/util/Timer.java
+++ after/net/minecraft/util/Timer.java
@@ -24,8 +24,8 @@
         long i = Minecraft.func_71386_F();
         this.field_194148_c = (float)(i - this.field_74277_g) / this.field_194149_e;
         this.field_74277_g = i;
-        this.field_194147_b = this.field_194147_b + this.field_194148_c;
+        this.field_194147_b += this.field_194148_c;
         this.field_74280_b = (int)this.field_194147_b;
-        this.field_194147_b = this.field_194147_b - (float)this.field_74280_b;
+        this.field_194147_b -= (float)this.field_74280_b;
     }
 }
