--- before/net/minecraft/network/EnumConnectionState.java
+++ after/net/minecraft/network/EnumConnectionState.java
@@ -313,7 +313,7 @@
     public Packet<?> getPacket(EnumPacketDirection direction, int packetId) throws InstantiationException, IllegalAccessException
     {
         Class <? extends Packet<? >> oclass = this.directionMaps.get(direction).get(Integer.valueOf(packetId));
-        return (Packet<?>)(oclass == null ? null : oclass.newInstance());
+        return oclass == null ? null : oclass.newInstance();
     }
 
     public int getId()
