--- before/net/minecraft/network/LegacyPingHandler.java
+++ after/net/minecraft/network/LegacyPingHandler.java
@@ -58,15 +58,15 @@
                         }
 
                         LOGGER.debug("Ping: (1.4-1.5.x) from {}:{}", inetsocketaddress.getAddress(), inetsocketaddress.getPort());
-                        String s1 = String.format(
-                                        "\u00a71\u0000%d\u0000%s\u0000%s\u0000%d\u0000%d",
-                                        127,
-                                        minecraftserver.getMinecraftVersion(),
-                                        minecraftserver.getMOTD(),
-                                        minecraftserver.getCurrentPlayerCount(),
-                                        minecraftserver.getMaxPlayers()
-                                    );
-                        this.writeAndFlush(p_channelRead_1_, this.getStringBuffer(s1));
+                        String s = String.format(
+                                       "\u00a71\u0000%d\u0000%s\u0000%s\u0000%d\u0000%d",
+                                       127,
+                                       minecraftserver.getMinecraftVersion(),
+                                       minecraftserver.getMOTD(),
+                                       minecraftserver.getCurrentPlayerCount(),
+                                       minecraftserver.getMaxPlayers()
+                                   );
+                        this.writeAndFlush(p_channelRead_1_, this.getStringBuffer(s));
                         break;
                     default:
                         boolean flag1 = bytebuf.readUnsignedByte() == 1;
@@ -84,15 +84,15 @@
                         }
 
                         LOGGER.debug("Ping: (1.6) from {}:{}", inetsocketaddress.getAddress(), inetsocketaddress.getPort());
-                        String s = String.format(
-                                       "\u00a71\u0000%d\u0000%s\u0000%s\u0000%d\u0000%d",
-                                       127,
-                                       minecraftserver.getMinecraftVersion(),
-                                       minecraftserver.getMOTD(),
-                                       minecraftserver.getCurrentPlayerCount(),
-                                       minecraftserver.getMaxPlayers()
-                                   );
-                        ByteBuf bytebuf1 = this.getStringBuffer(s);
+                        String s1 = String.format(
+                                        "\u00a71\u0000%d\u0000%s\u0000%s\u0000%d\u0000%d",
+                                        127,
+                                        minecraftserver.getMinecraftVersion(),
+                                        minecraftserver.getMOTD(),
+                                        minecraftserver.getCurrentPlayerCount(),
+                                        minecraftserver.getMaxPlayers()
+                                    );
+                        ByteBuf bytebuf1 = this.getStringBuffer(s1);
 
                         try
                         {
