--- before/net/minecraft/network/NetworkManager.java
+++ after/net/minecraft/network/NetworkManager.java
@@ -37,6 +37,7 @@
 import net.minecraft.util.LazyLoadBase;
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.fml.common.network.internal.FMLProxyPacket;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.commons.lang3.ArrayUtils;
@@ -51,33 +52,38 @@
     private static final Logger LOGGER = LogManager.getLogger();
     public static final Marker NETWORK_MARKER = MarkerManager.getMarker("NETWORK");
     public static final Marker NETWORK_PACKETS_MARKER = MarkerManager.getMarker("NETWORK_PACKETS", NETWORK_MARKER);
-    public static final AttributeKey<EnumConnectionState> PROTOCOL_ATTRIBUTE_KEY = AttributeKey.valueOf("protocol");
+    public static final AttributeKey<EnumConnectionState> PROTOCOL_ATTRIBUTE_KEY = AttributeKey.<EnumConnectionState>valueOf("protocol");
     public static final LazyLoadBase<NioEventLoopGroup> CLIENT_NIO_EVENTLOOP = new LazyLoadBase<NioEventLoopGroup>()
     {
         protected NioEventLoopGroup load()
         {
-            return new NioEventLoopGroup(0, new ThreadFactoryBuilder().setNameFormat("Netty Client IO #%d").setDaemon(true).build());
+            return new NioEventLoopGroup(0, (new ThreadFactoryBuilder()).setNameFormat("Netty Client IO #%d").setDaemon(true).build());
         }
     };
     public static final LazyLoadBase<EpollEventLoopGroup> CLIENT_EPOLL_EVENTLOOP = new LazyLoadBase<EpollEventLoopGroup>()
     {
         protected EpollEventLoopGroup load()
         {
-            return new EpollEventLoopGroup(0, new ThreadFactoryBuilder().setNameFormat("Netty Epoll Client IO #%d").setDaemon(true).build());
+            return new EpollEventLoopGroup(0, (new ThreadFactoryBuilder()).setNameFormat("Netty Epoll Client IO #%d").setDaemon(true).build());
         }
     };
     public static final LazyLoadBase<LocalEventLoopGroup> CLIENT_LOCAL_EVENTLOOP = new LazyLoadBase<LocalEventLoopGroup>()
     {
         protected LocalEventLoopGroup load()
         {
-            return new LocalEventLoopGroup(0, new ThreadFactoryBuilder().setNameFormat("Netty Local Client IO #%d").setDaemon(true).build());
+            return new LocalEventLoopGroup(0, (new ThreadFactoryBuilder()).setNameFormat("Netty Local Client IO #%d").setDaemon(true).build());
         }
     };
     private final EnumPacketDirection direction;
-    private final Queue<NetworkManager.InboundHandlerTuplePacketListener> outboundPacketsQueue = Queues.newConcurrentLinkedQueue();
+    private final Queue<NetworkManager.InboundHandlerTuplePacketListener> outboundPacketsQueue = Queues.<NetworkManager.InboundHandlerTuplePacketListener>newConcurrentLinkedQueue();
     private final ReentrantReadWriteLock readWriteLock = new ReentrantReadWriteLock();
-    private Channel channel;
-    private SocketAddress socketAddress;
+    public Channel channel; // CatServer - private -> public
+    // Spigot Start // PAIL
+    public SocketAddress socketAddress;
+    public java.util.UUID spoofedUUID;
+    public com.mojang.authlib.properties.Property[] spoofedProfile;
+    // Spigot End
+    public boolean preparing = true; // Spigot
     private INetHandler packetListener;
     private ITextComponent terminationReason;
     private boolean isEncrypted;
@@ -88,12 +94,17 @@
         this.direction = packetDirection;
     }
 
-    @Override
+    public EnumPacketDirection getDirection()
+    {
+        return this.direction;
+    }
+
     public void channelActive(ChannelHandlerContext p_channelActive_1_) throws Exception
     {
         super.channelActive(p_channelActive_1_);
         this.channel = p_channelActive_1_.channel();
         this.socketAddress = this.channel.remoteAddress();
+        this.preparing = false; // Spigot
 
         try
         {
@@ -112,40 +123,39 @@
         LOGGER.debug("Enabled auto read");
     }
 
-    @Override
     public void channelInactive(ChannelHandlerContext p_channelInactive_1_) throws Exception
     {
-        this.closeChannel(new TextComponentTranslation("disconnect.endOfStream"));
+        this.closeChannel(new TextComponentTranslation("disconnect.endOfStream", new Object[0]));
     }
 
-    @Override
     public void exceptionCaught(ChannelHandlerContext p_exceptionCaught_1_, Throwable p_exceptionCaught_2_) throws Exception
     {
         TextComponentTranslation textcomponenttranslation;
 
         if (p_exceptionCaught_2_ instanceof TimeoutException)
         {
-            textcomponenttranslation = new TextComponentTranslation("disconnect.timeout");
+            textcomponenttranslation = new TextComponentTranslation("disconnect.timeout", new Object[0]);
         }
         else
         {
-            textcomponenttranslation = new TextComponentTranslation("disconnect.genericReason", "Internal Exception: " + p_exceptionCaught_2_);
+            textcomponenttranslation = new TextComponentTranslation("disconnect.genericReason", new Object[] {"Internal Exception: " + p_exceptionCaught_2_});
         }
 
         LOGGER.debug(textcomponenttranslation.getUnformattedText(), p_exceptionCaught_2_);
         this.closeChannel(textcomponenttranslation);
     }
 
-    protected void channelRead0(ChannelHandlerContext p_channelRead0_1_, Packet p_channelRead0_2_) throws Exception
+    protected void channelRead0(ChannelHandlerContext p_channelRead0_1_, Packet<?> p_channelRead0_2_) throws Exception
     {
         if (this.channel.isOpen())
         {
             try
             {
-                p_channelRead0_2_.processPacket(this.packetListener);
+                ((Packet<INetHandler>)p_channelRead0_2_).processPacket(this.packetListener);
             }
-            catch (ThreadQuickExitException threadquickexitexception)
+            catch (ThreadQuickExitException var4)
             {
+                ;
             }
         }
     }
@@ -162,7 +172,7 @@
         if (this.isChannelOpen())
         {
             this.flushOutboundQueue();
-            this.dispatchPacket(packetIn, null);
+            this.dispatchPacket(packetIn, (GenericFutureListener[])null);
         }
         else
         {
@@ -170,7 +180,7 @@
 
             try
             {
-                this.outboundPacketsQueue.add(new NetworkManager.InboundHandlerTuplePacketListener(packetIn));
+                this.outboundPacketsQueue.add(new NetworkManager.InboundHandlerTuplePacketListener(packetIn, new GenericFutureListener[0]));
             }
             finally
             {
@@ -179,16 +189,12 @@
         }
     }
 
-    public void sendPacket(
-        Packet<?> packetIn,
-        GenericFutureListener <? extends Future <? super Void >> listener,
-        GenericFutureListener <? extends Future <? super Void >> ... listeners
-    )
+    public void sendPacket(Packet<?> packetIn, GenericFutureListener <? extends Future <? super Void >> listener, GenericFutureListener <? extends Future <? super Void >> ... listeners)
     {
         if (this.isChannelOpen())
         {
             this.flushOutboundQueue();
-            this.dispatchPacket(packetIn, ArrayUtils.add(listeners, 0, listener));
+            this.dispatchPacket(packetIn, (GenericFutureListener[])ArrayUtils.add(listeners, 0, listener));
         }
         else
         {
@@ -196,7 +202,7 @@
 
             try
             {
-                this.outboundPacketsQueue.add(new NetworkManager.InboundHandlerTuplePacketListener(packetIn, ArrayUtils.add(listeners, 0, listener)));
+                this.outboundPacketsQueue.add(new NetworkManager.InboundHandlerTuplePacketListener(packetIn, (GenericFutureListener[])ArrayUtils.add(listeners, 0, listener)));
             }
             finally
             {
@@ -210,7 +216,7 @@
         final EnumConnectionState enumconnectionstate = EnumConnectionState.getFromPacket(inPacket);
         final EnumConnectionState enumconnectionstate1 = this.channel.attr(PROTOCOL_ATTRIBUTE_KEY).get();
 
-        if (enumconnectionstate1 != enumconnectionstate)
+        if (enumconnectionstate1 != enumconnectionstate && !( inPacket instanceof FMLProxyPacket))
         {
             LOGGER.debug("Disabled auto read");
             this.channel.config().setAutoRead(false);
@@ -218,7 +224,7 @@
 
         if (this.channel.eventLoop().inEventLoop())
         {
-            if (enumconnectionstate != enumconnectionstate1)
+            if (enumconnectionstate != enumconnectionstate1 && !( inPacket instanceof FMLProxyPacket))
             {
                 this.setConnectionState(enumconnectionstate);
             }
@@ -236,10 +242,9 @@
         {
             this.channel.eventLoop().execute(new Runnable()
             {
-                @Override
                 public void run()
                 {
-                    if (enumconnectionstate != enumconnectionstate1)
+                    if (enumconnectionstate != enumconnectionstate1  && !( inPacket instanceof FMLProxyPacket))
                     {
                         NetworkManager.this.setConnectionState(enumconnectionstate);
                     }
@@ -268,9 +273,7 @@
                 while (!this.outboundPacketsQueue.isEmpty())
                 {
                     NetworkManager.InboundHandlerTuplePacketListener networkmanager$inboundhandlertuplepacketlistener = this.outboundPacketsQueue.poll();
-                    this.dispatchPacket(
-                        networkmanager$inboundhandlertuplepacketlistener.packet, networkmanager$inboundhandlertuplepacketlistener.futureListeners
-                    );
+                    this.dispatchPacket(networkmanager$inboundhandlertuplepacketlistener.packet, networkmanager$inboundhandlertuplepacketlistener.futureListeners);
                 }
             }
             finally
@@ -300,11 +303,12 @@
         return this.socketAddress;
     }
 
-    public void closeChannel(ITextComponent message)
-    {
-        if (this.channel.isOpen())
-        {
-            this.channel.close().awaitUninterruptibly();
+    public void closeChannel(ITextComponent message) {
+        this.preparing = false; // Spigot
+        if (this.channel.isOpen()) {
+            // We can't wait as this may be called from an event loop.
+            // this.channel.close().awaitUninterruptibly();
+            this.channel.close();
             this.terminationReason = message;
         }
     }
@@ -317,6 +321,7 @@
     @SideOnly(Side.CLIENT)
     public static NetworkManager createNetworkManagerAndConnect(InetAddress address, int serverPort, boolean useNativeTransport)
     {
+        if (address instanceof java.net.Inet6Address) System.setProperty("java.net.preferIPv4Stack", "false");
         final NetworkManager networkmanager = new NetworkManager(EnumPacketDirection.CLIENTBOUND);
         Class <? extends SocketChannel > oclass;
         LazyLoadBase <? extends EventLoopGroup > lazyloadbase;
@@ -332,35 +337,22 @@
             lazyloadbase = CLIENT_NIO_EVENTLOOP;
         }
 
-        new Bootstrap()
-        .group(lazyloadbase.getValue())
-        .handler(
-            new ChannelInitializer<Channel>()
+        ((Bootstrap)((Bootstrap)((Bootstrap)(new Bootstrap()).group(lazyloadbase.getValue())).handler(new ChannelInitializer<Channel>()
         {
-            @Override
             protected void initChannel(Channel p_initChannel_1_) throws Exception
             {
                 try
                 {
-                    p_initChannel_1_.config().setOption(ChannelOption.TCP_NODELAY, true);
+                    p_initChannel_1_.config().setOption(ChannelOption.TCP_NODELAY, Boolean.valueOf(true));
                 }
-                catch (ChannelException channelexception)
+                catch (ChannelException var3)
                 {
+                    ;
                 }
 
-                p_initChannel_1_.pipeline()
-                .addLast("timeout", new ReadTimeoutHandler(30))
-                .addLast("splitter", new NettyVarint21FrameDecoder())
-                .addLast("decoder", new NettyPacketDecoder(EnumPacketDirection.CLIENTBOUND))
-                .addLast("prepender", new NettyVarint21FrameEncoder())
-                .addLast("encoder", new NettyPacketEncoder(EnumPacketDirection.SERVERBOUND))
-                .addLast("packet_handler", networkmanager);
+                p_initChannel_1_.pipeline().addLast("timeout", new ReadTimeoutHandler(30)).addLast("splitter", new NettyVarint21FrameDecoder()).addLast("decoder", new NettyPacketDecoder(EnumPacketDirection.CLIENTBOUND)).addLast("prepender", new NettyVarint21FrameEncoder()).addLast("encoder", new NettyPacketEncoder(EnumPacketDirection.SERVERBOUND)).addLast("packet_handler", networkmanager);
             }
-        }
-        )
-        .channel(oclass)
-        .connect(address, serverPort)
-        .syncUninterruptibly();
+        })).channel(oclass)).connect(address, serverPort).syncUninterruptibly();
         return networkmanager;
     }
 
@@ -368,14 +360,13 @@
     public static NetworkManager provideLocalClient(SocketAddress address)
     {
         final NetworkManager networkmanager = new NetworkManager(EnumPacketDirection.CLIENTBOUND);
-        new Bootstrap().group(CLIENT_LOCAL_EVENTLOOP.getValue()).handler(new ChannelInitializer<Channel>()
+        ((Bootstrap)((Bootstrap)((Bootstrap)(new Bootstrap()).group(CLIENT_LOCAL_EVENTLOOP.getValue())).handler(new ChannelInitializer<Channel>()
         {
-            @Override
             protected void initChannel(Channel p_initChannel_1_) throws Exception
             {
                 p_initChannel_1_.pipeline().addLast("packet_handler", networkmanager);
             }
-        }).channel(LocalChannel.class).connect(address).syncUninterruptibly();
+        })).channel(LocalChannel.class)).connect(address).syncUninterruptibly();
         return networkmanager;
     }
 
@@ -427,7 +418,7 @@
             }
             else
             {
-                this.channel.pipeline().addBefore("decoder", "decompress", new NettyCompressionDecoder(threshold));
+                this.channel.pipeline().addAfter("splitter", "decompress", new NettyCompressionDecoder(threshold)); // CatServer
             }
 
             if (this.channel.pipeline().get("compress") instanceof NettyCompressionEncoder)
@@ -471,10 +462,15 @@
                 }
                 else if (this.getNetHandler() != null)
                 {
-                    this.getNetHandler().onDisconnect(new TextComponentTranslation("multiplayer.disconnect.generic"));
+                    this.getNetHandler().onDisconnect(new TextComponentTranslation("multiplayer.disconnect.generic", new Object[0]));
                 }
             }
         }
+    }
+
+    public Channel channel()
+    {
+        return channel;
     }
 
     static class InboundHandlerTuplePacketListener
@@ -488,4 +484,11 @@
                 this.futureListeners = inFutureListeners;
             }
         }
+    // Spigot Start
+    public SocketAddress getRawAddress()
+    {
+        return this.channel.remoteAddress();
+    }
+// Spigot End
+
 }
