--- before/net/minecraft/entity/item/EntityMinecartTNT.java
+++ after/net/minecraft/entity/item/EntityMinecartTNT.java
@@ -56,7 +56,7 @@
 
         if (this.minecartTNTFuse > 0)
         {
-            this.minecartTNTFuse--;
+            --this.minecartTNTFuse;
             this.world.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, this.posX, this.posY + 0.5, this.posZ, 0.0, 0.0, 0.0);
         }
         else if (this.minecartTNTFuse == 0)
