--- before/net/minecraft/client/model/ModelDragon.java
+++ after/net/minecraft/client/model/ModelDragon.java
@@ -130,7 +130,7 @@
         f2 = 20.0F;
         float f3 = -12.0F;
 
-        for (int i = 0; i < 5; i++)
+        for (int i = 0; i < 5; ++i)
         {
             double[] adouble1 = entitydragon.getMovementOffsets(5 - i, this.partialTicks);
             float f9 = (float)Math.cos((double)((float)i * 0.45F + f8)) * 0.15F;
@@ -161,7 +161,7 @@
         this.body.rotateAngleZ = 0.0F;
         this.body.render(scale);
 
-        for (int j = 0; j < 2; j++)
+        for (int j = 0; j < 2; ++j)
         {
             GlStateManager.enableCull();
             float f11 = f * (float)(Math.PI * 2);
@@ -196,7 +196,7 @@
         f4 = 0.0F;
         adouble = entitydragon.getMovementOffsets(11, this.partialTicks);
 
-        for (int k = 0; k < 12; k++)
+        for (int k = 0; k < 12; ++k)
         {
             adouble2 = entitydragon.getMovementOffsets(12 + k, this.partialTicks);
             f10 = (float)((double)f10 + Math.sin((double)((float)k * 0.45F + f8)) * 0.05F);
