--- before/net/minecraft/client/model/ModelOcelot.java
+++ after/net/minecraft/client/model/ModelOcelot.java
@@ -156,9 +156,9 @@
 
         if (entityocelot.isSneaking())
         {
-            this.ocelotBody.rotationPointY++;
+            ++this.ocelotBody.rotationPointY;
             this.ocelotHead.rotationPointY += 2.0F;
-            this.ocelotTail.rotationPointY++;
+            ++this.ocelotTail.rotationPointY;
             this.ocelotTail2.rotationPointY += -4.0F;
             this.ocelotTail2.rotationPointZ += 2.0F;
             this.ocelotTail.rotateAngleX = (float)(Math.PI / 2);
@@ -179,7 +179,7 @@
             this.ocelotBody.rotationPointY += -4.0F;
             this.ocelotBody.rotationPointZ += 5.0F;
             this.ocelotHead.rotationPointY += -3.3F;
-            this.ocelotHead.rotationPointZ++;
+            ++this.ocelotHead.rotationPointZ;
             this.ocelotTail.rotationPointY += 8.0F;
             this.ocelotTail.rotationPointZ += -2.0F;
             this.ocelotTail2.rotationPointY += 2.0F;
