--- before/net/minecraft/client/model/ModelParrot.java
+++ after/net/minecraft/client/model/ModelParrot.java
@@ -120,9 +120,8 @@
                 return;
             }
 
-            this.legLeft.rotateAngleX = this.legLeft.rotateAngleX + MathHelper.cos(limbSwing * 0.6662F) * 1.4F * limbSwingAmount;
-            this.legRight.rotateAngleX = this.legRight.rotateAngleX
-                                                + MathHelper.cos(limbSwing * 0.6662F + (float) Math.PI) * 1.4F * limbSwingAmount;
+            this.legLeft.rotateAngleX += MathHelper.cos(limbSwing * 0.6662F) * 1.4F * limbSwingAmount;
+            this.legRight.rotateAngleX += MathHelper.cos(limbSwing * 0.6662F + (float) Math.PI) * 1.4F * limbSwingAmount;
         }
 
         this.head.rotationPointY = 15.69F + f;
@@ -174,10 +173,10 @@
                 this.wingLeft.rotationPointY = 18.84F;
                 this.wingRight.rotateAngleZ = 0.0873F;
                 this.wingRight.rotationPointY = 18.84F;
-                this.legLeft.rotationPointY++;
-                this.legRight.rotationPointY++;
-                this.legLeft.rotateAngleX++;
-                this.legRight.rotateAngleX++;
+                ++this.legLeft.rotationPointY;
+                ++this.legRight.rotationPointY;
+                ++this.legLeft.rotateAngleX;
+                ++this.legRight.rotateAngleX;
                 this.state = ModelParrot.State.SITTING;
             }
             else if (entityparrot.isFlying())
