--- before/net/minecraft/client/model/ModelIllager.java
+++ after/net/minecraft/client/model/ModelIllager.java
@@ -121,10 +121,10 @@
                 this.field_191224_h.field_78795_f += f * 2.2F - f1 * 0.4F;
             }
 
-            this.field_191223_g.field_78808_h = this.field_191223_g.field_78808_h + MathHelper.func_76134_b(p_78087_3_ * 0.09F) * 0.05F + 0.05F;
-            this.field_191224_h.field_78808_h = this.field_191224_h.field_78808_h - (MathHelper.func_76134_b(p_78087_3_ * 0.09F) * 0.05F + 0.05F);
-            this.field_191223_g.field_78795_f = this.field_191223_g.field_78795_f + MathHelper.func_76126_a(p_78087_3_ * 0.067F) * 0.05F;
-            this.field_191224_h.field_78795_f = this.field_191224_h.field_78795_f - MathHelper.func_76126_a(p_78087_3_ * 0.067F) * 0.05F;
+            this.field_191223_g.field_78808_h += MathHelper.func_76134_b(p_78087_3_ * 0.09F) * 0.05F + 0.05F;
+            this.field_191224_h.field_78808_h -= MathHelper.func_76134_b(p_78087_3_ * 0.09F) * 0.05F + 0.05F;
+            this.field_191223_g.field_78795_f += MathHelper.func_76126_a(p_78087_3_ * 0.067F) * 0.05F;
+            this.field_191224_h.field_78795_f -= MathHelper.func_76126_a(p_78087_3_ * 0.067F) * 0.05F;
         }
         else if (abstractillager$illagerarmpose == AbstractIllager.IllagerArmPose.SPELLCASTING)
         {
