--- before/net/minecraft/client/model/ModelLlama.java
+++ after/net/minecraft/client/model/ModelLlama.java
@@ -48,14 +48,14 @@
         this.leg4 = new ModelRenderer(this, 29, 29);
         this.leg4.addBox(-2.0F, 0.0F, -2.0F, 4, 14, 4, p_i47226_1_);
         this.leg4.setRotationPoint(2.5F, 10.0F, -4.0F);
-        this.leg1.rotationPointX--;
-        this.leg2.rotationPointX++;
+        --this.leg1.rotationPointX;
+        ++this.leg2.rotationPointX;
         this.leg1.rotationPointZ += 0.0F;
         this.leg2.rotationPointZ += 0.0F;
-        this.leg3.rotationPointX--;
-        this.leg4.rotationPointX++;
-        this.leg3.rotationPointZ--;
-        this.leg4.rotationPointZ--;
+        --this.leg3.rotationPointX;
+        ++this.leg4.rotationPointX;
+        --this.leg3.rotationPointZ;
+        --this.leg4.rotationPointZ;
         this.childZOffset += 2.0F;
     }
 
