--- before/net/minecraft/client/model/ModelWither.java
+++ after/net/minecraft/client/model/ModelWither.java
@@ -78,7 +78,7 @@
     {
         EntityWither entitywither = (EntityWither)entitylivingbaseIn;
 
-        for (int i = 1; i < 3; i++)
+        for (int i = 1; i < 3; ++i)
         {
             this.heads[i].rotateAngleY = (entitywither.getHeadYRotation(i - 1) - entitylivingbaseIn.renderYawOffset) * (float)(Math.PI / 180.0);
             this.heads[i].rotateAngleX = entitywither.getHeadXRotation(i - 1) * (float)(Math.PI / 180.0);
