--- before/net/minecraft/client/model/ModelIllager.java
+++ after/net/minecraft/client/model/ModelIllager.java
@@ -121,10 +121,10 @@
                 this.leftArm.rotateAngleX += f * 2.2F - f1 * 0.4F;
             }
 
-            this.rightArm.rotateAngleZ = this.rightArm.rotateAngleZ + MathHelper.cos(ageInTicks * 0.09F) * 0.05F + 0.05F;
-            this.leftArm.rotateAngleZ = this.leftArm.rotateAngleZ - (MathHelper.cos(ageInTicks * 0.09F) * 0.05F + 0.05F);
-            this.rightArm.rotateAngleX = this.rightArm.rotateAngleX + MathHelper.sin(ageInTicks * 0.067F) * 0.05F;
-            this.leftArm.rotateAngleX = this.leftArm.rotateAngleX - MathHelper.sin(ageInTicks * 0.067F) * 0.05F;
+            this.rightArm.rotateAngleZ += MathHelper.cos(ageInTicks * 0.09F) * 0.05F + 0.05F;
+            this.leftArm.rotateAngleZ -= MathHelper.cos(ageInTicks * 0.09F) * 0.05F + 0.05F;
+            this.rightArm.rotateAngleX += MathHelper.sin(ageInTicks * 0.067F) * 0.05F;
+            this.leftArm.rotateAngleX -= MathHelper.sin(ageInTicks * 0.067F) * 0.05F;
         }
         else if (abstractillager$illagerarmpose == AbstractIllager.IllagerArmPose.SPELLCASTING)
         {
