--- before/net/minecraft/scoreboard/ScoreboardSaveData.java
+++ after/net/minecraft/scoreboard/ScoreboardSaveData.java
@@ -60,7 +60,7 @@
 
     protected void readTeams(NBTTagList tagList)
     {
-        for (int i = 0; i < tagList.tagCount(); i++)
+        for (int i = 0; i < tagList.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = tagList.getCompoundTagAt(i);
             String s = nbttagcompound.getString("Name");
@@ -134,7 +134,7 @@
 
     protected void loadTeamPlayers(ScorePlayerTeam playerTeam, NBTTagList tagList)
     {
-        for (int i = 0; i < tagList.tagCount(); i++)
+        for (int i = 0; i < tagList.tagCount(); ++i)
         {
             this.scoreboard.addPlayerToTeam(tagList.getStringTagAt(i), playerTeam.getName());
         }
@@ -142,7 +142,7 @@
 
     protected void readDisplayConfig(NBTTagCompound compound)
     {
-        for (int i = 0; i < 19; i++)
+        for (int i = 0; i < 19; ++i)
         {
             if (compound.hasKey("slot_" + i, 8))
             {
@@ -155,7 +155,7 @@
 
     protected void readObjectives(NBTTagList nbt)
     {
-        for (int i = 0; i < nbt.tagCount(); i++)
+        for (int i = 0; i < nbt.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = nbt.getCompoundTagAt(i);
             IScoreCriteria iscorecriteria = IScoreCriteria.INSTANCES.get(nbttagcompound.getString("CriteriaName"));
@@ -178,7 +178,7 @@
 
     protected void readScores(NBTTagList nbt)
     {
-        for (int i = 0; i < nbt.tagCount(); i++)
+        for (int i = 0; i < nbt.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = nbt.getCompoundTagAt(i);
             ScoreObjective scoreobjective = this.scoreboard.getObjective(nbttagcompound.getString("Objective"));
@@ -258,7 +258,7 @@
         NBTTagCompound nbttagcompound = new NBTTagCompound();
         boolean flag = false;
 
-        for (int i = 0; i < 19; i++)
+        for (int i = 0; i < 19; ++i)
         {
             ScoreObjective scoreobjective = this.scoreboard.getObjectiveInDisplaySlot(i);
 
