--- before/net/minecraft/scoreboard/ServerScoreboard.java
+++ after/net/minecraft/scoreboard/ServerScoreboard.java
@@ -15,13 +15,13 @@
 
 public class ServerScoreboard extends Scoreboard
 {
-    private final MinecraftServer server;
+    private final MinecraftServer scoreboardMCServer;
     private final Set<ScoreObjective> addedObjectives = Sets.newHashSet();
     private Runnable[] dirtyRunnables = new Runnable[0];
 
     public ServerScoreboard(MinecraftServer mcServer)
     {
-        this.server = mcServer;
+        this.scoreboardMCServer = mcServer;
     }
 
     @Override
@@ -31,7 +31,7 @@
 
         if (this.addedObjectives.contains(scoreIn.getObjective()))
         {
-            this.server.getPlayerList().sendPacketToAllPlayers(new SPacketUpdateScore(scoreIn));
+            this.sendAll(new SPacketUpdateScore(scoreIn));
         }
 
         this.markSaveDataDirty();
@@ -41,7 +41,7 @@
     public void broadcastScoreUpdate(String scoreName)
     {
         super.broadcastScoreUpdate(scoreName);
-        this.server.getPlayerList().sendPacketToAllPlayers(new SPacketUpdateScore(scoreName));
+        this.sendAll(new SPacketUpdateScore(scoreName));
         this.markSaveDataDirty();
     }
 
@@ -49,7 +49,7 @@
     public void broadcastScoreUpdate(String scoreName, ScoreObjective objective)
     {
         super.broadcastScoreUpdate(scoreName, objective);
-        this.server.getPlayerList().sendPacketToAllPlayers(new SPacketUpdateScore(scoreName, objective));
+        this.sendAll(new SPacketUpdateScore(scoreName, objective));
         this.markSaveDataDirty();
     }
 
@@ -63,7 +63,7 @@
         {
             if (this.getObjectiveDisplaySlotCount(scoreobjective) > 0)
             {
-                this.server.getPlayerList().sendPacketToAllPlayers(new SPacketDisplayObjective(objectiveSlot, objective));
+                this.sendAll(new SPacketDisplayObjective(objectiveSlot, objective));
             }
             else
             {
@@ -75,7 +75,7 @@
         {
             if (this.addedObjectives.contains(objective))
             {
-                this.server.getPlayerList().sendPacketToAllPlayers(new SPacketDisplayObjective(objectiveSlot, objective));
+                this.sendAll(new SPacketDisplayObjective(objectiveSlot, objective));
             }
             else
             {
@@ -92,7 +92,7 @@
         if (super.addPlayerToTeam(player, newTeam))
         {
             ScorePlayerTeam scoreplayerteam = this.getTeam(newTeam);
-            this.server.getPlayerList().sendPacketToAllPlayers(new SPacketTeams(scoreplayerteam, Arrays.asList(player), 3));
+            this.sendAll(new SPacketTeams(scoreplayerteam, Arrays.asList(player), 3));
             this.markSaveDataDirty();
             return true;
         }
@@ -106,7 +106,7 @@
     public void removePlayerFromTeam(String username, ScorePlayerTeam playerTeam)
     {
         super.removePlayerFromTeam(username, playerTeam);
-        this.server.getPlayerList().sendPacketToAllPlayers(new SPacketTeams(playerTeam, Arrays.asList(username), 4));
+        this.sendAll(new SPacketTeams(playerTeam, Arrays.asList(username), 4));
         this.markSaveDataDirty();
     }
 
@@ -124,7 +124,7 @@
 
         if (this.addedObjectives.contains(objective))
         {
-            this.server.getPlayerList().sendPacketToAllPlayers(new SPacketScoreboardObjective(objective, 2));
+            this.sendAll(new SPacketScoreboardObjective(objective, 2));
         }
 
         this.markSaveDataDirty();
@@ -147,7 +147,7 @@
     public void broadcastTeamCreated(ScorePlayerTeam playerTeam)
     {
         super.broadcastTeamCreated(playerTeam);
-        this.server.getPlayerList().sendPacketToAllPlayers(new SPacketTeams(playerTeam, 0));
+        this.sendAll(new SPacketTeams(playerTeam, 0));
         this.markSaveDataDirty();
     }
 
@@ -155,7 +155,7 @@
     public void broadcastTeamInfoUpdate(ScorePlayerTeam playerTeam)
     {
         super.broadcastTeamInfoUpdate(playerTeam);
-        this.server.getPlayerList().sendPacketToAllPlayers(new SPacketTeams(playerTeam, 2));
+        this.sendAll(new SPacketTeams(playerTeam, 2));
         this.markSaveDataDirty();
     }
 
@@ -163,7 +163,7 @@
     public void broadcastTeamRemove(ScorePlayerTeam playerTeam)
     {
         super.broadcastTeamRemove(playerTeam);
-        this.server.getPlayerList().sendPacketToAllPlayers(new SPacketTeams(playerTeam, 1));
+        this.sendAll(new SPacketTeams(playerTeam, 1));
         this.markSaveDataDirty();
     }
 
@@ -186,7 +186,7 @@
         List < Packet<? >> list = Lists.newArrayList();
         list.add(new SPacketScoreboardObjective(objective, 0));
 
-        for (int i = 0; i < 19; i++)
+        for (int i = 0; i < 19; ++i)
         {
             if (this.getObjectiveInDisplaySlot(i) == objective)
             {
@@ -206,8 +206,9 @@
     {
         List < Packet<? >> list = this.getCreatePackets(objective);
 
-        for (EntityPlayerMP entityplayermp : this.server.getPlayerList().getPlayers())
+        for (EntityPlayerMP entityplayermp : this.scoreboardMCServer.getPlayerList().getPlayers())
         {
+            if (entityplayermp.getBukkitEntity().getScoreboard().getHandle() != this) continue; // CraftBukkit - Only players on this board
             for (Packet<?> packet : list)
             {
                 entityplayermp.connection.sendPacket(packet);
@@ -222,7 +223,7 @@
         List < Packet<? >> list = Lists.newArrayList();
         list.add(new SPacketScoreboardObjective(p_96548_1_, 1));
 
-        for (int i = 0; i < 19; i++)
+        for (int i = 0; i < 19; ++i)
         {
             if (this.getObjectiveInDisplaySlot(i) == p_96548_1_)
             {
@@ -237,8 +238,9 @@
     {
         List < Packet<? >> list = this.getDestroyPackets(p_96546_1_);
 
-        for (EntityPlayerMP entityplayermp : this.server.getPlayerList().getPlayers())
+        for (EntityPlayerMP entityplayermp : this.scoreboardMCServer.getPlayerList().getPlayers())
         {
+            if (entityplayermp.getBukkitEntity().getScoreboard().getHandle() != this) continue; // CraftBukkit - Only players on this board
             for (Packet<?> packet : list)
             {
                 entityplayermp.connection.sendPacket(packet);
@@ -252,14 +254,22 @@
     {
         int i = 0;
 
-        for (int j = 0; j < 19; j++)
+        for (int j = 0; j < 19; ++j)
         {
             if (this.getObjectiveInDisplaySlot(j) == p_96552_1_)
             {
-                i++;
+                ++i;
             }
         }
 
         return i;
+    }
+
+    private void sendAll(Packet packet) {
+        for (EntityPlayerMP entityplayer : this.scoreboardMCServer.getPlayerList().getPlayers()) {
+            if (entityplayer.getBukkitEntity().getScoreboard().getHandle() == this) {
+                entityplayer.connection.sendPacket(packet);
+            }
+        }
     }
 }
