--- before/net/minecraft/scoreboard/Team.java
+++ after/net/minecraft/scoreboard/Team.java
@@ -12,7 +12,14 @@
 {
     public boolean isSameTeam(@Nullable Team other)
     {
-        return other == null ? false : this == other;
+        if (other == null)
+        {
+            return false;
+        }
+        else
+        {
+            return this == other;
+        }
     }
 
     public abstract String getName();
