--- before/net/minecraft/item/ItemPickaxe.java
+++ after/net/minecraft/item/ItemPickaxe.java
@@ -81,9 +81,13 @@
             {
                 return true;
             }
+            else if (material == Material.IRON)
+            {
+                return true;
+            }
             else
             {
-                return material == Material.IRON ? true : material == Material.ANVIL;
+                return material == Material.ANVIL;
             }
         }
         else
