--- before/net/minecraft/item/ItemRecord.java
+++ after/net/minecraft/item/ItemRecord.java
@@ -26,6 +26,7 @@
     private static final Map<SoundEvent, ItemRecord> RECORDS = Maps.newHashMap();
     private final SoundEvent sound;
     private final String displayName;
+    public boolean playAfter = false;
 
     protected ItemRecord(String recordName, SoundEvent soundIn)
     {
@@ -56,7 +57,7 @@
             {
                 ItemStack itemstack = player.getHeldItem(hand);
                 ((BlockJukebox)Blocks.JUKEBOX).insertRecord(worldIn, pos, iblockstate, itemstack);
-                worldIn.playEvent(null, 1010, pos, Item.getIdFromItem(this));
+                playAfter = true; // CatServer - handle in ForgeHook
                 itemstack.shrink(1);
                 player.addStat(StatList.RECORD_PLAYED);
             }
