--- before/net/minecraft/item/ItemBanner.java
+++ after/net/minecraft/item/ItemBanner.java
@@ -121,7 +121,7 @@
         {
             NBTTagList nbttaglist = nbttagcompound.func_150295_c("Patterns", 10);
 
-            for (int i = 0; i < nbttaglist.func_74745_c() && i < 6; i++)
+            for (int i = 0; i < nbttaglist.func_74745_c() && i < 6; ++i)
             {
                 NBTTagCompound nbttagcompound1 = nbttaglist.func_150305_b(i);
                 EnumDyeColor enumdyecolor = EnumDyeColor.func_176766_a(nbttagcompound1.func_74762_e("Color"));
