--- before/net/minecraft/item/ItemSpade.java
+++ after/net/minecraft/item/ItemSpade.java
@@ -41,7 +41,15 @@
     public boolean canHarvestBlock(IBlockState blockIn)
     {
         Block block = blockIn.getBlock();
-        return block == Blocks.SNOW_LAYER ? true : block == Blocks.SNOW;
+
+        if (block == Blocks.SNOW_LAYER)
+        {
+            return true;
+        }
+        else
+        {
+            return block == Blocks.SNOW;
+        }
     }
 
     @Override
