--- before/net/minecraft/item/ItemKnowledgeBook.java
+++ after/net/minecraft/item/ItemKnowledgeBook.java
@@ -43,7 +43,7 @@
                 NBTTagList nbttaglist = nbttagcompound.getTagList("Recipes", 8);
                 List<IRecipe> list = Lists.newArrayList();
 
-                for (int i = 0; i < nbttaglist.tagCount(); i++)
+                for (int i = 0; i < nbttaglist.tagCount(); ++i)
                 {
                     String s = nbttaglist.getStringTagAt(i);
                     IRecipe irecipe = CraftingManager.getRecipe(new ResourceLocation(s));
