--- before/net/minecraft/item/ItemEnchantedBook.java
+++ after/net/minecraft/item/ItemEnchantedBook.java
@@ -48,7 +48,7 @@
         super.addInformation(stack, worldIn, tooltip, flagIn);
         NBTTagList nbttaglist = getEnchantments(stack);
 
-        for (int i = 0; i < nbttaglist.tagCount(); i++)
+        for (int i = 0; i < nbttaglist.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
             int j = nbttagcompound.getShort("id");
@@ -66,7 +66,7 @@
         NBTTagList nbttaglist = getEnchantments(p_92115_0_);
         boolean flag = true;
 
-        for (int i = 0; i < nbttaglist.tagCount(); i++)
+        for (int i = 0; i < nbttaglist.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
 
@@ -114,7 +114,7 @@
             {
                 if (enchantment.type != null)
                 {
-                    for (int i = enchantment.getMinLevel(); i <= enchantment.getMaxLevel(); i++)
+                    for (int i = enchantment.getMinLevel(); i <= enchantment.getMaxLevel(); ++i)
                     {
                         items.add(getEnchantedItemStack(new EnchantmentData(enchantment, i)));
                     }
