--- before/net/minecraft/item/ItemFireworkCharge.java
+++ after/net/minecraft/item/ItemFireworkCharge.java
@@ -74,7 +74,7 @@
                 flag = false;
                 boolean flag1 = false;
 
-                for (int j = 0; j < ItemDye.DYE_COLORS.length; j++)
+                for (int j = 0; j < ItemDye.DYE_COLORS.length; ++j)
                 {
                     if (i == ItemDye.DYE_COLORS[j])
                     {
@@ -110,7 +110,7 @@
                 flag2 = false;
                 boolean flag5 = false;
 
-                for (int k = 0; k < 16; k++)
+                for (int k = 0; k < 16; ++k)
                 {
                     if (l == ItemDye.DYE_COLORS[k])
                     {
