--- before/net/minecraft/world/gen/layer/GenLayerShore.java
+++ after/net/minecraft/world/gen/layer/GenLayerShore.java
@@ -19,9 +19,9 @@
         int[] aint = this.parent.getInts(areaX - 1, areaY - 1, areaWidth + 2, areaHeight + 2);
         int[] aint1 = IntCache.getIntCache(areaWidth * areaHeight);
 
-        for (int i = 0; i < areaHeight; i++)
+        for (int i = 0; i < areaHeight; ++i)
         {
-            for (int j = 0; j < areaWidth; j++)
+            for (int j = 0; j < areaWidth; ++j)
             {
                 this.initChunkSeed((long)(j + areaX), (long)(i + areaY));
                 int k = aint[j + 1 + (i + 1) * (areaWidth + 2)];
@@ -151,14 +151,19 @@
 
     private boolean isJungleCompatible(int p_151631_1_)
     {
-        return Biome.getBiome(p_151631_1_) != null && Biome.getBiome(p_151631_1_).getBiomeClass() == BiomeJungle.class
-               ? true
-               : p_151631_1_ == Biome.getIdForBiome(Biomes.JUNGLE_EDGE)
-               || p_151631_1_ == Biome.getIdForBiome(Biomes.JUNGLE)
-               || p_151631_1_ == Biome.getIdForBiome(Biomes.JUNGLE_HILLS)
-               || p_151631_1_ == Biome.getIdForBiome(Biomes.FOREST)
-               || p_151631_1_ == Biome.getIdForBiome(Biomes.TAIGA)
-               || isBiomeOceanic(p_151631_1_);
+        if (Biome.getBiome(p_151631_1_) != null && Biome.getBiome(p_151631_1_).getBiomeClass() == BiomeJungle.class)
+        {
+            return true;
+        }
+        else
+        {
+            return p_151631_1_ == Biome.getIdForBiome(Biomes.JUNGLE_EDGE)
+                   || p_151631_1_ == Biome.getIdForBiome(Biomes.JUNGLE)
+                   || p_151631_1_ == Biome.getIdForBiome(Biomes.JUNGLE_HILLS)
+                   || p_151631_1_ == Biome.getIdForBiome(Biomes.FOREST)
+                   || p_151631_1_ == Biome.getIdForBiome(Biomes.TAIGA)
+                   || isBiomeOceanic(p_151631_1_);
+        }
     }
 
     private boolean isMesa(int p_151633_1_)
