--- before/net/minecraft/world/gen/layer/GenLayerDeepOcean.java
+++ after/net/minecraft/world/gen/layer/GenLayerDeepOcean.java
@@ -21,9 +21,9 @@
         int[] aint = this.parent.getInts(i, j, k, l);
         int[] aint1 = IntCache.getIntCache(areaWidth * areaHeight);
 
-        for (int i1 = 0; i1 < areaHeight; i1++)
+        for (int i1 = 0; i1 < areaHeight; ++i1)
         {
-            for (int j1 = 0; j1 < areaWidth; j1++)
+            for (int j1 = 0; j1 < areaWidth; ++j1)
             {
                 int k1 = aint[j1 + 1 + (i1 + 1 - 1) * (areaWidth + 2)];
                 int l1 = aint[j1 + 1 + 1 + (i1 + 1) * (areaWidth + 2)];
@@ -34,22 +34,22 @@
 
                 if (k1 == 0)
                 {
-                    l2++;
+                    ++l2;
                 }
 
                 if (l1 == 0)
                 {
-                    l2++;
+                    ++l2;
                 }
 
                 if (i2 == 0)
                 {
-                    l2++;
+                    ++l2;
                 }
 
                 if (j2 == 0)
                 {
-                    l2++;
+                    ++l2;
                 }
 
                 if (k2 == 0 && l2 > 3)
