--- before/net/minecraft/world/gen/layer/GenLayerZoom.java
+++ after/net/minecraft/world/gen/layer/GenLayerZoom.java
@@ -20,13 +20,13 @@
         int j1 = l - 1 << 1;
         int[] aint1 = IntCache.func_76445_a(i1 * j1);
 
-        for (int k1 = 0; k1 < l - 1; k1++)
+        for (int k1 = 0; k1 < l - 1; ++k1)
         {
             int l1 = (k1 << 1) * i1;
             int i2 = 0;
             int j2 = aint[i2 + 0 + (k1 + 0) * k];
 
-            for (int k2 = aint[i2 + 0 + (k1 + 1) * k]; i2 < k - 1; i2++)
+            for (int k2 = aint[i2 + 0 + (k1 + 1) * k]; i2 < k - 1; ++i2)
             {
                 this.func_75903_a((long)(i2 + i << 1), (long)(k1 + j << 1));
                 int l2 = aint[i2 + 1 + (k1 + 0) * k];
@@ -42,7 +42,7 @@
 
         int[] aint2 = IntCache.func_76445_a(p_75904_3_ * p_75904_4_);
 
-        for (int j3 = 0; j3 < p_75904_4_; j3++)
+        for (int j3 = 0; j3 < p_75904_4_; ++j3)
         {
             System.arraycopy(aint1, (j3 + (p_75904_2_ & 1)) * i1 + (p_75904_1_ & 1), aint2, j3 * p_75904_3_, p_75904_3_);
         }
@@ -54,7 +54,7 @@
     {
         GenLayer genlayer = p_75915_2_;
 
-        for (int i = 0; i < p_75915_3_; i++)
+        for (int i = 0; i < p_75915_3_; ++i)
         {
             genlayer = new GenLayerZoom(p_75915_0_ + (long)i, genlayer);
         }
