--- before/net/minecraft/world/gen/layer/IntCache.java
+++ after/net/minecraft/world/gen/layer/IntCache.java
@@ -23,7 +23,7 @@
             }
             else
             {
-                int[] aint3 = field_76449_b.remove(field_76449_b.size() - 1);
+                int[] aint3 = (int[])field_76449_b.remove(field_76449_b.size() - 1);
                 field_76450_c.add(aint3);
                 return aint3;
             }
@@ -45,7 +45,7 @@
         }
         else
         {
-            int[] aint = field_76447_d.remove(field_76447_d.size() - 1);
+            int[] aint = (int[])field_76447_d.remove(field_76447_d.size() - 1);
             field_76448_e.add(aint);
             return aint;
         }
