--- before/net/minecraft/world/gen/structure/MapGenScatteredFeature.java
+++ after/net/minecraft/world/gen/structure/MapGenScatteredFeature.java
@@ -64,11 +64,11 @@
 
         int k = chunkX / this.maxDistanceBetweenScatteredFeatures;
         int l = chunkZ / this.maxDistanceBetweenScatteredFeatures;
-        Random random = this.world.setRandomSeed(k, l, 14357617);
-        k *= this.maxDistanceBetweenScatteredFeatures;
-        l *= this.maxDistanceBetweenScatteredFeatures;
-        k += random.nextInt(this.maxDistanceBetweenScatteredFeatures - 8);
-        l += random.nextInt(this.maxDistanceBetweenScatteredFeatures - 8);
+        Random random = this.world.setRandomSeed(k, l, this.world.spigotConfig.largeFeatureSeed); // Spigot
+        k = k * this.maxDistanceBetweenScatteredFeatures;
+        l = l * this.maxDistanceBetweenScatteredFeatures;
+        k = k + random.nextInt(this.maxDistanceBetweenScatteredFeatures - 8);
+        l = l + random.nextInt(this.maxDistanceBetweenScatteredFeatures - 8);
 
         if (i == k && j == l)
         {
