--- before/net/minecraft/world/gen/structure/MapGenNetherBridge.java
+++ after/net/minecraft/world/gen/structure/MapGenNetherBridge.java
@@ -48,9 +48,13 @@
         {
             return false;
         }
+        else if (chunkX != (i << 4) + 4 + this.rand.nextInt(8))
+        {
+            return false;
+        }
         else
         {
-            return chunkX != (i << 4) + 4 + this.rand.nextInt(8) ? false : chunkZ == (j << 4) + 4 + this.rand.nextInt(8);
+            return chunkZ == (j << 4) + 4 + this.rand.nextInt(8);
         }
     }
 
@@ -67,13 +71,13 @@
         int j = pos.getX() >> 4;
         int k = pos.getZ() >> 4;
 
-        for (int l = 0; l <= 1000; l++)
+        for (int l = 0; l <= 1000; ++l)
         {
-            for (int i1 = -l; i1 <= l; i1++)
+            for (int i1 = -l; i1 <= l; ++i1)
             {
                 boolean flag = i1 == -l || i1 == l;
 
-                for (int j1 = -l; j1 <= l; j1++)
+                for (int j1 = -l; j1 <= l; ++j1)
                 {
                     boolean flag1 = j1 == -l || j1 == l;
 
