--- before/net/minecraft/world/gen/structure/StructureMineshaftPieces.java
+++ after/net/minecraft/world/gen/structure/StructureMineshaftPieces.java
@@ -178,7 +178,7 @@
                 StructureBoundingBox structureboundingbox = new StructureBoundingBox(x, y, z, x, y + 2, z);
                 int i;
 
-                for (i = rand.nextInt(3) + 2; i > 0; i--)
+                for (i = rand.nextInt(3) + 2; i > 0; --i)
                 {
                     int j = i * 5;
 
@@ -535,7 +535,7 @@
                         );
                     }
 
-                    for (int j1 = 0; j1 < this.sectionCount; j1++)
+                    for (int j1 = 0; j1 < this.sectionCount; ++j1)
                     {
                         int k1 = 2 + j1 * 5;
                         this.placeSupport(worldIn, structureBoundingBoxIn, 0, 0, k1, 2, 2, randomIn);
@@ -580,9 +580,9 @@
                         }
                     }
 
-                    for (int l2 = 0; l2 <= 2; l2++)
+                    for (int l2 = 0; l2 <= 2; ++l2)
                     {
-                        for (int i3 = 0; i3 <= i1; i3++)
+                        for (int i3 = 0; i3 <= i1; ++i3)
                         {
                             int k3 = -1;
                             IBlockState iblockstate3 = this.getBlockStateFromPos(worldIn, l2, -1, i3, structureBoundingBoxIn);
@@ -601,7 +601,7 @@
                                                    .getDefaultState()
                                                    .withProperty(BlockRail.SHAPE, BlockRailBase.EnumRailDirection.NORTH_SOUTH);
 
-                        for (int j3 = 0; j3 <= i1; j3++)
+                        for (int j3 = 0; j3 <= i1; ++j3)
                         {
                             IBlockState iblockstate2 = this.getBlockStateFromPos(worldIn, 1, -1, j3, structureBoundingBoxIn);
 
@@ -1099,9 +1099,9 @@
                         this.boundingBox.maxY
                     );
 
-                    for (int i = this.boundingBox.minX; i <= this.boundingBox.maxX; i++)
+                    for (int i = this.boundingBox.minX; i <= this.boundingBox.maxX; ++i)
                     {
-                        for (int j = this.boundingBox.minZ; j <= this.boundingBox.maxZ; j++)
+                        for (int j = this.boundingBox.minZ; j <= this.boundingBox.maxZ; ++j)
                         {
                             if (this.getBlockStateFromPos(worldIn, i, this.boundingBox.minY - 1, j, structureBoundingBoxIn).getMaterial() == Material.AIR
                                     && this.getSkyBrightness(worldIn, i, this.boundingBox.minY - 1, j, structureBoundingBoxIn) < 8)
@@ -1188,7 +1188,7 @@
 
             protected boolean isSupportingBox(World p_189918_1_, StructureBoundingBox p_189918_2_, int p_189918_3_, int p_189918_4_, int p_189918_5_, int p_189918_6_)
             {
-                for (int i = p_189918_3_; i <= p_189918_4_; i++)
+                for (int i = p_189918_3_; i <= p_189918_4_; ++i)
                 {
                     if (this.getBlockStateFromPos(p_189918_1_, i, p_189918_5_ + 1, p_189918_6_, p_189918_2_).getMaterial() == Material.AIR)
                     {
@@ -1221,20 +1221,20 @@
             public void buildComponent(StructureComponent componentIn, List<StructureComponent> listIn, Random rand)
             {
                 int i = this.getComponentType();
-                int k = this.boundingBox.getYSize() - 3 - 1;
+                int j = this.boundingBox.getYSize() - 3 - 1;
 
-                if (k <= 0)
+                if (j <= 0)
                 {
-                    k = 1;
+                    j = 1;
                 }
 
-                int j = 0;
+                int k;
 
-                while (j < this.boundingBox.getXSize())
+                for (k = 0; k < this.boundingBox.getXSize(); k += 4)
                 {
-                    j += rand.nextInt(this.boundingBox.getXSize());
+                    k += rand.nextInt(this.boundingBox.getXSize());
 
-                    if (j + 3 > this.boundingBox.getXSize())
+                    if (k + 3 > this.boundingBox.getXSize())
                     {
                         break;
                     }
@@ -1243,8 +1243,8 @@
                                 componentIn,
                                 listIn,
                                 rand,
-                                this.boundingBox.minX + j,
-                                this.boundingBox.minY + rand.nextInt(k) + 1,
+                                this.boundingBox.minX + k,
+                                this.boundingBox.minY + rand.nextInt(j) + 1,
                                 this.boundingBox.minZ - 1,
                                 EnumFacing.NORTH,
                                 i
@@ -1265,17 +1265,13 @@
                             )
                         );
                     }
-
-                    j += 4;
                 }
 
-                j = 0;
-
-                while (j < this.boundingBox.getXSize())
+                for (k = 0; k < this.boundingBox.getXSize(); k += 4)
                 {
-                    j += rand.nextInt(this.boundingBox.getXSize());
+                    k += rand.nextInt(this.boundingBox.getXSize());
 
-                    if (j + 3 > this.boundingBox.getXSize())
+                    if (k + 3 > this.boundingBox.getXSize())
                     {
                         break;
                     }
@@ -1284,8 +1280,8 @@
                                 componentIn,
                                 listIn,
                                 rand,
-                                this.boundingBox.minX + j,
-                                this.boundingBox.minY + rand.nextInt(k) + 1,
+                                this.boundingBox.minX + k,
+                                this.boundingBox.minY + rand.nextInt(j) + 1,
                                 this.boundingBox.maxZ + 1,
                                 EnumFacing.SOUTH,
                                 i
@@ -1306,17 +1302,13 @@
                             )
                         );
                     }
-
-                    j += 4;
                 }
 
-                j = 0;
-
-                while (j < this.boundingBox.getZSize())
+                for (k = 0; k < this.boundingBox.getZSize(); k += 4)
                 {
-                    j += rand.nextInt(this.boundingBox.getZSize());
+                    k += rand.nextInt(this.boundingBox.getZSize());
 
-                    if (j + 3 > this.boundingBox.getZSize())
+                    if (k + 3 > this.boundingBox.getZSize())
                     {
                         break;
                     }
@@ -1326,8 +1318,8 @@
                                 listIn,
                                 rand,
                                 this.boundingBox.minX - 1,
-                                this.boundingBox.minY + rand.nextInt(k) + 1,
-                                this.boundingBox.minZ + j,
+                                this.boundingBox.minY + rand.nextInt(j) + 1,
+                                this.boundingBox.minZ + k,
                                 EnumFacing.WEST,
                                 i
                             );
@@ -1347,17 +1339,13 @@
                             )
                         );
                     }
-
-                    j += 4;
                 }
 
-                j = 0;
-
-                while (j < this.boundingBox.getZSize())
+                for (k = 0; k < this.boundingBox.getZSize(); k += 4)
                 {
-                    j += rand.nextInt(this.boundingBox.getZSize());
+                    k += rand.nextInt(this.boundingBox.getZSize());
 
-                    if (j + 3 > this.boundingBox.getZSize())
+                    if (k + 3 > this.boundingBox.getZSize())
                     {
                         break;
                     }
@@ -1367,8 +1355,8 @@
                                                                 listIn,
                                                                 rand,
                                                                 this.boundingBox.maxX + 1,
-                                                                this.boundingBox.minY + rand.nextInt(k) + 1,
-                                                                this.boundingBox.minZ + j,
+                                                                this.boundingBox.minY + rand.nextInt(j) + 1,
+                                                                this.boundingBox.minZ + k,
                                                                 EnumFacing.EAST,
                                                                 i
                                                             );
@@ -1388,8 +1376,6 @@
                             )
                         );
                     }
-
-                    j += 4;
                 }
             }
 
@@ -1493,7 +1479,7 @@
                 super.readStructureFromNBT(tagCompound, p_143011_2_);
                 NBTTagList nbttaglist = tagCompound.getTagList("Entrances", 11);
 
-                for (int i = 0; i < nbttaglist.tagCount(); i++)
+                for (int i = 0; i < nbttaglist.tagCount(); ++i)
                 {
                     this.connectedRooms.add(new StructureBoundingBox(nbttaglist.getIntArrayAt(i)));
                 }
@@ -1618,7 +1604,7 @@
                     this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 5, 0, 2, 7, 1, Blocks.AIR.getDefaultState(), Blocks.AIR.getDefaultState(), false);
                     this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 0, 7, 2, 2, 8, Blocks.AIR.getDefaultState(), Blocks.AIR.getDefaultState(), false);
 
-                    for (int i = 0; i < 5; i++)
+                    for (int i = 0; i < 5; ++i)
                     {
                         this.fillWithBlocks(
                             worldIn,
