--- before/net/minecraft/world/gen/structure/StructureOceanMonument.java
+++ after/net/minecraft/world/gen/structure/StructureOceanMonument.java
@@ -203,7 +203,7 @@
                 {
                     NBTTagList nbttaglist = tagCompound.getTagList("Processed", 10);
 
-                    for (int i = 0; i < nbttaglist.tagCount(); i++)
+                    for (int i = 0; i < nbttaglist.tagCount(); ++i)
                     {
                         NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
                         this.processed.add(new ChunkPos(nbttagcompound.getInteger("X"), nbttagcompound.getInteger("Z")));
