--- before/net/minecraft/world/gen/structure/StructureNetherBridgePieces.java
+++ after/net/minecraft/world/gen/structure/StructureNetherBridgePieces.java
@@ -223,9 +223,9 @@
 
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i = 0; i <= 4; i++)
+                for (int i = 0; i <= 4; ++i)
                 {
-                    for (int j = 0; j <= 4; j++)
+                    for (int j = 0; j <= 4; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
@@ -302,9 +302,9 @@
 
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i = 0; i <= 4; i++)
+                for (int i = 0; i <= 4; ++i)
                 {
-                    for (int j = 0; j <= 4; j++)
+                    for (int j = 0; j <= 4; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
@@ -350,7 +350,7 @@
             {
                 IBlockState iblockstate = Blocks.NETHER_BRICK_STAIRS.getDefaultState().withProperty(BlockStairs.FACING, EnumFacing.SOUTH);
 
-                for (int i = 0; i <= 9; i++)
+                for (int i = 0; i <= 9; ++i)
                 {
                     int j = Math.max(1, 7 - i);
                     int k = Math.min(Math.max(j + 5, 14 - i), 13);
@@ -385,7 +385,7 @@
                         );
                     }
 
-                    for (int i1 = 0; i1 <= 4; i1++)
+                    for (int i1 = 0; i1 <= 4; ++i1)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i1, -1, l, structureBoundingBoxIn);
                     }
@@ -456,9 +456,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 1, 4, 5, 1, 5, 5, Blocks.NETHER_BRICK_FENCE.getDefaultState(), Blocks.NETHER_BRICK_FENCE.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 7, 4, 5, 7, 5, 5, Blocks.NETHER_BRICK_FENCE.getDefaultState(), Blocks.NETHER_BRICK_FENCE.getDefaultState(), false);
 
-                for (int i = 0; i <= 5; i++)
+                for (int i = 0; i <= 5; ++i)
                 {
-                    for (int j = 0; j <= 8; j++)
+                    for (int j = 0; j <= 8; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), j, -1, i, structureBoundingBoxIn);
                     }
@@ -510,9 +510,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 4, 3, 3, 4, 4, 3, Blocks.NETHER_BRICK_FENCE.getDefaultState(), Blocks.NETHER_BRICK_FENCE.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i = 0; i <= 4; i++)
+                for (int i = 0; i <= 4; ++i)
                 {
-                    for (int j = 0; j <= 4; j++)
+                    for (int j = 0; j <= 4; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
@@ -575,9 +575,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 6, 6, 2, 6, 6, 4, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 6, 5, 2, 6, 5, 4, Blocks.NETHER_BRICK_FENCE.getDefaultState(), Blocks.NETHER_BRICK_FENCE.getDefaultState(), false);
 
-                for (int i = 0; i <= 6; i++)
+                for (int i = 0; i <= 6; ++i)
                 {
-                    for (int j = 0; j <= 6; j++)
+                    for (int j = 0; j <= 6; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
@@ -629,9 +629,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 4, 2, 4, 4, 5, 4, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 6, 0, 4, 6, 4, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i = 0; i <= 4; i++)
+                for (int i = 0; i <= 4; ++i)
                 {
-                    for (int j = 0; j <= 4; j++)
+                    for (int j = 0; j <= 4; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
@@ -709,9 +709,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 7, 0, 0, 11, 1, 3, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 7, 0, 15, 11, 1, 18, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i = 7; i <= 11; i++)
+                for (int i = 7; i <= 11; ++i)
                 {
-                    for (int j = 0; j <= 2; j++)
+                    for (int j = 0; j <= 2; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, 18 - j, structureBoundingBoxIn);
@@ -723,9 +723,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 0, 7, 3, 1, 11, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 15, 0, 7, 18, 1, 11, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int k = 0; k <= 2; k++)
+                for (int k = 0; k <= 2; ++k)
                 {
-                    for (int l = 7; l <= 11; l++)
+                    for (int l = 7; l <= 11; ++l)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), k, -1, l, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), 18 - k, -1, l, structureBoundingBoxIn);
@@ -781,9 +781,9 @@
             {
                 Random random = new Random((long)this.fillSeed);
 
-                for (int i = 0; i <= 4; i++)
+                for (int i = 0; i <= 4; ++i)
                 {
-                    for (int j = 3; j <= 4; j++)
+                    for (int j = 3; j <= 4; ++j)
                     {
                         int k = random.nextInt(8);
                         this.fillWithBlocks(
@@ -797,7 +797,7 @@
                 l = random.nextInt(8);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 4, 5, 0, 4, 5, l, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i1 = 0; i1 <= 4; i1++)
+                for (int i1 = 0; i1 <= 4; ++i1)
                 {
                     int k1 = random.nextInt(5);
                     this.fillWithBlocks(
@@ -805,9 +805,9 @@
                     );
                 }
 
-                for (int j1 = 0; j1 <= 4; j1++)
+                for (int j1 = 0; j1 <= 4; ++j1)
                 {
-                    for (int l1 = 0; l1 <= 1; l1++)
+                    for (int l1 = 0; l1 <= 1; ++l1)
                     {
                         int i2 = random.nextInt(3);
                         this.fillWithBlocks(
@@ -911,18 +911,18 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 0, 4, 3, 1, 8, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 9, 0, 4, 12, 1, 8, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int l = 4; l <= 8; l++)
+                for (int l = 4; l <= 8; ++l)
                 {
-                    for (int j = 0; j <= 2; j++)
+                    for (int j = 0; j <= 2; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), l, -1, j, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), l, -1, 12 - j, structureBoundingBoxIn);
                     }
                 }
 
-                for (int i1 = 0; i1 <= 2; i1++)
+                for (int i1 = 0; i1 <= 2; ++i1)
                 {
-                    for (int j1 = 4; j1 <= 8; j1++)
+                    for (int j1 = 4; j1 <= 8; ++j1)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i1, -1, j1, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), 12 - i1, -1, j1, structureBoundingBoxIn);
@@ -1033,11 +1033,11 @@
 
                 IBlockState iblockstate = Blocks.NETHER_BRICK_STAIRS.getDefaultState().withProperty(BlockStairs.FACING, EnumFacing.NORTH);
 
-                for (int j = 0; j <= 6; j++)
+                for (int j = 0; j <= 6; ++j)
                 {
                     int k = j + 4;
 
-                    for (int l = 5; l <= 7; l++)
+                    for (int l = 5; l <= 7; ++l)
                     {
                         this.setBlockState(worldIn, iblockstate, l, 5 + j, k, structureBoundingBoxIn);
                     }
@@ -1063,7 +1063,7 @@
                     }
                 }
 
-                for (int k1 = 5; k1 <= 7; k1++)
+                for (int k1 = 5; k1 <= 7; ++k1)
                 {
                     this.setBlockState(worldIn, iblockstate, k1, 12, 11, structureBoundingBoxIn);
                 }
@@ -1098,18 +1098,18 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 0, 4, 3, 1, 8, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 9, 0, 4, 12, 1, 8, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int l1 = 4; l1 <= 8; l1++)
+                for (int l1 = 4; l1 <= 8; ++l1)
                 {
-                    for (int i1 = 0; i1 <= 2; i1++)
+                    for (int i1 = 0; i1 <= 2; ++i1)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), l1, -1, i1, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), l1, -1, 12 - i1, structureBoundingBoxIn);
                     }
                 }
 
-                for (int i2 = 0; i2 <= 2; i2++)
+                for (int i2 = 0; i2 <= 2; ++i2)
                 {
-                    for (int j2 = 4; j2 <= 8; j2++)
+                    for (int j2 = 4; j2 <= 8; ++j2)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i2, -1, j2, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), 12 - i2, -1, j2, structureBoundingBoxIn);
@@ -1178,7 +1178,7 @@
 
                 while (j < 5 && flag)
                 {
-                    j++;
+                    ++j;
                     int k = p_175871_4_.nextInt(i);
 
                     for (StructureNetherBridgePieces.PieceWeight structurenetherbridgepieces$pieceweight : p_175871_2_)
@@ -1199,7 +1199,7 @@
 
                             if (structurenetherbridgepieces$piece != null)
                             {
-                                structurenetherbridgepieces$pieceweight.placeCount++;
+                                ++structurenetherbridgepieces$pieceweight.placeCount;
                                 p_175871_1_.lastPlaced = structurenetherbridgepieces$pieceweight;
 
                                 if (!structurenetherbridgepieces$pieceweight.isValid())
@@ -1556,9 +1556,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 2, 6, 0, 4, 8, 0, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 2, 5, 0, 4, 5, 0, Blocks.NETHER_BRICK_FENCE.getDefaultState(), Blocks.NETHER_BRICK_FENCE.getDefaultState(), false);
 
-                for (int i = 0; i <= 6; i++)
+                for (int i = 0; i <= 6; ++i)
                 {
-                    for (int j = 0; j <= 6; j++)
+                    for (int j = 0; j <= 6; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
@@ -1643,9 +1643,9 @@
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 0, 0, 4, 1, 3, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
                 this.fillWithBlocks(worldIn, structureBoundingBoxIn, 0, 0, 15, 4, 1, 18, Blocks.NETHER_BRICK.getDefaultState(), Blocks.NETHER_BRICK.getDefaultState(), false);
 
-                for (int i = 0; i <= 4; i++)
+                for (int i = 0; i <= 4; ++i)
                 {
-                    for (int j = 0; j <= 2; j++)
+                    for (int j = 0; j <= 2; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, 18 - j, structureBoundingBoxIn);
@@ -1742,9 +1742,9 @@
                     }
                 }
 
-                for (int i = 0; i <= 6; i++)
+                for (int i = 0; i <= 6; ++i)
                 {
-                    for (int j = 0; j <= 6; j++)
+                    for (int j = 0; j <= 6; ++j)
                     {
                         this.replaceAirAndLiquidDownwards(worldIn, Blocks.NETHER_BRICK.getDefaultState(), i, -1, j, structureBoundingBoxIn);
                     }
