--- before/net/minecraft/world/gen/structure/WoodlandMansion.java
+++ after/net/minecraft/world/gen/structure/WoodlandMansion.java
@@ -148,9 +148,9 @@
                 super.generateStructure(worldIn, rand, structurebb);
                 int i = this.boundingBox.minY;
 
-                for (int j = structurebb.minX; j <= structurebb.maxX; j++)
+                for (int j = structurebb.minX; j <= structurebb.maxX; ++j)
                 {
-                    for (int k = structurebb.minZ; k <= structurebb.maxZ; k++)
+                    for (int k = structurebb.minZ; k <= structurebb.maxZ; ++k)
                     {
                         BlockPos blockpos = new BlockPos(j, i, k);
 
@@ -169,7 +169,7 @@
 
                             if (flag)
                             {
-                                for (int l = i - 1; l > 1; l--)
+                                for (int l = i - 1; l > 1; --l)
                                 {
                                     BlockPos blockpos1 = new BlockPos(j, l, k);
 
