--- before/net/minecraft/world/gen/structure/WoodlandMansionPieces.java
+++ after/net/minecraft/world/gen/structure/WoodlandMansionPieces.java
@@ -167,7 +167,7 @@
                     p_191110_1_.set(p_191110_2_, p_191110_3_, 1);
                     p_191110_1_.setIf(p_191110_2_ + p_191110_4_.getXOffset(), p_191110_3_ + p_191110_4_.getZOffset(), 0, 1);
 
-                    for (int i = 0; i < 8; i++)
+                    for (int i = 0; i < 8; ++i)
                     {
                         EnumFacing enumfacing = EnumFacing.byHorizontalIndex(this.random.nextInt(4));
 
@@ -217,9 +217,9 @@
             {
                 boolean flag = false;
 
-                for (int i = 0; i < p_191111_1_.height; i++)
+                for (int i = 0; i < p_191111_1_.height; ++i)
                 {
-                    for (int j = 0; j < p_191111_1_.width; j++)
+                    for (int j = 0; j < p_191111_1_.width; ++j)
                     {
                         if (p_191111_1_.get(j, i) == 0)
                         {
@@ -260,9 +260,9 @@
                 List<Tuple<Integer, Integer>> list = Lists.newArrayList();
                 WoodlandMansionPieces.SimpleGrid woodlandmansionpieces$simplegrid = this.floorRooms[1];
 
-                for (int i = 0; i < this.thirdFloorGrid.height; i++)
+                for (int i = 0; i < this.thirdFloorGrid.height; ++i)
                 {
-                    for (int j = 0; j < this.thirdFloorGrid.width; j++)
+                    for (int j = 0; j < this.thirdFloorGrid.width; ++j)
                     {
                         int k = woodlandmansionpieces$simplegrid.get(j, i);
                         int l = k & 983040;
@@ -287,9 +287,9 @@
                     int i2 = tuple.getFirst() + enumfacing1.getXOffset();
                     int i1 = tuple.getSecond() + enumfacing1.getZOffset();
 
-                    for (int j1 = 0; j1 < this.thirdFloorGrid.height; j1++)
+                    for (int j1 = 0; j1 < this.thirdFloorGrid.height; ++j1)
                     {
-                        for (int k1 = 0; k1 < this.thirdFloorGrid.width; k1++)
+                        for (int k1 = 0; k1 < this.thirdFloorGrid.width; ++k1)
                         {
                             if (!isHouse(this.baseGrid, k1, j1))
                             {
@@ -338,9 +338,9 @@
             {
                 List<Tuple<Integer, Integer>> list = Lists.newArrayList();
 
-                for (int i = 0; i < p_191116_1_.height; i++)
+                for (int i = 0; i < p_191116_1_.height; ++i)
                 {
-                    for (int j = 0; j < p_191116_1_.width; j++)
+                    for (int j = 0; j < p_191116_1_.width; ++j)
                     {
                         if (p_191116_1_.get(j, i) == 2)
                         {
@@ -447,9 +447,9 @@
                             }
                         }
 
-                        for (int i3 = k1; i3 <= l1; i3++)
+                        for (int i3 = k1; i3 <= l1; ++i3)
                         {
-                            for (int j3 = i1; j3 <= j1; j3++)
+                            for (int j3 = i1; j3 <= j1; ++j3)
                             {
                                 if (j3 == j2 && i3 == k2)
                                 {
@@ -462,7 +462,7 @@
                             }
                         }
 
-                        k3++;
+                        ++k3;
                     }
                 }
             }
@@ -643,9 +643,9 @@
                 woodlandmansionpieces$placementdata2.wallType = "wall_window";
                 boolean flag = false;
 
-                for (int k = 0; k < woodlandmansionpieces$simplegrid1.height && !flag; k++)
+                for (int k = 0; k < woodlandmansionpieces$simplegrid1.height && !flag; ++k)
                 {
-                    for (int l = woodlandmansionpieces$simplegrid1.width - 1; l >= 0 && !flag; l--)
+                    for (int l = woodlandmansionpieces$simplegrid1.width - 1; l >= 0 && !flag; --l)
                     {
                         if (WoodlandMansionPieces.Grid.isHouse(woodlandmansionpieces$simplegrid1, l, k))
                         {
@@ -672,7 +672,7 @@
                     new WoodlandMansionPieces.FirstFloor(), new WoodlandMansionPieces.SecondFloor(), new WoodlandMansionPieces.ThirdFloor()
                 };
 
-                for (int l2 = 0; l2 < 3; l2++)
+                for (int l2 = 0; l2 < 3; ++l2)
                 {
                     BlockPos blockpos = p_191125_1_.up(8 * l2 + (l2 == 2 ? 3 : 0));
                     WoodlandMansionPieces.SimpleGrid woodlandmansionpieces$simplegrid2 = p_191125_4_.floorRooms[l2];
@@ -680,9 +680,9 @@
                     String s = l2 == 0 ? "carpet_south" : "carpet_south_2";
                     String s1 = l2 == 0 ? "carpet_west" : "carpet_west_2";
 
-                    for (int i1 = 0; i1 < woodlandmansionpieces$simplegrid3.height; i1++)
+                    for (int i1 = 0; i1 < woodlandmansionpieces$simplegrid3.height; ++i1)
                     {
-                        for (int j1 = 0; j1 < woodlandmansionpieces$simplegrid3.width; j1++)
+                        for (int j1 = 0; j1 < woodlandmansionpieces$simplegrid3.width; ++j1)
                         {
                             if (woodlandmansionpieces$simplegrid3.get(j1, i1) == 1)
                             {
@@ -753,9 +753,9 @@
                     String s3 = l2 == 0 ? "indoors_door" : "indoors_door_2";
                     List<EnumFacing> list = Lists.newArrayList();
 
-                    for (int k1 = 0; k1 < woodlandmansionpieces$simplegrid3.height; k1++)
+                    for (int k1 = 0; k1 < woodlandmansionpieces$simplegrid3.height; ++k1)
                     {
-                        for (int l1 = 0; l1 < woodlandmansionpieces$simplegrid3.width; l1++)
+                        for (int l1 = 0; l1 < woodlandmansionpieces$simplegrid3.width; ++l1)
                         {
                             boolean flag1 = l2 == 2 && woodlandmansionpieces$simplegrid3.get(l1, k1) == 3;
 
@@ -925,9 +925,9 @@
                 @Nullable WoodlandMansionPieces.SimpleGrid p_191123_5_
             )
             {
-                for (int i = 0; i < p_191123_4_.height; i++)
+                for (int i = 0; i < p_191123_4_.height; ++i)
                 {
-                    for (int j = 0; j < p_191123_4_.width; j++)
+                    for (int j = 0; j < p_191123_4_.width; ++j)
                     {
                         BlockPos lvt_8_3_ = p_191123_2_.offset(p_191123_3_.rotate(EnumFacing.SOUTH), 8 + (i - this.startY) * 8);
                         lvt_8_3_ = lvt_8_3_.offset(p_191123_3_.rotate(EnumFacing.EAST), (j - this.startX) * 8);
@@ -980,9 +980,9 @@
 
                 if (p_191123_5_ != null)
                 {
-                    for (int k = 0; k < p_191123_4_.height; k++)
+                    for (int k = 0; k < p_191123_4_.height; ++k)
                     {
-                        for (int i1 = 0; i1 < p_191123_4_.width; i1++)
+                        for (int i1 = 0; i1 < p_191123_4_.width; ++i1)
                         {
                             BlockPos blockpos3 = p_191123_2_.offset(p_191123_3_.rotate(EnumFacing.SOUTH), 8 + (k - this.startY) * 8);
                             blockpos3 = blockpos3.offset(p_191123_3_.rotate(EnumFacing.EAST), (i1 - this.startX) * 8);
@@ -1079,9 +1079,9 @@
                     }
                 }
 
-                for (int l = 0; l < p_191123_4_.height; l++)
+                for (int l = 0; l < p_191123_4_.height; ++l)
                 {
-                    for (int j1 = 0; j1 < p_191123_4_.width; j1++)
+                    for (int j1 = 0; j1 < p_191123_4_.width; ++j1)
                     {
                         BlockPos blockpos4 = p_191123_2_.offset(p_191123_3_.rotate(EnumFacing.SOUTH), 8 + (l - this.startY) * 8);
                         blockpos4 = blockpos4.offset(p_191123_3_.rotate(EnumFacing.EAST), (j1 - this.startX) * 8);
@@ -1589,9 +1589,9 @@
 
             public void set(int p_191142_1_, int p_191142_2_, int p_191142_3_, int p_191142_4_, int p_191142_5_)
             {
-                for (int i = p_191142_2_; i <= p_191142_4_; i++)
+                for (int i = p_191142_2_; i <= p_191142_4_; ++i)
                 {
-                    for (int j = p_191142_1_; j <= p_191142_3_; j++)
+                    for (int j = p_191142_1_; j <= p_191142_3_; ++j)
                     {
                         this.set(j, i, p_191142_5_);
                     }
