--- before/net/minecraft/block/state/pattern/BlockStateMatcher.java
+++ after/net/minecraft/block/state/pattern/BlockStateMatcher.java
@@ -60,7 +60,7 @@
     }
     protected <T extends Comparable<T>> boolean matches(IBlockState blockState, IProperty<T> property, Predicate<T> predicate)
     {
-        return ((Predicate<T>)predicate).apply(blockState.getValue(property));
+        return predicate.apply(blockState.getValue(property));
     }
 
     public <V extends Comparable<V>> BlockStateMatcher where(IProperty<V> property, Predicate <? extends V > is)
