--- before/net/minecraft/block/state/pattern/BlockPattern.java
+++ after/net/minecraft/block/state/pattern/BlockPattern.java
@@ -63,11 +63,11 @@
         BlockPos p_177682_1_, EnumFacing p_177682_2_, EnumFacing p_177682_3_, LoadingCache<BlockPos, BlockWorldState> p_177682_4_
     )
     {
-        for (int i = 0; i < this.field_177686_d; i++)
+        for (int i = 0; i < this.field_177686_d; ++i)
         {
-            for (int j = 0; j < this.field_177688_c; j++)
+            for (int j = 0; j < this.field_177688_c; ++j)
             {
-                for (int k = 0; k < this.field_177687_b; k++)
+                for (int k = 0; k < this.field_177687_b; ++k)
                 {
                     if (!this.field_177689_a[k][j][i].apply(p_177682_4_.getUnchecked(func_177683_a(p_177682_1_, p_177682_2_, p_177682_3_, i, j, k))))
                     {
