--- before/net/minecraft/block/state/pattern/BlockStateMatcher.java
+++ after/net/minecraft/block/state/pattern/BlockStateMatcher.java
@@ -60,7 +60,7 @@
     }
     protected <T extends Comparable<T>> boolean func_185927_a(IBlockState p_185927_1_, IProperty<T> p_185927_2_, Predicate<T> p_185927_3_)
     {
-        return ((Predicate<T>)p_185927_3_).apply(p_185927_1_.func_177229_b(p_185927_2_));
+        return p_185927_3_.apply(p_185927_1_.func_177229_b(p_185927_2_));
     }
 
     public <V extends Comparable<V>> BlockStateMatcher func_177637_a(IProperty<V> p_177637_1_, Predicate <? extends V > p_177637_2_)
