--- before/net/minecraft/block/state/pattern/BlockPattern.java
+++ after/net/minecraft/block/state/pattern/BlockPattern.java
@@ -63,11 +63,11 @@
         BlockPos pos, EnumFacing finger, EnumFacing thumb, LoadingCache<BlockPos, BlockWorldState> lcache
     )
     {
-        for (int i = 0; i < this.palmLength; i++)
+        for (int i = 0; i < this.palmLength; ++i)
         {
-            for (int j = 0; j < this.thumbLength; j++)
+            for (int j = 0; j < this.thumbLength; ++j)
             {
-                for (int k = 0; k < this.fingerLength; k++)
+                for (int k = 0; k < this.fingerLength; ++k)
                 {
                     if (!this.blockMatches[k][j][i].apply(lcache.getUnchecked(translateOffset(pos, finger, thumb, i, j, k))))
                     {
