--- before/net/minecraft/block/state/pattern/FactoryBlockPattern.java
+++ after/net/minecraft/block/state/pattern/FactoryBlockPattern.java
@@ -91,15 +91,15 @@
     private Predicate<BlockWorldState>[][][] func_177658_c()
     {
         this.func_177657_d();
-        Predicate<BlockWorldState>[][][] predicate = (Predicate<BlockWorldState>[][][])Array.newInstance(
+        Predicate<BlockWorldState>[][][] predicate = (Predicate[][][])Array.newInstance(
                     Predicate.class, this.field_177665_b.size(), this.field_177663_d, this.field_177664_e
                 );
 
-        for (int i = 0; i < this.field_177665_b.size(); i++)
+        for (int i = 0; i < this.field_177665_b.size(); ++i)
         {
-            for (int j = 0; j < this.field_177663_d; j++)
+            for (int j = 0; j < this.field_177663_d; ++j)
             {
-                for (int k = 0; k < this.field_177664_e; k++)
+                for (int k = 0; k < this.field_177664_e; ++k)
                 {
                     predicate[i][j][k] = this.field_177666_c.get(this.field_177665_b.get(i)[j].charAt(k));
                 }
