--- before/net/minecraft/util/registry/RegistryNamespaced.java
+++ after/net/minecraft/util/registry/RegistryNamespaced.java
@@ -10,8 +10,8 @@
 
 public class RegistryNamespaced<K, V> extends RegistrySimple<K, V> implements IObjectIntIterable<V>
 {
-    protected final IntIdentityHashBiMap<V> field_148759_a = new IntIdentityHashBiMap(256);
-    protected final Map<V, K> field_148758_b = ((BiMap)this.field_82596_a).inverse();
+    protected final IntIdentityHashBiMap<V> field_148759_a = new IntIdentityHashBiMap<>(256);
+    protected final Map<V, K> field_148758_b = ((BiMap) this.field_82596_a).inverse();
 
     public void func_177775_a(int p_177775_1_, K p_177775_2_, V p_177775_3_)
     {
