--- before/net/minecraft/util/registry/RegistryNamespacedDefaultedByKey.java
+++ after/net/minecraft/util/registry/RegistryNamespacedDefaultedByKey.java
@@ -43,7 +43,7 @@
     public K getNameForObject(V value)
     {
         K k = super.getNameForObject(value);
-        return k == null ? this.defaultValueKey : k;
+        return (K)(k == null ? this.defaultValueKey : k);
     }
 
     @Nonnull
@@ -51,7 +51,7 @@
     public V getObject(@Nullable K name)
     {
         V v = super.getObject(name);
-        return v == null ? this.defaultValue : v;
+        return (V)(v == null ? this.defaultValue : v);
     }
 
     @Nonnull
@@ -59,7 +59,7 @@
     public V getObjectById(int id)
     {
         V v = super.getObjectById(id);
-        return v == null ? this.defaultValue : v;
+        return (V)(v == null ? this.defaultValue : v);
     }
 
     @Nonnull
@@ -67,6 +67,6 @@
     public V getRandomObject(Random random)
     {
         V v = super.getRandomObject(random);
-        return v == null ? this.defaultValue : v;
+        return (V)(v == null ? this.defaultValue : v);
     }
 }
