--- before/net/minecraft/util/registry/RegistryNamespaced.java
+++ after/net/minecraft/util/registry/RegistryNamespaced.java
@@ -10,8 +10,8 @@
 
 public class RegistryNamespaced<K, V> extends RegistrySimple<K, V> implements IObjectIntIterable<V>
 {
-    protected final IntIdentityHashBiMap<V> underlyingIntegerMap = new IntIdentityHashBiMap(256);
-    protected final Map<V, K> inverseObjectRegistry = ((BiMap)this.registryObjects).inverse();
+    protected final IntIdentityHashBiMap<V> underlyingIntegerMap = new IntIdentityHashBiMap<>(256);
+    protected final Map<V, K> inverseObjectRegistry = ((BiMap) this.registryObjects).inverse();
 
     public void register(int id, K key, V value)
     {
