--- before/net/minecraft/pathfinding/PathFinder.java
+++ after/net/minecraft/pathfinding/PathFinder.java
@@ -88,7 +88,7 @@
             pathpoint1.visited = true;
             int j = this.nodeProcessor.findPathOptions(this.pathOptions, pathpoint1, pathTo, maxDistance);
 
-            for (int k = 0; k < j; k++)
+            for (int k = 0; k < j; ++k)
             {
                 PathPoint pathpoint2 = this.pathOptions[k];
                 float f = pathpoint1.distanceManhattan(pathpoint2);
@@ -124,17 +124,17 @@
 
         for (PathPoint pathpoint = end; pathpoint.previous != null; pathpoint = pathpoint.previous)
         {
-            i++;
+            ++i;
         }
 
         PathPoint[] apathpoint = new PathPoint[i];
         PathPoint pathpoint1 = end;
-        i--;
+        --i;
 
         for (apathpoint[i] = end; pathpoint1.previous != null; apathpoint[i] = pathpoint1)
         {
             pathpoint1 = pathpoint1.previous;
-            i--;
+            --i;
         }
 
         return new Path(apathpoint);
