--- before/net/minecraft/pathfinding/PathHeap.java
+++ after/net/minecraft/pathfinding/PathHeap.java
@@ -65,11 +65,11 @@
     private void func_75847_a(int p_75847_1_)
     {
         PathPoint pathpoint = this.field_75852_a[p_75847_1_];
-        float f = pathpoint.field_75834_g;
+        int i;
 
-        while (p_75847_1_ > 0)
+        for (float f = pathpoint.field_75834_g; p_75847_1_ > 0; p_75847_1_ = i)
         {
-            int i = p_75847_1_ - 1 >> 1;
+            i = p_75847_1_ - 1 >> 1;
             PathPoint pathpoint1 = this.field_75852_a[i];
 
             if (!(f < pathpoint1.field_75834_g))
@@ -79,7 +79,6 @@
 
             this.field_75852_a[p_75847_1_] = pathpoint1;
             pathpoint1.field_75835_d = p_75847_1_;
-            p_75847_1_ = i;
         }
 
         this.field_75852_a[p_75847_1_] = pathpoint;
