--- before/net/minecraft/pathfinding/PathHeap.java
+++ after/net/minecraft/pathfinding/PathHeap.java
@@ -65,11 +65,11 @@
     private void sortBack(int index)
     {
         PathPoint pathpoint = this.pathPoints[index];
-        float f = pathpoint.distanceToTarget;
+        int i;
 
-        while (index > 0)
+        for (float f = pathpoint.distanceToTarget; index > 0; index = i)
         {
-            int i = index - 1 >> 1;
+            i = index - 1 >> 1;
             PathPoint pathpoint1 = this.pathPoints[i];
 
             if (!(f < pathpoint1.distanceToTarget))
@@ -79,7 +79,6 @@
 
             this.pathPoints[index] = pathpoint1;
             pathpoint1.index = index;
-            index = i;
         }
 
         this.pathPoints[index] = pathpoint;
