--- before/net/minecraft/pathfinding/Path.java
+++ after/net/minecraft/pathfinding/Path.java
@@ -25,7 +25,7 @@
 
     public void incrementPathIndex()
     {
-        this.currentPathIndex++;
+        ++this.currentPathIndex;
     }
 
     public boolean isFinished()
@@ -100,7 +100,7 @@
         }
         else
         {
-            for (int i = 0; i < this.points.length; i++)
+            for (int i = 0; i < this.points.length; ++i)
             {
                 if (this.points[i].x != pathentityIn.points[i].x
                         || this.points[i].y != pathentityIn.points[i].y
@@ -139,21 +139,21 @@
         PathPoint pathpoint = PathPoint.createFromBuffer(buf);
         PathPoint[] apathpoint = new PathPoint[buf.readInt()];
 
-        for (int j = 0; j < apathpoint.length; j++)
+        for (int j = 0; j < apathpoint.length; ++j)
         {
             apathpoint[j] = PathPoint.createFromBuffer(buf);
         }
 
         PathPoint[] apathpoint1 = new PathPoint[buf.readInt()];
 
-        for (int k = 0; k < apathpoint1.length; k++)
+        for (int k = 0; k < apathpoint1.length; ++k)
         {
             apathpoint1[k] = PathPoint.createFromBuffer(buf);
         }
 
         PathPoint[] apathpoint2 = new PathPoint[buf.readInt()];
 
-        for (int l = 0; l < apathpoint2.length; l++)
+        for (int l = 0; l < apathpoint2.length; ++l)
         {
             apathpoint2[l] = PathPoint.createFromBuffer(buf);
         }
