--- before/net/minecraft/pathfinding/PathNavigateSwimmer.java
+++ after/net/minecraft/pathfinding/PathNavigateSwimmer.java
@@ -45,7 +45,7 @@
             this.currentPath.incrementPathIndex();
         }
 
-        for (int j = Math.min(this.currentPath.getCurrentPathIndex() + 6, this.currentPath.getCurrentPathLength() - 1); j > this.currentPath.getCurrentPathIndex(); j--)
+        for (int j = Math.min(this.currentPath.getCurrentPathIndex() + 6, this.currentPath.getCurrentPathLength() - 1); j > this.currentPath.getCurrentPathIndex(); --j)
         {
             Vec3d vec3d1 = this.currentPath.getVectorFromIndex(this.entity, j);
 
