--- before/net/minecraft/pathfinding/PathNavigateGround.java
+++ after/net/minecraft/pathfinding/PathNavigateGround.java
@@ -143,7 +143,7 @@
                 return;
             }
 
-            for (int i = 0; i < this.field_75514_c.func_75874_d(); i++)
+            for (int i = 0; i < this.field_75514_c.func_75874_d(); ++i)
             {
                 PathPoint pathpoint = this.field_75514_c.func_75877_a(i);
 
@@ -192,12 +192,12 @@
 
                 if (d0 >= 0.0)
                 {
-                    d6++;
+                    ++d6;
                 }
 
                 if (d1 >= 0.0)
                 {
-                    d7++;
+                    ++d7;
                 }
 
                 d6 /= d0;
@@ -256,9 +256,9 @@
         }
         else
         {
-            for (int k = i; k < i + p_179683_4_; k++)
+            for (int k = i; k < i + p_179683_4_; ++k)
             {
-                for (int l = j; l < j + p_179683_6_; l++)
+                for (int l = j; l < j + p_179683_6_; ++l)
                 {
                     double d0 = (double)k + 0.5 - p_179683_7_.field_72450_a;
                     double d1 = (double)l + 0.5 - p_179683_7_.field_72449_c;
