--- before/net/minecraft/entity/boss/dragon/phase/PhaseHoldingPattern.java
+++ after/net/minecraft/entity/boss/dragon/phase/PhaseHoldingPattern.java
@@ -100,11 +100,11 @@
 
             if (this.field_188679_d)
             {
-                k++;
+                ++k;
             }
             else
             {
-                k--;
+                --k;
             }
 
             if (this.field_188661_a.func_184664_cU() != null && this.field_188661_a.func_184664_cU().func_186092_c() >= 0)
