--- before/net/minecraft/entity/boss/dragon/phase/PhaseStrafePlayer.java
+++ after/net/minecraft/entity/boss/dragon/phase/PhaseStrafePlayer.java
@@ -62,7 +62,7 @@
             {
                 if (this.dragon.canEntityBeSeen(this.attackTarget))
                 {
-                    this.fireballCharge++;
+                    ++this.fireballCharge;
                     Vec3d vec3d1 = new Vec3d(
                         this.attackTarget.posX - this.dragon.posX,
                         0.0,
@@ -110,12 +110,12 @@
                 }
                 else if (this.fireballCharge > 0)
                 {
-                    this.fireballCharge--;
+                    --this.fireballCharge;
                 }
             }
             else if (this.fireballCharge > 0)
             {
-                this.fireballCharge--;
+                --this.fireballCharge;
             }
         }
     }
@@ -135,11 +135,11 @@
 
             if (this.holdingPatternClockwise)
             {
-                j++;
+                ++j;
             }
             else
             {
-                j--;
+                --j;
             }
 
             if (this.dragon.getFightManager() != null && this.dragon.getFightManager().getNumAliveCrystals() > 0)
