--- before/net/minecraft/entity/boss/dragon/phase/PhaseSittingScanning.java
+++ after/net/minecraft/entity/boss/dragon/phase/PhaseSittingScanning.java
@@ -17,7 +17,7 @@
     @Override
     public void doLocalUpdate()
     {
-        this.scanningTime++;
+        ++this.scanningTime;
         EntityLivingBase entitylivingbase = this.dragon.world.getNearestAttackablePlayer(this.dragon, 20.0, 10.0);
 
         if (entitylivingbase != null)
@@ -60,7 +60,7 @@
                     }
 
                     this.dragon.randomYawVelocity = (float)((double)this.dragon.randomYawVelocity + d2 * (double)(0.7F / f2 / f3));
-                    this.dragon.rotationYaw = this.dragon.rotationYaw + this.dragon.randomYawVelocity;
+                    this.dragon.rotationYaw += this.dragon.randomYawVelocity;
                 }
             }
         }
