--- before/net/minecraft/entity/boss/dragon/phase/PhaseChargingPlayer.java
+++ after/net/minecraft/entity/boss/dragon/phase/PhaseChargingPlayer.java
@@ -35,7 +35,7 @@
 
             if (d0 < 100.0 || d0 > 22500.0 || this.dragon.collidedHorizontally || this.dragon.collidedVertically)
             {
-                this.timeSinceCharge++;
+                ++this.timeSinceCharge;
             }
         }
     }
