--- before/net/minecraft/enchantment/EnchantmentProtection.java
+++ after/net/minecraft/enchantment/EnchantmentProtection.java
@@ -80,9 +80,15 @@
         if (ench instanceof EnchantmentProtection)
         {
             EnchantmentProtection enchantmentprotection = (EnchantmentProtection)ench;
-            return this.protectionType == enchantmentprotection.protectionType
-                   ? false
-                   : this.protectionType == EnchantmentProtection.Type.FALL || enchantmentprotection.protectionType == EnchantmentProtection.Type.FALL;
+
+            if (this.protectionType == enchantmentprotection.protectionType)
+            {
+                return false;
+            }
+            else
+            {
+                return this.protectionType == EnchantmentProtection.Type.FALL || enchantmentprotection.protectionType == EnchantmentProtection.Type.FALL;
+            }
         }
         else
         {
