--- before/net/minecraft/enchantment/EnchantmentDurability.java
+++ after/net/minecraft/enchantment/EnchantmentDurability.java
@@ -39,6 +39,13 @@
 
     public static boolean negateDamage(ItemStack stack, int level, Random rand)
     {
-        return stack.getItem() instanceof ItemArmor && rand.nextFloat() < 0.6F ? false : rand.nextInt(level + 1) > 0;
+        if (stack.getItem() instanceof ItemArmor && rand.nextFloat() < 0.6F)
+        {
+            return false;
+        }
+        else
+        {
+            return rand.nextInt(level + 1) > 0;
+        }
     }
 }
