--- before/net/minecraft/client/tutorial/MovementStep.java
+++ after/net/minecraft/client/tutorial/MovementStep.java
@@ -38,17 +38,17 @@
     @Override
     public void update()
     {
-        this.timeWaiting++;
+        ++this.timeWaiting;
 
         if (this.moved)
         {
-            this.timeMoved++;
+            ++this.timeMoved;
             this.moved = false;
         }
 
         if (this.turned)
         {
-            this.timeLooked++;
+            ++this.timeLooked;
             this.turned = false;
         }
 
