--- before/net/minecraft/client/tutorial/PunchTreeStep.java
+++ after/net/minecraft/client/tutorial/PunchTreeStep.java
@@ -36,7 +36,7 @@
     @Override
     public void update()
     {
-        this.timeWaiting++;
+        ++this.timeWaiting;
 
         if (this.tutorial.getGameType() != GameType.SURVIVAL)
         {
@@ -108,7 +108,7 @@
         }
         else if (flag)
         {
-            this.resetCount++;
+            ++this.resetCount;
         }
     }
 
