--- before/net/minecraft/block/BlockRail.java
+++ after/net/minecraft/block/BlockRail.java
@@ -46,6 +46,7 @@
         return state.getValue(SHAPE).getMetadata();
     }
 
+    @SuppressWarnings("incomplete-switch")
     @Override
     public IBlockState withRotation(IBlockState state, Rotation rot)
     {
@@ -130,6 +131,7 @@
         }
     }
 
+    @SuppressWarnings("incomplete-switch")
     @Override
     public IBlockState withMirror(IBlockState state, Mirror mirrorIn)
     {
