--- before/net/minecraft/block/BlockDaylightDetector.java
+++ after/net/minecraft/block/BlockDaylightDetector.java
@@ -77,6 +77,7 @@
 
             if (iblockstate.getValue(POWER) != i)
             {
+                i = org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callRedstoneChange(worldIn, pos.getX(), pos.getY(), pos.getZ(), iblockstate.getValue(POWER), i).getNewCurrent();
                 worldIn.setBlockState(pos, iblockstate.withProperty(POWER, i), 3);
             }
         }
