--- before/net/minecraft/block/BlockPressurePlate.java
+++ after/net/minecraft/block/BlockPressurePlate.java
@@ -7,11 +7,13 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.SoundEvents;
 import net.minecraft.util.SoundCategory;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
+import org.bukkit.event.entity.EntityInteractEvent;
 
 public class BlockPressurePlate extends BlockBasePressurePlate
 {
@@ -85,6 +87,24 @@
         {
             for (Entity entity : list)
             {
+                if (this.getRedstoneStrength(worldIn.getBlockState(pos)) == 0) {
+                    org.bukkit.World bworld = worldIn.getWorld();
+                    org.bukkit.plugin.PluginManager manager = worldIn.getServer().getPluginManager();
+                    org.bukkit.event.Cancellable cancellable;
+
+                    if (entity instanceof EntityPlayer) {
+                        cancellable = org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callPlayerInteractEvent((EntityPlayer) entity, org.bukkit.event.block.Action.PHYSICAL, pos, null, null, null);
+                    } else {
+                        cancellable = new EntityInteractEvent(entity.getBukkitEntity(), bworld.getBlockAt(pos.getX(), pos.getY(), pos.getZ()));
+                        manager.callEvent((EntityInteractEvent) cancellable);
+                    }
+
+                    // We only want to block turning the plate on if all events are cancelled
+                    if (cancellable.isCancelled()) {
+                        continue;
+                    }
+                }
+
                 if (!entity.doesEntityNotTriggerPressurePlate())
                 {
                     return 15;
