--- before/net/minecraft/block/BlockEndGateway.java
+++ after/net/minecraft/block/BlockEndGateway.java
@@ -77,7 +77,7 @@
         {
             int i = ((TileEntityEndGateway)tileentity).getParticleAmount();
 
-            for (int j = 0; j < i; j++)
+            for (int j = 0; j < i; ++j)
             {
                 double d0 = (double)((float)pos.getX() + rand.nextFloat());
                 double d1 = (double)((float)pos.getY() + rand.nextFloat());
