--- before/net/minecraft/block/BlockShulkerBox.java
+++ after/net/minecraft/block/BlockShulkerBox.java
@@ -43,7 +43,7 @@
 public class BlockShulkerBox extends BlockContainer
 {
     public static final PropertyEnum<EnumFacing> field_190957_a = PropertyDirection.func_177714_a("facing");
-    private final EnumDyeColor field_190958_b;
+    public final EnumDyeColor field_190958_b;
 
     public BlockShulkerBox(EnumDyeColor p_i47248_1_)
     {
@@ -273,11 +273,11 @@
                 {
                     if (!itemstack.func_190926_b())
                     {
-                        j++;
+                        ++j;
 
                         if (i <= 4)
                         {
-                            i++;
+                            ++i;
                             p_190948_3_.add(String.format("%s x%d", itemstack.func_82833_r(), itemstack.func_190916_E()));
                         }
                     }
@@ -345,42 +345,24 @@
 
     public static Block func_190952_a(EnumDyeColor p_190952_0_)
     {
-        switch (p_190952_0_)
-        {
-            case WHITE:
-                return Blocks.field_190977_dl;
-            case ORANGE:
-                return Blocks.field_190978_dm;
-            case MAGENTA:
-                return Blocks.field_190979_dn;
-            case LIGHT_BLUE:
-                return Blocks.field_190980_do;
-            case YELLOW:
-                return Blocks.field_190981_dp;
-            case LIME:
-                return Blocks.field_190982_dq;
-            case PINK:
-                return Blocks.field_190983_dr;
-            case GRAY:
-                return Blocks.field_190984_ds;
-            case SILVER:
-                return Blocks.field_190985_dt;
-            case CYAN:
-                return Blocks.field_190986_du;
-            case PURPLE:
-            default:
-                return Blocks.field_190987_dv;
-            case BLUE:
-                return Blocks.field_190988_dw;
-            case BROWN:
-                return Blocks.field_190989_dx;
-            case GREEN:
-                return Blocks.field_190990_dy;
-            case RED:
-                return Blocks.field_190991_dz;
-            case BLACK:
-                return Blocks.field_190975_dA;
-        }
+        return switch (p_190952_0_) {
+            case WHITE -> Blocks.field_190977_dl;
+            case ORANGE -> Blocks.field_190978_dm;
+            case MAGENTA -> Blocks.field_190979_dn;
+            case LIGHT_BLUE -> Blocks.field_190980_do;
+            case YELLOW -> Blocks.field_190981_dp;
+            case LIME -> Blocks.field_190982_dq;
+            case PINK -> Blocks.field_190983_dr;
+            case GRAY -> Blocks.field_190984_ds;
+            case SILVER -> Blocks.field_190985_dt;
+            case CYAN -> Blocks.field_190986_du;
+            case BLUE -> Blocks.field_190988_dw;
+            case BROWN -> Blocks.field_190989_dx;
+            case GREEN -> Blocks.field_190990_dy;
+            case RED -> Blocks.field_190991_dz;
+            case BLACK -> Blocks.field_190975_dA;
+            default -> Blocks.field_190987_dv;
+        };
     }
 
     @SideOnly(Side.CLIENT)
@@ -411,8 +393,12 @@
     {
         p_193383_2_ = this.func_176221_a(p_193383_2_, p_193383_1_, p_193383_3_);
         EnumFacing enumfacing = p_193383_2_.func_177229_b(field_190957_a);
-        TileEntityShulkerBox.AnimationStatus tileentityshulkerbox$animationstatus = ((TileEntityShulkerBox)p_193383_1_.func_175625_s(p_193383_3_))
-                .func_190591_p();
+        TileEntityShulkerBox te = (TileEntityShulkerBox) p_193383_1_.func_175625_s(p_193383_3_);
+
+        // CREF: getTileEntity may return null
+        TileEntityShulkerBox.AnimationStatus tileentityshulkerbox$animationstatus =
+            te == null ? TileEntityShulkerBox.AnimationStatus.CLOSED : te.func_190591_p();
+
         return tileentityshulkerbox$animationstatus != TileEntityShulkerBox.AnimationStatus.CLOSED
                && (
                    tileentityshulkerbox$animationstatus != TileEntityShulkerBox.AnimationStatus.OPENED
