--- before/net/minecraft/block/BlockSponge.java
+++ after/net/minecraft/block/BlockSponge.java
@@ -87,7 +87,7 @@
                 {
                     worldIn.setBlockState(blockpos1, Blocks.AIR.getDefaultState(), 2);
                     list.add(blockpos1);
-                    i++;
+                    ++i;
 
                     if (j < 6)
                     {
@@ -165,7 +165,7 @@
 
                         if (enumfacing == EnumFacing.EAST)
                         {
-                            d0++;
+                            ++d0;
                         }
                         else
                         {
@@ -178,7 +178,7 @@
 
                         if (enumfacing == EnumFacing.SOUTH)
                         {
-                            d2++;
+                            ++d2;
                         }
                         else
                         {
