--- before/net/minecraft/block/BlockLilyPad.java
+++ after/net/minecraft/block/BlockLilyPad.java
@@ -44,7 +44,7 @@
     {
         super.onEntityCollision(worldIn, pos, state, entityIn);
 
-        if (entityIn instanceof EntityBoat)
+        if (entityIn instanceof EntityBoat && !org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callEntityChangeBlockEvent(entityIn, pos, Blocks.AIR, 0).isCancelled())
         {
             worldIn.destroyBlock(new BlockPos(pos), true);
         }
