--- before/net/minecraft/block/BlockStainedGlassPane.java
+++ after/net/minecraft/block/BlockStainedGlassPane.java
@@ -46,7 +46,7 @@
     @Override
     public void getSubBlocks(CreativeTabs itemIn, NonNullList<ItemStack> items)
     {
-        for (int i = 0; i < EnumDyeColor.values().length; i++)
+        for (int i = 0; i < EnumDyeColor.values().length; ++i)
         {
             items.add(new ItemStack(this, 1, i));
         }
