--- before/net/minecraft/block/BlockTripWireHook.java
+++ after/net/minecraft/block/BlockTripWireHook.java
@@ -11,6 +11,7 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
 import net.minecraft.init.SoundEvents;
 import net.minecraft.item.ItemStack;
@@ -25,6 +26,7 @@
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import org.bukkit.event.block.BlockRedstoneEvent;
 
 public class BlockTripWireHook extends Block
 {
@@ -177,7 +179,7 @@
         int i = 0;
         IBlockState[] aiblockstate = new IBlockState[42];
 
-        for (int j = 1; j < 42; j++)
+        for (int j = 1; j < 42; ++j)
         {
             BlockPos blockpos = pos.offset(enumfacing, j);
             IBlockState iblockstate = worldIn.getBlockState(blockpos);
@@ -230,6 +232,15 @@
             this.playSound(worldIn, blockpos1, flag2, flag3, flag, flag1);
         }
 
+        org.bukkit.block.Block block = worldIn.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+
+        BlockRedstoneEvent eventRedstone = new BlockRedstoneEvent(block, 15, 0);
+        worldIn.getServer().getPluginManager().callEvent(eventRedstone);
+
+        if (eventRedstone.getNewCurrent() > 0) {
+            return;
+        }
+
         this.playSound(worldIn, pos, flag2, flag3, flag, flag1);
 
         if (!p_176260_4_)
@@ -244,7 +255,7 @@
 
         if (flag != flag2)
         {
-            for (int k = 1; k < i; k++)
+            for (int k = 1; k < i; ++k)
             {
                 BlockPos blockpos2 = pos.offset(enumfacing, k);
                 IBlockState iblockstate2 = aiblockstate[k];
