--- before/net/minecraft/block/BlockShulkerBox.java
+++ after/net/minecraft/block/BlockShulkerBox.java
@@ -43,7 +43,7 @@
 public class BlockShulkerBox extends BlockContainer
 {
     public static final PropertyEnum<EnumFacing> FACING = PropertyDirection.create("facing");
-    private final EnumDyeColor color;
+    public final EnumDyeColor color;
 
     public BlockShulkerBox(EnumDyeColor colorIn)
     {
@@ -273,11 +273,11 @@
                 {
                     if (!itemstack.isEmpty())
                     {
-                        j++;
+                        ++j;
 
                         if (i <= 4)
                         {
-                            i++;
+                            ++i;
                             tooltip.add(String.format("%s x%d", itemstack.getDisplayName(), itemstack.getCount()));
                         }
                     }
@@ -345,42 +345,24 @@
 
     public static Block getBlockByColor(EnumDyeColor colorIn)
     {
-        switch (colorIn)
-        {
-            case WHITE:
-                return Blocks.WHITE_SHULKER_BOX;
-            case ORANGE:
-                return Blocks.ORANGE_SHULKER_BOX;
-            case MAGENTA:
-                return Blocks.MAGENTA_SHULKER_BOX;
-            case LIGHT_BLUE:
-                return Blocks.LIGHT_BLUE_SHULKER_BOX;
-            case YELLOW:
-                return Blocks.YELLOW_SHULKER_BOX;
-            case LIME:
-                return Blocks.LIME_SHULKER_BOX;
-            case PINK:
-                return Blocks.PINK_SHULKER_BOX;
-            case GRAY:
-                return Blocks.GRAY_SHULKER_BOX;
-            case SILVER:
-                return Blocks.SILVER_SHULKER_BOX;
-            case CYAN:
-                return Blocks.CYAN_SHULKER_BOX;
-            case PURPLE:
-            default:
-                return Blocks.PURPLE_SHULKER_BOX;
-            case BLUE:
-                return Blocks.BLUE_SHULKER_BOX;
-            case BROWN:
-                return Blocks.BROWN_SHULKER_BOX;
-            case GREEN:
-                return Blocks.GREEN_SHULKER_BOX;
-            case RED:
-                return Blocks.RED_SHULKER_BOX;
-            case BLACK:
-                return Blocks.BLACK_SHULKER_BOX;
-        }
+        return switch (colorIn) {
+            case WHITE -> Blocks.WHITE_SHULKER_BOX;
+            case ORANGE -> Blocks.ORANGE_SHULKER_BOX;
+            case MAGENTA -> Blocks.MAGENTA_SHULKER_BOX;
+            case LIGHT_BLUE -> Blocks.LIGHT_BLUE_SHULKER_BOX;
+            case YELLOW -> Blocks.YELLOW_SHULKER_BOX;
+            case LIME -> Blocks.LIME_SHULKER_BOX;
+            case PINK -> Blocks.PINK_SHULKER_BOX;
+            case GRAY -> Blocks.GRAY_SHULKER_BOX;
+            case SILVER -> Blocks.SILVER_SHULKER_BOX;
+            case CYAN -> Blocks.CYAN_SHULKER_BOX;
+            case BLUE -> Blocks.BLUE_SHULKER_BOX;
+            case BROWN -> Blocks.BROWN_SHULKER_BOX;
+            case GREEN -> Blocks.GREEN_SHULKER_BOX;
+            case RED -> Blocks.RED_SHULKER_BOX;
+            case BLACK -> Blocks.BLACK_SHULKER_BOX;
+            default -> Blocks.PURPLE_SHULKER_BOX;
+        };
     }
 
     @SideOnly(Side.CLIENT)
@@ -411,8 +393,12 @@
     {
         state = this.getActualState(state, worldIn, pos);
         EnumFacing enumfacing = state.getValue(FACING);
-        TileEntityShulkerBox.AnimationStatus tileentityshulkerbox$animationstatus = ((TileEntityShulkerBox)worldIn.getTileEntity(pos))
-                .getAnimationStatus();
+        TileEntityShulkerBox te = (TileEntityShulkerBox) worldIn.getTileEntity(pos);
+
+        // CREF: getTileEntity may return null
+        TileEntityShulkerBox.AnimationStatus tileentityshulkerbox$animationstatus =
+            te == null ? TileEntityShulkerBox.AnimationStatus.CLOSED : te.getAnimationStatus();
+
         return tileentityshulkerbox$animationstatus != TileEntityShulkerBox.AnimationStatus.CLOSED
                && (
                    tileentityshulkerbox$animationstatus != TileEntityShulkerBox.AnimationStatus.OPENED
