--- before/net/minecraft/block/BlockDispenser.java
+++ after/net/minecraft/block/BlockDispenser.java
@@ -132,7 +132,7 @@
         }
     }
 
-    protected void dispense(World worldIn, BlockPos pos)
+    public void dispense(World worldIn, BlockPos pos)
     {
         BlockSourceImpl blocksourceimpl = new BlockSourceImpl(worldIn, pos);
         TileEntityDispenser tileentitydispenser = blocksourceimpl.getBlockTileEntity();
