--- before/net/minecraft/block/BlockShulkerBox.java
+++ after/net/minecraft/block/BlockShulkerBox.java
@@ -43,7 +43,7 @@
 public class BlockShulkerBox extends BlockContainer
 {
     public static final PropertyEnum<EnumFacing> field_190957_a = PropertyDirection.func_177714_a("facing");
-    private final EnumDyeColor field_190958_b;
+    public final EnumDyeColor field_190958_b;
 
     public BlockShulkerBox(EnumDyeColor p_i47248_1_)
     {
@@ -273,11 +273,11 @@
                 {
                     if (!itemstack.func_190926_b())
                     {
-                        j++;
+                        ++j;
 
                         if (i <= 4)
                         {
-                            i++;
+                            ++i;
                             p_190948_3_.add(String.format("%s x%d", itemstack.func_82833_r(), itemstack.func_190916_E()));
                         }
                     }
