--- before/net/minecraft/nbt/JsonToNBT.java
+++ after/net/minecraft/nbt/JsonToNBT.java
@@ -30,7 +30,7 @@
 
         if (this.canRead())
         {
-            this.cursor++;
+            ++this.cursor;
             throw this.exception("Trailing data found");
         }
         else
@@ -197,7 +197,7 @@
 
         while (this.canRead() && this.isAllowedInKey(this.peek()))
         {
-            this.cursor++;
+            ++this.cursor;
         }
 
         return this.string.substring(i, this.cursor);
@@ -341,7 +341,7 @@
 
     private <T extends Number> List<T> readArray(byte p_193603_1_, byte p_193603_2_) throws NBTException
     {
-        List<T> list = Lists.newArrayList();
+        List list = Lists.newArrayList();
 
         while (this.peek() != ']')
         {
@@ -355,15 +355,15 @@
 
             if (p_193603_2_ == 1)
             {
-                list.add((T)(Byte)((NBTPrimitive)nbtbase).getByte());
+                list.add(((NBTPrimitive)nbtbase).getByte());
             }
             else if (p_193603_2_ == 4)
             {
-                list.add((T)(Long)((NBTPrimitive)nbtbase).getLong());
+                list.add(((NBTPrimitive)nbtbase).getLong());
             }
             else
             {
-                list.add((T)(Integer)((NBTPrimitive)nbtbase).getInt());
+                list.add(((NBTPrimitive)nbtbase).getInt());
             }
 
             if (!this.hasElementSeparator())
@@ -385,7 +385,7 @@
     {
         while (this.canRead() && Character.isWhitespace(this.peek()))
         {
-            this.cursor++;
+            ++this.cursor;
         }
     }
 
@@ -395,7 +395,7 @@
 
         if (this.canRead() && this.peek() == ',')
         {
-            this.cursor++;
+            ++this.cursor;
             this.skipWhitespace();
             return true;
         }
@@ -412,7 +412,7 @@
 
         if (flag && this.peek() == expected)
         {
-            this.cursor++;
+            ++this.cursor;
         }
         else
         {
