--- before/net/minecraft/nbt/NBTUtil.java
+++ after/net/minecraft/nbt/NBTUtil.java
@@ -2,6 +2,7 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Optional;
+import com.google.common.collect.UnmodifiableIterator;
 import com.mojang.authlib.GameProfile;
 import com.mojang.authlib.properties.Property;
 import java.util.UUID;
@@ -61,7 +62,7 @@
                 {
                     NBTTagList nbttaglist = nbttagcompound.func_150295_c(s2, 10);
 
-                    for (int i = 0; i < nbttaglist.func_74745_c(); i++)
+                    for (int i = 0; i < nbttaglist.func_74745_c(); ++i)
                     {
                         NBTTagCompound nbttagcompound1 = nbttaglist.func_150305_b(i);
                         String s3 = nbttagcompound1.func_74779_i("Value");
@@ -175,12 +176,12 @@
             }
             else
             {
-                for (int i = 0; i < nbttaglist.func_74745_c(); i++)
+                for (int i = 0; i < nbttaglist.func_74745_c(); ++i)
                 {
                     NBTBase nbtbase = nbttaglist.func_179238_g(i);
                     boolean flag = false;
 
-                    for (int j = 0; j < nbttaglist1.func_74745_c(); j++)
+                    for (int j = 0; j < nbttaglist1.func_74745_c(); ++j)
                     {
                         if (func_181123_a(nbtbase, nbttaglist1.func_179238_g(j), p_181123_2_))
                         {
@@ -288,10 +289,12 @@
         if (!p_190009_1_.func_177228_b().isEmpty())
         {
             NBTTagCompound nbttagcompound = new NBTTagCompound();
+            UnmodifiableIterator unmodifiableiterator = p_190009_1_.func_177228_b().entrySet().iterator();
 
-            for (Entry < IProperty<?>, Comparable<? >> entry : p_190009_1_.func_177228_b().entrySet())
+            while (unmodifiableiterator.hasNext())
             {
-                IProperty<?> iproperty = entry.getKey();
+                Entry < IProperty<?>, Comparable<? >> entry = (Entry)unmodifiableiterator.next();
+                IProperty<?> iproperty = (IProperty)entry.getKey();
                 nbttagcompound.func_74778_a(iproperty.func_177701_a(), func_190010_a(iproperty, entry.getValue()));
             }
 
@@ -300,6 +303,7 @@
 
         return p_190009_0_;
     }
+
     @SuppressWarnings("unchecked")
     private static <T extends Comparable<T>> String func_190010_a(IProperty<T> p_190010_0_, Comparable<?> p_190010_1_)
     {
