--- before/net/minecraft/world/chunk/BlockStatePaletteLinear.java
+++ after/net/minecraft/world/chunk/BlockStatePaletteLinear.java
@@ -24,7 +24,7 @@
     @Override
     public int idFor(IBlockState state)
     {
-        for (int i = 0; i < this.arraySize; i++)
+        for (int i = 0; i < this.arraySize; ++i)
         {
             if (this.states[i] == state)
             {
@@ -37,7 +37,7 @@
         if (j < this.states.length)
         {
             this.states[j] = state;
-            this.arraySize++;
+            ++this.arraySize;
             return j;
         }
         else
@@ -59,7 +59,7 @@
     {
         this.arraySize = buf.readVarInt();
 
-        for (int i = 0; i < this.arraySize; i++)
+        for (int i = 0; i < this.arraySize; ++i)
         {
             this.states[i] = Block.BLOCK_STATE_IDS.getByValue(buf.readVarInt());
         }
@@ -70,7 +70,7 @@
     {
         buf.writeVarInt(this.arraySize);
 
-        for (int i = 0; i < this.arraySize; i++)
+        for (int i = 0; i < this.arraySize; ++i)
         {
             buf.writeVarInt(Block.BLOCK_STATE_IDS.get(this.states[i]));
         }
@@ -81,7 +81,7 @@
     {
         int i = PacketBuffer.getVarIntSize(this.arraySize);
 
-        for (int j = 0; j < this.arraySize; j++)
+        for (int j = 0; j < this.arraySize; ++j)
         {
             i += PacketBuffer.getVarIntSize(Block.BLOCK_STATE_IDS.get(this.states[j]));
         }
