--- before/net/minecraft/world/storage/MapData.java
+++ after/net/minecraft/world/storage/MapData.java
@@ -4,6 +4,7 @@
 import com.google.common.collect.Maps;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 import javax.annotation.Nullable;
 import net.minecraft.entity.item.EntityItemFrame;
 import net.minecraft.entity.player.EntityPlayer;
@@ -15,38 +16,70 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.world.World;
+import org.bukkit.craftbukkit.v1_12_R1.CraftServer;
+import org.bukkit.craftbukkit.v1_12_R1.CraftWorld;
+import org.bukkit.craftbukkit.v1_12_R1.map.CraftMapView;
 
 public class MapData extends WorldSavedData
 {
     public int xCenter;
     public int zCenter;
-    public byte dimension;
+    public int dimension; //FML byte -> int
     public boolean trackingPosition;
     public boolean unlimitedTracking;
     public byte scale;
     public byte[] colors = new byte[16384];
-    public List<MapData.MapInfo> playersArrayList = Lists.newArrayList();
-    private final Map<EntityPlayer, MapData.MapInfo> playersHashMap = Maps.newHashMap();
-    public Map<String, MapDecoration> mapDecorations = Maps.newLinkedHashMap();
+    public List<MapData.MapInfo> playersArrayList = Lists.<MapData.MapInfo>newArrayList();
+    private final Map<EntityPlayer, MapData.MapInfo> playersHashMap = Maps.<EntityPlayer, MapData.MapInfo>newHashMap();
+    public Map<String, MapDecoration> mapDecorations = Maps.<String, MapDecoration>newLinkedHashMap();
+
+    public final CraftMapView mapView;
+    private CraftServer server;
+    private UUID uniqueId = null;
 
     public MapData(String mapname)
     {
         super(mapname);
+        mapView = new CraftMapView(this);
+        server = (CraftServer) org.bukkit.Bukkit.getServer();
     }
 
     public void calculateMapCenter(double x, double z, int mapScale)
     {
         int i = 128 * (1 << mapScale);
-        int j = MathHelper.floor((x + 64.0) / (double)i);
-        int k = MathHelper.floor((z + 64.0) / (double)i);
+        int j = MathHelper.floor((x + 64.0D) / (double)i);
+        int k = MathHelper.floor((z + 64.0D) / (double)i);
         this.xCenter = j * i + i / 2 - 64;
         this.zCenter = k * i + i / 2 - 64;
     }
 
-    @Override
     public void readFromNBT(NBTTagCompound nbt)
     {
-        this.dimension = nbt.getByte("dimension");
+        // this.dimension = nbt.getInteger("dimension");
+        // CraftBukkit start
+        int dimension = nbt.getInteger("dimension");
+
+        if (dimension >= 10) {
+            long least = nbt.getLong("UUIDLeast");
+            long most = nbt.getLong("UUIDMost");
+
+            if (least != 0L && most != 0L) {
+                this.uniqueId = new UUID(most, least);
+
+                CraftWorld world = (CraftWorld) server.getWorld(this.uniqueId);
+                // Check if the stored world details are correct.
+                if (world == null) {
+                    /* All Maps which do not have their valid world loaded are set to a dimension which hopefully won't be reached.
+                       This is to prevent them being corrupted with the wrong map data. */
+                    dimension = 127;
+                } else {
+                    dimension = (byte) world.getHandle().dimension;
+                }
+            }
+        }
+
+        this.dimension = dimension;
+        // CraftBukkit end
         this.xCenter = nbt.getInteger("xCenter");
         this.zCenter = nbt.getInteger("zCenter");
         this.scale = nbt.getByte("scale");
@@ -76,13 +109,13 @@
             int k = (128 - i) / 2;
             int l = (128 - j) / 2;
 
-            for (int i1 = 0; i1 < j; i1++)
+            for (int i1 = 0; i1 < j; ++i1)
             {
                 int j1 = i1 + l;
 
                 if (j1 >= 0 || j1 < 128)
                 {
-                    for (int k1 = 0; k1 < i; k1++)
+                    for (int k1 = 0; k1 < i; ++k1)
                     {
                         int l1 = k1 + k;
 
@@ -96,10 +129,28 @@
         }
     }
 
-    @Override
     public NBTTagCompound writeToNBT(NBTTagCompound compound)
     {
-        compound.setByte("dimension", this.dimension);
+        // CraftBukkit start
+        if (this.dimension >= 10) {
+            if (this.uniqueId == null) {
+                for (org.bukkit.World world : server.getWorlds()) {
+                    CraftWorld cWorld = (CraftWorld) world;
+                    if (cWorld.getHandle().dimension == this.dimension) {
+                        this.uniqueId = cWorld.getUID();
+                        break;
+                    }
+                }
+            }
+            /* Perform a second check to see if a matching world was found, this is a necessary
+               change incase Maps are forcefully unlinked from a World and lack a UID.*/
+            if (this.uniqueId != null) {
+                compound.setLong("UUIDLeast", this.uniqueId.getLeastSignificantBits());
+                compound.setLong("UUIDMost", this.uniqueId.getMostSignificantBits());
+            }
+        }
+        // CraftBukkit end
+        compound.setInteger("dimension", this.dimension);
         compound.setInteger("xCenter", this.xCenter);
         compound.setInteger("zCenter", this.zCenter);
         compound.setByte("scale", this.scale);
@@ -125,23 +176,15 @@
             this.mapDecorations.remove(player.getName());
         }
 
-        for (int i = 0; i < this.playersArrayList.size(); i++)
+        for (int i = 0; i < this.playersArrayList.size(); ++i)
         {
             MapData.MapInfo mapdata$mapinfo1 = this.playersArrayList.get(i);
 
-            if (!mapdata$mapinfo1.player.isDead
-                    && (mapdata$mapinfo1.player.inventory.hasItemStack(mapStack) || mapStack.isOnItemFrame()))
+            if (!mapdata$mapinfo1.player.isDead && (mapdata$mapinfo1.player.inventory.hasItemStack(mapStack) || mapStack.isOnItemFrame()))
             {
                 if (!mapStack.isOnItemFrame() && mapdata$mapinfo1.player.dimension == this.dimension && this.trackingPosition)
                 {
-                    this.updateDecorations(
-                        MapDecoration.Type.PLAYER,
-                        mapdata$mapinfo1.player.world,
-                        mapdata$mapinfo1.player.getName(),
-                        mapdata$mapinfo1.player.posX,
-                        mapdata$mapinfo1.player.posZ,
-                        (double)mapdata$mapinfo1.player.rotationYaw
-                    );
+                    this.updateDecorations(MapDecoration.Type.PLAYER, mapdata$mapinfo1.player.world, mapdata$mapinfo1.player.getName(), mapdata$mapinfo1.player.posX, mapdata$mapinfo1.player.posZ, (double)mapdata$mapinfo1.player.rotationYaw);
                 }
             }
             else
@@ -155,34 +198,20 @@
         {
             EntityItemFrame entityitemframe = mapStack.getItemFrame();
             BlockPos blockpos = entityitemframe.getHangingPosition();
-            this.updateDecorations(
-                MapDecoration.Type.FRAME,
-                player.world,
-                "frame-" + entityitemframe.getEntityId(),
-                (double)blockpos.getX(),
-                (double)blockpos.getZ(),
-                (double)(entityitemframe.facingDirection.getHorizontalIndex() * 90)
-            );
+            this.updateDecorations(MapDecoration.Type.FRAME, player.world, "frame-" + entityitemframe.getEntityId(), (double)blockpos.getX(), (double)blockpos.getZ(), (double)(entityitemframe.facingDirection.getHorizontalIndex() * 90));
         }
 
         if (mapStack.hasTagCompound() && mapStack.getTagCompound().hasKey("Decorations", 9))
         {
             NBTTagList nbttaglist = mapStack.getTagCompound().getTagList("Decorations", 10);
 
-            for (int j = 0; j < nbttaglist.tagCount(); j++)
+            for (int j = 0; j < nbttaglist.tagCount(); ++j)
             {
                 NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(j);
 
                 if (!this.mapDecorations.containsKey(nbttagcompound.getString("id")))
                 {
-                    this.updateDecorations(
-                        MapDecoration.Type.byIcon(nbttagcompound.getByte("type")),
-                        player.world,
-                        nbttagcompound.getString("id"),
-                        nbttagcompound.getDouble("x"),
-                        nbttagcompound.getDouble("z"),
-                        nbttagcompound.getDouble("rot")
-                    );
+                    this.updateDecorations(MapDecoration.Type.byIcon(nbttagcompound.getByte("type")), player.world, nbttagcompound.getString("id"), nbttagcompound.getDouble("x"), nbttagcompound.getDouble("z"), nbttagcompound.getDouble("rot"));
                 }
             }
         }
@@ -207,7 +236,7 @@
         nbttagcompound.setString("id", decorationName);
         nbttagcompound.setDouble("x", (double)target.getX());
         nbttagcompound.setDouble("z", (double)target.getZ());
-        nbttagcompound.setDouble("rot", 180.0);
+        nbttagcompound.setDouble("rot", 180.0D);
         nbttaglist.appendTag(nbttagcompound);
 
         if (type.hasMapColor())
@@ -222,19 +251,19 @@
         int i = 1 << this.scale;
         float f = (float)(worldX - (double)this.xCenter) / (float)i;
         float f1 = (float)(worldZ - (double)this.zCenter) / (float)i;
-        byte b0 = (byte)((int)((double)(f * 2.0F) + 0.5));
-        byte b1 = (byte)((int)((double)(f1 * 2.0F) + 0.5));
+        byte b0 = (byte)((int)((double)(f * 2.0F) + 0.5D));
+        byte b1 = (byte)((int)((double)(f1 * 2.0F) + 0.5D));
         int j = 63;
         byte b2;
 
         if (f >= -63.0F && f1 >= -63.0F && f <= 63.0F && f1 <= 63.0F)
         {
-            rotationIn += rotationIn < 0.0 ? -8.0 : 8.0;
-            b2 = (byte)((int)(rotationIn * 16.0 / 360.0));
+            rotationIn = rotationIn + (rotationIn < 0.0D ? -8.0D : 8.0D);
+            b2 = (byte)((int)(rotationIn * 16.0D / 360.0D));
 
-            if (this.dimension < 0)
+            if (worldIn.provider.shouldMapSpin(decorationName, worldX, worldZ, rotationIn))
             {
-                int l = (int)(worldIn.getWorldInfo().getWorldTime() / 10L);
+                int l = (int)(worldIn.getWorldTime() / 10L);
                 b2 = (byte)(l * l * 34187121 + l * 121 >> 15 & 15);
             }
         }
@@ -339,36 +368,25 @@
         @Nullable
         public Packet<?> getPacket(ItemStack stack)
         {
+            org.bukkit.craftbukkit.v1_12_R1.map.RenderData render = MapData.this.mapView.render((org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer) this.player.getBukkitEntity());
+
+            java.util.Collection<MapDecoration> icons = new java.util.ArrayList<>();
+
+            for ( org.bukkit.map.MapCursor cursor : render.cursors) {
+
+                if (cursor.isVisible()) {
+                    icons.add(new MapDecoration(MapDecoration.Type.byIcon(cursor.getRawType()), cursor.getX(), cursor.getY(), cursor.getDirection()));
+                }
+            }
+
             if (this.isDirty)
             {
                 this.isDirty = false;
-                return new SPacketMaps(
-                           stack.getMetadata(),
-                           MapData.this.scale,
-                           MapData.this.trackingPosition,
-                           MapData.this.mapDecorations.values(),
-                           MapData.this.colors,
-                           this.minX,
-                           this.minY,
-                           this.maxX + 1 - this.minX,
-                           this.maxY + 1 - this.minY
-                       );
+                return new SPacketMaps(stack.getMetadata(), MapData.this.scale, MapData.this.trackingPosition, icons, render.buffer, this.minX, this.minY, this.maxX + 1 - this.minX, this.maxY + 1 - this.minY);
             }
             else
             {
-                return this.tick++ % 5 == 0
-                       ? new SPacketMaps(
-                           stack.getMetadata(),
-                           MapData.this.scale,
-                           MapData.this.trackingPosition,
-                           MapData.this.mapDecorations.values(),
-                           MapData.this.colors,
-                           0,
-                           0,
-                           0,
-                           0
-                       )
-                       : null;
+                return this.tick++ % 5 == 0 ? new SPacketMaps(stack.getMetadata(), MapData.this.scale, MapData.this.trackingPosition, icons, render.buffer, 0, 0, 0, 0) : null;
             }
         }
 
