--- before/net/minecraft/world/storage/ThreadedFileIOBase.java
+++ after/net/minecraft/world/storage/ThreadedFileIOBase.java
@@ -35,7 +35,7 @@
 
     private void processQueue()
     {
-        for (int i = 0; i < this.threadedIOQueue.size(); i++)
+        for (int i = 0; i < this.threadedIOQueue.size(); ++i)
         {
             IThreadedFileIO ithreadedfileio = this.threadedIOQueue.get(i);
             boolean flag = ithreadedfileio.writeNextIO();
@@ -43,9 +43,10 @@
             if (!flag)
             {
                 this.threadedIOQueue.remove(i--);
-                this.savedIOCounter++;
+                ++this.savedIOCounter;
             }
 
+            /*
             try
             {
                 Thread.sleep(this.isThreadWaiting ? 0L : 10L);
@@ -54,6 +55,7 @@
             {
                 interruptedexception1.printStackTrace();
             }
+            */
         }
 
         if (this.threadedIOQueue.isEmpty())
@@ -73,7 +75,7 @@
     {
         if (!this.threadedIOQueue.contains(fileIo))
         {
-            this.writeQueuedCounter++;
+            ++this.writeQueuedCounter;
             this.threadedIOQueue.add(fileIo);
         }
     }
