--- before/net/minecraft/world/storage/loot/functions/LootFunctionManager.java
+++ after/net/minecraft/world/storage/loot/functions/LootFunctionManager.java
@@ -56,7 +56,7 @@
 
     public static <T extends LootFunction> LootFunction.Serializer<T> getSerializerFor(T functionClass)
     {
-        LootFunction.Serializer<T> serializer = (LootFunction.Serializer<T>)CLASS_TO_SERIALIZER_MAP.get(functionClass.getClass());
+        LootFunction.Serializer<T> serializer = (LootFunction.Serializer<T>) CLASS_TO_SERIALIZER_MAP.get(functionClass.getClass());
 
         if (serializer == null)
         {
