--- before/net/minecraft/world/gen/feature/WorldGenSand.java
+++ after/net/minecraft/world/gen/feature/WorldGenSand.java
@@ -30,16 +30,16 @@
             int i = rand.nextInt(this.radius - 2) + 2;
             int j = 2;
 
-            for (int k = position.getX() - i; k <= position.getX() + i; k++)
+            for (int k = position.getX() - i; k <= position.getX() + i; ++k)
             {
-                for (int l = position.getZ() - i; l <= position.getZ() + i; l++)
+                for (int l = position.getZ() - i; l <= position.getZ() + i; ++l)
                 {
                     int i1 = k - position.getX();
                     int j1 = l - position.getZ();
 
                     if (i1 * i1 + j1 * j1 <= i * i)
                     {
-                        for (int k1 = position.getY() - 2; k1 <= position.getY() + 2; k1++)
+                        for (int k1 = position.getY() - 2; k1 <= position.getY() + 2; ++k1)
                         {
                             BlockPos blockpos = new BlockPos(k, k1, l);
                             Block block = worldIn.getBlockState(blockpos).getBlock();
