--- before/net/minecraft/world/gen/feature/WorldGenCactus.java
+++ after/net/minecraft/world/gen/feature/WorldGenCactus.java
@@ -10,7 +10,7 @@
     @Override
     public boolean generate(World worldIn, Random rand, BlockPos position)
     {
-        for (int i = 0; i < 10; i++)
+        for (int i = 0; i < 10; ++i)
         {
             BlockPos blockpos = position.add(
                                     rand.nextInt(8) - rand.nextInt(8), rand.nextInt(4) - rand.nextInt(4), rand.nextInt(8) - rand.nextInt(8)
@@ -20,7 +20,7 @@
             {
                 int j = 1 + rand.nextInt(rand.nextInt(3) + 1);
 
-                for (int k = 0; k < j; k++)
+                for (int k = 0; k < j; ++k)
                 {
                     if (Blocks.CACTUS.canBlockStay(worldIn, blockpos))
                     {
