--- before/net/minecraft/world/gen/feature/WorldGenVines.java
+++ after/net/minecraft/world/gen/feature/WorldGenVines.java
@@ -13,7 +13,7 @@
     @Override
     public boolean generate(World worldIn, Random rand, BlockPos position)
     {
-        while (position.getY() < 128)
+        for (; position.getY() < 128; position = position.up())
         {
             if (worldIn.isAirBlock(position))
             {
@@ -36,8 +36,6 @@
             {
                 position = position.add(rand.nextInt(4) - rand.nextInt(4), 0, rand.nextInt(4) - rand.nextInt(4));
             }
-
-            position = position.up();
         }
 
         return true;
