--- before/net/minecraft/world/gen/feature/WorldGenSpikes.java
+++ after/net/minecraft/world/gen/feature/WorldGenSpikes.java
@@ -57,9 +57,9 @@
 
             if (this.field_186146_b.func_186150_e())
             {
-                for (int j = -2; j <= 2; j++)
+                for (int j = -2; j <= 2; ++j)
                 {
-                    for (int k = -2; k <= 2; k++)
+                    for (int k = -2; k <= 2; ++k)
                     {
                         if (MathHelper.func_76130_a(j) == 2 || MathHelper.func_76130_a(k) == 2)
                         {
