--- before/net/minecraft/world/gen/feature/WorldGenFossils.java
+++ after/net/minecraft/world/gen/feature/WorldGenFossils.java
@@ -61,9 +61,9 @@
         int k = random.nextInt(16 - blockpos.getZ());
         int l = 256;
 
-        for (int i1 = 0; i1 < blockpos.getX(); i1++)
+        for (int i1 = 0; i1 < blockpos.getX(); ++i1)
         {
-            for (int j1 = 0; j1 < blockpos.getX(); j1++)
+            for (int j1 = 0; j1 < blockpos.getX(); ++j1)
             {
                 l = Math.min(l, worldIn.getHeight(position.getX() + i1 + j, position.getZ() + j1 + k));
             }
