--- before/net/minecraft/world/gen/feature/WorldGenReed.java
+++ after/net/minecraft/world/gen/feature/WorldGenReed.java
@@ -11,7 +11,7 @@
     @Override
     public boolean generate(World worldIn, Random rand, BlockPos position)
     {
-        for (int i = 0; i < 20; i++)
+        for (int i = 0; i < 20; ++i)
         {
             BlockPos blockpos = position.add(rand.nextInt(4) - rand.nextInt(4), 0, rand.nextInt(4) - rand.nextInt(4));
 
@@ -26,7 +26,7 @@
                 {
                     int j = 2 + rand.nextInt(rand.nextInt(3) + 1);
 
-                    for (int k = 0; k < j; k++)
+                    for (int k = 0; k < j; ++k)
                     {
                         if (Blocks.REEDS.canBlockStay(worldIn, blockpos))
                         {
