--- before/net/minecraft/world/gen/feature/WorldGenBlockBlob.java
+++ after/net/minecraft/world/gen/feature/WorldGenBlockBlob.java
@@ -21,7 +21,7 @@
     @Override
     public boolean generate(World worldIn, Random rand, BlockPos position)
     {
-        while (position.getY() > 3)
+        for (; position.getY() > 3; position = position.down())
         {
             if (!worldIn.isAirBlock(position.down()))
             {
@@ -32,8 +32,6 @@
                     break;
                 }
             }
-
-            position = position.down();
         }
 
         if (position.getY() <= 3)
@@ -44,7 +42,7 @@
         {
             int i1 = this.startRadius;
 
-            for (int i = 0; i1 >= 0 && i < 3; i++)
+            for (int i = 0; i1 >= 0 && i < 3; ++i)
             {
                 int j = i1 + rand.nextInt(2);
                 int k = i1 + rand.nextInt(2);
