--- before/net/minecraft/world/gen/feature/WorldGenFlowers.java
+++ after/net/minecraft/world/gen/feature/WorldGenFlowers.java
@@ -25,7 +25,7 @@
     @Override
     public boolean generate(World worldIn, Random rand, BlockPos position)
     {
-        for (int i = 0; i < 64; i++)
+        for (int i = 0; i < 64; ++i)
         {
             BlockPos blockpos = position.add(
                                     rand.nextInt(8) - rand.nextInt(8), rand.nextInt(4) - rand.nextInt(4), rand.nextInt(8) - rand.nextInt(8)
