--- before/net/minecraft/command/AdvancementCommand.java
+++ after/net/minecraft/command/AdvancementCommand.java
@@ -429,7 +429,7 @@
             {
                 if (this.func_193537_a(p_193532_1_, advancement))
                 {
-                    i++;
+                    ++i;
                 }
             }
 
@@ -492,7 +492,7 @@
 
         static
         {
-            for (int i = 0; i < values().length; i++)
+            for (int i = 0; i < values().length; ++i)
             {
                 field_193553_f[i] = values()[i].field_193554_g;
             }
