--- before/net/minecraft/command/CommandEnchant.java
+++ after/net/minecraft/command/CommandEnchant.java
@@ -80,7 +80,7 @@
                     {
                         NBTTagList nbttaglist = itemstack.func_77986_q();
 
-                        for (int j = 0; j < nbttaglist.func_74745_c(); j++)
+                        for (int j = 0; j < nbttaglist.func_74745_c(); ++j)
                         {
                             int k = nbttaglist.func_150305_b(j).func_74765_d("id");
 
