--- before/net/minecraft/command/CommandCompare.java
+++ after/net/minecraft/command/CommandCompare.java
@@ -78,11 +78,11 @@
                     BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
                     BlockPos.MutableBlockPos blockpos$mutableblockpos1 = new BlockPos.MutableBlockPos();
 
-                    for (int j = structureboundingbox.minZ; j <= structureboundingbox.maxZ; j++)
+                    for (int j = structureboundingbox.minZ; j <= structureboundingbox.maxZ; ++j)
                     {
-                        for (int k = structureboundingbox.minY; k <= structureboundingbox.maxY; k++)
+                        for (int k = structureboundingbox.minY; k <= structureboundingbox.maxY; ++k)
                         {
-                            for (int l = structureboundingbox.minX; l <= structureboundingbox.maxX; l++)
+                            for (int l = structureboundingbox.minX; l <= structureboundingbox.maxX; ++l)
                             {
                                 blockpos$mutableblockpos.setPos(l, k, j);
                                 blockpos$mutableblockpos1.setPos(l + blockpos3.getX(), k + blockpos3.getY(), j + blockpos3.getZ());
@@ -122,7 +122,7 @@
                                         flag1 = true;
                                     }
 
-                                    i++;
+                                    ++i;
 
                                     if (flag1)
                                     {
