--- before/net/minecraft/command/CommandSpreadPlayers.java
+++ after/net/minecraft/command/CommandSpreadPlayers.java
@@ -10,6 +10,7 @@
 import java.util.Set;
 import javax.annotation.Nullable;
 import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.scoreboard.Team;
@@ -18,6 +19,7 @@
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.text.TextComponentTranslation;
 import net.minecraft.world.World;
+import net.minecraft.world.gen.ChunkProviderServer;
 
 public class CommandSpreadPlayers extends CommandBase
 {
@@ -172,18 +174,18 @@
         double d0 = Float.MAX_VALUE;
         int i;
 
-        for (i = 0; i < 10000 && flag; i++)
+        for (i = 0; i < 10000 && flag; ++i)
         {
             flag = false;
             d0 = Float.MAX_VALUE;
 
-            for (int j = 0; j < p_110668_14_.length; j++)
+            for (int j = 0; j < p_110668_14_.length; ++j)
             {
                 CommandSpreadPlayers.Position commandspreadplayers$position = p_110668_14_[j];
                 int k = 0;
                 CommandSpreadPlayers.Position commandspreadplayers$position1 = new CommandSpreadPlayers.Position();
 
-                for (int l = 0; l < p_110668_14_.length; l++)
+                for (int l = 0; l < p_110668_14_.length; ++l)
                 {
                     if (j != l)
                     {
@@ -193,11 +195,11 @@
 
                         if (d1 < p_110668_2_)
                         {
-                            k++;
-                            commandspreadplayers$position1.field_111101_a = commandspreadplayers$position1.field_111101_a
-                                    + (commandspreadplayers$position2.field_111101_a - commandspreadplayers$position.field_111101_a);
-                            commandspreadplayers$position1.field_111100_b = commandspreadplayers$position1.field_111100_b
-                                    + (commandspreadplayers$position2.field_111100_b - commandspreadplayers$position.field_111100_b);
+                            ++k;
+                            commandspreadplayers$position1.field_111101_a += commandspreadplayers$position2.field_111101_a
+                                    - commandspreadplayers$position.field_111101_a;
+                            commandspreadplayers$position1.field_111100_b += commandspreadplayers$position2.field_111100_b
+                                    - commandspreadplayers$position.field_111100_b;
                         }
                     }
                 }
@@ -262,7 +264,7 @@
         int i = 0;
         Map<Team, CommandSpreadPlayers.Position> map = Maps.newHashMap();
 
-        for (int j = 0; j < p_110671_1_.size(); j++)
+        for (int j = 0; j < p_110671_1_.size(); ++j)
         {
             Entity entity = p_110671_1_.get(j);
             CommandSpreadPlayers.Position commandspreadplayers$position;
@@ -311,7 +313,7 @@
     {
         CommandSpreadPlayers.Position[] acommandspreadplayers$position = new CommandSpreadPlayers.Position[p_110670_2_];
 
-        for (int i = 0; i < acommandspreadplayers$position.length; i++)
+        for (int i = 0; i < acommandspreadplayers$position.length; ++i)
         {
             CommandSpreadPlayers.Position commandspreadplayers$position = new CommandSpreadPlayers.Position();
             commandspreadplayers$position.func_111097_a(p_110670_1_, p_110670_3_, p_110670_5_, p_110670_7_, p_110670_9_);
@@ -363,8 +365,8 @@
 
             public void func_111094_b(CommandSpreadPlayers.Position p_111094_1_)
             {
-                this.field_111101_a = this.field_111101_a - p_111094_1_.field_111101_a;
-                this.field_111100_b = this.field_111100_b - p_111094_1_.field_111100_b;
+                this.field_111101_a -= p_111094_1_.field_111101_a;
+                this.field_111100_b -= p_111094_1_.field_111100_b;
             }
 
             public boolean func_111093_a(double p_111093_1_, double p_111093_3_, double p_111093_5_, double p_111093_7_)
@@ -404,7 +406,7 @@
                 {
                     blockpos = blockpos.func_177977_b();
 
-                    if (p_111092_1_.func_180495_p(blockpos).func_185904_a() != Material.field_151579_a)
+                    if (getBlockState(p_111092_1_, blockpos).func_185904_a() != Material.field_151579_a)
                     {
                         return blockpos.func_177956_o() + 1;
                     }
@@ -420,7 +422,7 @@
                 while (blockpos.func_177956_o() > 0)
                 {
                     blockpos = blockpos.func_177977_b();
-                    Material material = p_111098_1_.func_180495_p(blockpos).func_185904_a();
+                    Material material = getBlockState(p_111098_1_, blockpos).func_185904_a();
 
                     if (material != Material.field_151579_a)
                     {
@@ -436,5 +438,12 @@
                 this.field_111101_a = MathHelper.func_82716_a(p_111097_1_, p_111097_2_, p_111097_6_);
                 this.field_111100_b = MathHelper.func_82716_a(p_111097_1_, p_111097_4_, p_111097_8_);
             }
+
+            // CraftBukkit start - add a version of getBlockState which force loads chunks
+            private static IBlockState getBlockState(World world, BlockPos position) {
+                ((ChunkProviderServer) world.func_72863_F()).func_186028_c(position.func_177958_n() >> 4, position.func_177952_p() >> 4);
+                return world.func_180495_p(position);
+            }
+            // CraftBukkit end
         }
 }
