--- before/net/minecraft/command/CommandEnchant.java
+++ after/net/minecraft/command/CommandEnchant.java
@@ -80,7 +80,7 @@
                     {
                         NBTTagList nbttaglist = itemstack.getEnchantmentTagList();
 
-                        for (int j = 0; j < nbttaglist.tagCount(); j++)
+                        for (int j = 0; j < nbttaglist.tagCount(); ++j)
                         {
                             int k = nbttaglist.getCompoundTagAt(j).getShort("id");
 
