--- before/net/minecraft/command/CommandClone.java
+++ after/net/minecraft/command/CommandClone.java
@@ -120,11 +120,11 @@
                                 structureboundingbox1.minZ - structureboundingbox.minZ
                             );
 
-                            for (int j = structureboundingbox.minZ; j <= structureboundingbox.maxZ; j++)
+                            for (int j = structureboundingbox.minZ; j <= structureboundingbox.maxZ; ++j)
                             {
-                                for (int k = structureboundingbox.minY; k <= structureboundingbox.maxY; k++)
+                                for (int k = structureboundingbox.minY; k <= structureboundingbox.maxY; ++k)
                                 {
-                                    for (int l = structureboundingbox.minX; l <= structureboundingbox.maxX; l++)
+                                    for (int l = structureboundingbox.minX; l <= structureboundingbox.maxX; ++l)
                                     {
                                         BlockPos blockpos4 = new BlockPos(l, k, j);
                                         BlockPos blockpos5 = blockpos4.add(blockpos3);
@@ -200,7 +200,7 @@
                             {
                                 if (world.setBlockState(commandclone$staticclonedata1.pos, commandclone$staticclonedata1.blockState, 2))
                                 {
-                                    i++;
+                                    ++i;
                                 }
                             }
 
