--- before/net/minecraft/command/CommandHelp.java
+++ after/net/minecraft/command/CommandHelp.java
@@ -101,7 +101,7 @@
             textcomponenttranslation1.getStyle().setColor(TextFormatting.DARK_GREEN);
             sender.sendMessage(textcomponenttranslation1);
 
-            for (int i1 = k * 7; i1 < l; i1++)
+            for (int i1 = k * 7; i1 < l; ++i1)
             {
                 ICommand icommand1 = list.get(i1);
                 TextComponentTranslation textcomponenttranslation = new TextComponentTranslation(icommand1.getUsage(sender));
