--- before/net/minecraft/client/gui/recipebook/GhostRecipe.java
+++ after/net/minecraft/client/gui/recipebook/GhostRecipe.java
@@ -66,7 +66,7 @@
         RenderHelper.enableGUIStandardItemLighting();
         GlStateManager.disableLighting();
 
-        for (int i = 0; i < this.ingredients.size(); i++)
+        for (int i = 0; i < this.ingredients.size(); ++i)
         {
             GhostRecipe.GhostIngredient ghostrecipe$ghostingredient = this.ingredients.get(i);
             int j = ghostrecipe$ghostingredient.getX() + p_194188_2_;
