--- before/net/minecraft/client/gui/recipebook/GuiButtonRecipe.java
+++ after/net/minecraft/client/gui/recipebook/GuiButtonRecipe.java
@@ -110,7 +110,7 @@
             if (this.list.hasSingleResultItem() && this.getOrderedRecipes().size() > 1)
             {
                 mc.getRenderItem().renderItemAndEffectIntoGUI(itemstack, this.x + k + 1, this.y + k + 1);
-                k--;
+                --k;
             }
 
             mc.getRenderItem().renderItemAndEffectIntoGUI(itemstack, this.x + k, this.y + k);
