--- before/net/minecraft/client/gui/recipebook/RecipeBookPage.java
+++ after/net/minecraft/client/gui/recipebook/RecipeBookPage.java
@@ -30,7 +30,7 @@
 
     public RecipeBookPage()
     {
-        for (int i = 0; i < 20; i++)
+        for (int i = 0; i < 20; ++i)
         {
             this.buttons.add(new GuiButtonRecipe());
         }
@@ -41,7 +41,7 @@
         this.minecraft = p_194194_1_;
         this.recipeBook = p_194194_1_.player.getRecipeBook();
 
-        for (int i = 0; i < this.buttons.size(); i++)
+        for (int i = 0; i < this.buttons.size(); ++i)
         {
             this.buttons.get(i).setPosition(p_194194_2_ + 11 + 25 * (i % 5), p_194194_3_ + 31 + 25 * (i / 5));
         }
@@ -75,7 +75,7 @@
     {
         int i = 20 * this.currentPage;
 
-        for (int j = 0; j < this.buttons.size(); j++)
+        for (int j = 0; j < this.buttons.size(); ++j)
         {
             GuiButtonRecipe guibuttonrecipe = this.buttons.get(j);
 
@@ -174,14 +174,14 @@
         else if (this.forwardButton.mousePressed(this.minecraft, p_194196_1_, p_194196_2_) && p_194196_3_ == 0)
         {
             this.forwardButton.playPressSound(this.minecraft.getSoundHandler());
-            this.currentPage++;
+            ++this.currentPage;
             this.updateButtonsForPage();
             return true;
         }
         else if (this.backButton.mousePressed(this.minecraft, p_194196_1_, p_194196_2_) && p_194196_3_ == 0)
         {
             this.backButton.playPressSound(this.minecraft.getSoundHandler());
-            this.currentPage--;
+            --this.currentPage;
             this.updateButtonsForPage();
             return true;
         }
