--- before/net/minecraft/client/gui/recipebook/RecipeList.java
+++ after/net/minecraft/client/gui/recipebook/RecipeList.java
@@ -26,7 +26,7 @@
 
     public void updateKnownRecipes(RecipeBook book)
     {
-        for (int i = 0; i < this.recipes.size(); i++)
+        for (int i = 0; i < this.recipes.size(); ++i)
         {
             this.inBook.set(i, book.isUnlocked(this.recipes.get(i)));
         }
@@ -34,7 +34,7 @@
 
     public void canCraft(RecipeItemHelper handler, int width, int height, RecipeBook book)
     {
-        for (int i = 0; i < this.recipes.size(); i++)
+        for (int i = 0; i < this.recipes.size(); ++i)
         {
             IRecipe irecipe = this.recipes.get(i);
             boolean flag = irecipe.canFit(width, height) && book.isUnlocked(irecipe);
