--- before/net/minecraft/util/datafix/walkers/Filtered.java
+++ after/net/minecraft/util/datafix/walkers/Filtered.java
@@ -16,11 +16,11 @@
     {
         if (Entity.class.isAssignableFrom(p_i47309_1_))
         {
-            this.key = EntityList.getKey((Class <? extends Entity >)p_i47309_1_);
+            this.key = EntityList.getKey((Class<Entity>)p_i47309_1_);
         }
         else if (TileEntity.class.isAssignableFrom(p_i47309_1_))
         {
-            this.key = TileEntity.getKey((Class <? extends TileEntity >)p_i47309_1_);
+            this.key = TileEntity.getKey((Class<TileEntity>)p_i47309_1_);
         }
         else
         {
