--- before/net/minecraft/client/renderer/entity/RenderIllusionIllager.java
+++ after/net/minecraft/client/renderer/entity/RenderIllusionIllager.java
@@ -69,7 +69,7 @@
             Vec3d[] avec3d = ((EntityIllusionIllager)entity).getRenderLocations(partialTicks);
             float f = this.handleRotationFloat(entity, partialTicks);
 
-            for (int i = 0; i < avec3d.length; i++)
+            for (int i = 0; i < avec3d.length; ++i)
             {
                 super.doRender(
                     entity,
