--- before/net/minecraft/client/renderer/entity/RenderArrow.java
+++ after/net/minecraft/client/renderer/entity/RenderArrow.java
@@ -72,7 +72,7 @@
         bufferbuilder.pos(-7.0, -2.0, -2.0).tex(0.0, 0.3125).endVertex();
         tessellator.draw();
 
-        for (int j = 0; j < 4; j++)
+        for (int j = 0; j < 4; ++j)
         {
             GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
             GlStateManager.glNormal3f(0.0F, 0.0F, 0.05625F);
