--- before/net/minecraft/client/renderer/entity/RenderLiving.java
+++ after/net/minecraft/client/renderer/entity/RenderLiving.java
@@ -117,7 +117,7 @@
             double d16 = 0.025;
             bufferbuilder.begin(5, DefaultVertexFormats.POSITION_COLOR);
 
-            for (int j = 0; j <= 24; j++)
+            for (int j = 0; j <= 24; ++j)
             {
                 float f = 0.5F;
                 float f1 = 0.4F;
@@ -150,7 +150,7 @@
             tessellator.draw();
             bufferbuilder.begin(5, DefaultVertexFormats.POSITION_COLOR);
 
-            for (int k = 0; k <= 24; k++)
+            for (int k = 0; k <= 24; ++k)
             {
                 float f4 = 0.5F;
                 float f5 = 0.4F;
