--- before/net/minecraft/client/renderer/entity/RenderArrow.java
+++ after/net/minecraft/client/renderer/entity/RenderArrow.java
@@ -72,7 +72,7 @@
         bufferbuilder.func_181662_b(-7.0, -2.0, -2.0).func_187315_a(0.0, 0.3125).func_181675_d();
         tessellator.func_78381_a();
 
-        for (int j = 0; j < 4; j++)
+        for (int j = 0; j < 4; ++j)
         {
             GlStateManager.func_179114_b(90.0F, 1.0F, 0.0F, 0.0F);
             GlStateManager.func_187432_a(0.0F, 0.0F, 0.05625F);
