--- before/net/minecraft/tileentity/TileEntityEnchantmentTable.java
+++ after/net/minecraft/tileentity/TileEntityEnchantmentTable.java
@@ -79,7 +79,7 @@
 
                 do
                 {
-                    this.flipT = this.flipT + (float)(rand.nextInt(4) - rand.nextInt(4));
+                    this.flipT += (float)(rand.nextInt(4) - rand.nextInt(4));
                 }
                 while (f1 == this.flipT);
             }
@@ -124,13 +124,13 @@
 
         this.bookRotation += f2 * 0.4F;
         this.bookSpread = MathHelper.clamp(this.bookSpread, 0.0F, 1.0F);
-        this.tickCount++;
+        ++this.tickCount;
         this.pageFlipPrev = this.pageFlip;
         float f = (this.flipT - this.pageFlip) * 0.4F;
         float f3 = 0.2F;
         f = MathHelper.clamp(f, -0.2F, 0.2F);
-        this.flipA = this.flipA + (f - this.flipA) * 0.9F;
-        this.pageFlip = this.pageFlip + this.flipA;
+        this.flipA += (f - this.flipA) * 0.9F;
+        this.pageFlip += this.flipA;
     }
 
     @Override
