--- before/net/minecraft/tileentity/TileEntityEnchantmentTable.java
+++ after/net/minecraft/tileentity/TileEntityEnchantmentTable.java
@@ -79,7 +79,7 @@
 
                 do
                 {
-                    this.field_145932_k = this.field_145932_k + (float)(field_145923_r.nextInt(4) - field_145923_r.nextInt(4));
+                    this.field_145932_k += (float)(field_145923_r.nextInt(4) - field_145923_r.nextInt(4));
                 }
                 while (f1 == this.field_145932_k);
             }
@@ -124,13 +124,13 @@
 
         this.field_145928_o += f2 * 0.4F;
         this.field_145930_m = MathHelper.func_76131_a(this.field_145930_m, 0.0F, 1.0F);
-        this.field_145926_a++;
+        ++this.field_145926_a;
         this.field_145931_j = this.field_145933_i;
         float f = (this.field_145932_k - this.field_145933_i) * 0.4F;
         float f3 = 0.2F;
         f = MathHelper.func_76131_a(f, -0.2F, 0.2F);
-        this.field_145929_l = this.field_145929_l + (f - this.field_145929_l) * 0.9F;
-        this.field_145933_i = this.field_145933_i + this.field_145929_l;
+        this.field_145929_l += (f - this.field_145929_l) * 0.9F;
+        this.field_145933_i += this.field_145929_l;
     }
 
     @Override
