--- before/net/minecraft/tileentity/TileEntityMobSpawner.java
+++ after/net/minecraft/tileentity/TileEntityMobSpawner.java
@@ -61,7 +61,7 @@
                     {
                         NBTTagList nbttaglist = compound.getTagList("SpawnPotentials", 10);
 
-                        for (int i = 0; i < nbttaglist.tagCount(); i++)
+                        for (int i = 0; i < nbttaglist.tagCount(); ++i)
                         {
                             NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
                             nbttagcompound.setTag("Entity", fixer.process(FixTypes.ENTITY, nbttagcompound.getCompoundTag("Entity"), versionIn));
