--- before/net/minecraft/tileentity/TileEntityEnderChest.java
+++ after/net/minecraft/tileentity/TileEntityEnderChest.java
@@ -111,27 +111,32 @@
 
     public void openChest()
     {
-        this.numPlayersUsing++;
+        ++this.numPlayersUsing;
         this.world.addBlockEvent(this.pos, Blocks.ENDER_CHEST, 1, this.numPlayersUsing);
     }
 
     public void closeChest()
     {
-        this.numPlayersUsing--;
+        --this.numPlayersUsing;
         this.world.addBlockEvent(this.pos, Blocks.ENDER_CHEST, 1, this.numPlayersUsing);
     }
 
     public boolean canBeUsed(EntityPlayer player)
     {
-        return this.world.getTileEntity(this.pos) != this
-               ? false
-               : !(
-                   player.getDistanceSq(
-                       (double)this.pos.getX() + 0.5,
-                       (double)this.pos.getY() + 0.5,
-                       (double)this.pos.getZ() + 0.5
-                   )
-                   > 64.0
-               );
+        if (this.world.getTileEntity(this.pos) != this)
+        {
+            return false;
+        }
+        else
+        {
+            return !(
+                       player.getDistanceSq(
+                           (double)this.pos.getX() + 0.5,
+                           (double)this.pos.getY() + 0.5,
+                           (double)this.pos.getZ() + 0.5
+                       )
+                       > 64.0
+                   );
+        }
     }
 }
