--- before/net/minecraft/client/renderer/block/model/ItemModelGenerator.java
+++ after/net/minecraft/client/renderer/block/model/ItemModelGenerator.java
@@ -24,7 +24,7 @@
         Map<String, String> map = Maps.newHashMap();
         List<BlockPart> list = Lists.newArrayList();
 
-        for (int i = 0; i < field_178398_a.size(); i++)
+        for (int i = 0; i < field_178398_a.size(); ++i)
         {
             String s = field_178398_a.get(i);
 
@@ -173,13 +173,13 @@
         int j = p_178393_1_.func_94216_b();
         List<ItemModelGenerator.Span> list = Lists.newArrayList();
 
-        for (int k = 0; k < p_178393_1_.func_110970_k(); k++)
+        for (int k = 0; k < p_178393_1_.func_110970_k(); ++k)
         {
             int[] aint = p_178393_1_.func_147965_a(k)[0];
 
-            for (int l = 0; l < j; l++)
+            for (int l = 0; l < j; ++l)
             {
-                for (int i1 = 0; i1 < i; i1++)
+                for (int i1 = 0; i1 < i; ++i1)
                 {
                     boolean flag = !this.func_178391_a(aint, i1, l, i, j);
                     this.func_178396_a(ItemModelGenerator.SpanFacing.UP, list, aint, i1, l, i, j, flag);
@@ -248,9 +248,14 @@
 
     private boolean func_178391_a(int[] p_178391_1_, int p_178391_2_, int p_178391_3_, int p_178391_4_, int p_178391_5_)
     {
-        return p_178391_2_ >= 0 && p_178391_3_ >= 0 && p_178391_2_ < p_178391_4_ && p_178391_3_ < p_178391_5_
-               ? (p_178391_1_[p_178391_3_ * p_178391_4_ + p_178391_2_] >> 24 & 0xFF) == 0
-               : true;
+        if (p_178391_2_ >= 0 && p_178391_3_ >= 0 && p_178391_2_ < p_178391_4_ && p_178391_3_ < p_178391_5_)
+        {
+            return (p_178391_1_[p_178391_3_ * p_178391_4_ + p_178391_2_] >> 24 & 0xFF) == 0;
+        }
+        else
+        {
+            return true;
+        }
     }
 
     @SideOnly(Side.CLIENT)
