--- before/net/minecraft/client/gui/inventory/GuiShulkerBox.java
+++ after/net/minecraft/client/gui/inventory/GuiShulkerBox.java
@@ -21,7 +21,7 @@
         super(new ContainerShulkerBox(playerInventoryIn, inventoryIn, Minecraft.getMinecraft().player));
         this.playerInventory = playerInventoryIn;
         this.inventory = inventoryIn;
-        this.ySize++;
+        ++this.ySize;
     }
 
     @Override
