--- before/net/minecraft/village/MerchantRecipeList.java
+++ after/net/minecraft/village/MerchantRecipeList.java
@@ -38,7 +38,7 @@
         }
         else
         {
-            for (int i = 0; i < this.size(); i++)
+            for (int i = 0; i < this.size(); ++i)
             {
                 MerchantRecipe merchantrecipe = this.get(i);
 
@@ -69,7 +69,7 @@
     {
         p_151391_1_.writeByte((byte)(this.size() & 0xFF));
 
-        for (int i = 0; i < this.size(); i++)
+        for (int i = 0; i < this.size(); ++i)
         {
             MerchantRecipe merchantrecipe = this.get(i);
             p_151391_1_.func_150788_a(merchantrecipe.func_77394_a());
@@ -94,7 +94,7 @@
         MerchantRecipeList merchantrecipelist = new MerchantRecipeList();
         int i = p_151390_0_.readByte() & 255;
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             ItemStack itemstack = p_151390_0_.func_150791_c();
             ItemStack itemstack1 = p_151390_0_.func_150791_c();
@@ -125,7 +125,7 @@
     {
         NBTTagList nbttaglist = p_77201_1_.func_150295_c("Recipes", 10);
 
-        for (int i = 0; i < nbttaglist.func_74745_c(); i++)
+        for (int i = 0; i < nbttaglist.func_74745_c(); ++i)
         {
             NBTTagCompound nbttagcompound = nbttaglist.func_150305_b(i);
             this.add(new MerchantRecipe(nbttagcompound));
@@ -137,7 +137,7 @@
         NBTTagCompound nbttagcompound = new NBTTagCompound();
         NBTTagList nbttaglist = new NBTTagList();
 
-        for (int i = 0; i < this.size(); i++)
+        for (int i = 0; i < this.size(); ++i)
         {
             MerchantRecipe merchantrecipe = this.get(i);
             nbttaglist.func_74742_a(merchantrecipe.func_77395_g());
