--- before/net/minecraft/village/MerchantRecipeList.java
+++ after/net/minecraft/village/MerchantRecipeList.java
@@ -38,7 +38,7 @@
         }
         else
         {
-            for (int i = 0; i < this.size(); i++)
+            for (int i = 0; i < this.size(); ++i)
             {
                 MerchantRecipe merchantrecipe = this.get(i);
 
@@ -69,7 +69,7 @@
     {
         buffer.writeByte((byte)(this.size() & 0xFF));
 
-        for (int i = 0; i < this.size(); i++)
+        for (int i = 0; i < this.size(); ++i)
         {
             MerchantRecipe merchantrecipe = this.get(i);
             buffer.writeItemStack(merchantrecipe.getItemToBuy());
@@ -94,7 +94,7 @@
         MerchantRecipeList merchantrecipelist = new MerchantRecipeList();
         int i = buffer.readByte() & 255;
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             ItemStack itemstack = buffer.readItemStack();
             ItemStack itemstack1 = buffer.readItemStack();
@@ -125,7 +125,7 @@
     {
         NBTTagList nbttaglist = compound.getTagList("Recipes", 10);
 
-        for (int i = 0; i < nbttaglist.tagCount(); i++)
+        for (int i = 0; i < nbttaglist.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
             this.add(new MerchantRecipe(nbttagcompound));
@@ -137,7 +137,7 @@
         NBTTagCompound nbttagcompound = new NBTTagCompound();
         NBTTagList nbttaglist = new NBTTagList();
 
-        for (int i = 0; i < this.size(); i++)
+        for (int i = 0; i < this.size(); ++i)
         {
             MerchantRecipe merchantrecipe = this.get(i);
             nbttaglist.appendTag(merchantrecipe.writeToTags());
