--- before/net/minecraft/network/rcon/RConThreadBase.java
+++ after/net/minecraft/network/rcon/RConThreadBase.java
@@ -154,7 +154,7 @@
         {
             if (this.closeSocket(datagramsocket, false))
             {
-                i++;
+                ++i;
             }
         }
 
@@ -164,7 +164,7 @@
         {
             if (this.closeServerSocket_do(serversocket, false))
             {
-                i++;
+                ++i;
             }
         }
 
