--- before/net/minecraft/entity/monster/EntityEndermite.java
+++ after/net/minecraft/entity/monster/EntityEndermite.java
@@ -158,7 +158,7 @@
 
         if (this.world.isRemote)
         {
-            for (int i = 0; i < 2; i++)
+            for (int i = 0; i < 2; ++i)
             {
                 this.world
                 .spawnParticle(
@@ -176,7 +176,7 @@
         {
             if (!this.isNoDespawnRequired())
             {
-                this.lifetime++;
+                ++this.lifetime;
             }
 
             if (this.lifetime >= 2400)
