--- before/net/minecraft/entity/monster/EntitySkeleton.java
+++ after/net/minecraft/entity/monster/EntitySkeleton.java
@@ -62,7 +62,7 @@
     @Override
     public void onDeath(DamageSource cause)
     {
-        super.onDeath(cause);
+        // super.onDeath(cause); // CraftBukkit - moved down
 
         if (cause.getTrueSource() instanceof EntityCreeper)
         {
@@ -74,6 +74,7 @@
                 this.entityDropItem(new ItemStack(Items.SKULL, 1, 0), 0.0F);
             }
         }
+        super.onDeath(cause); // CraftBukkit - moved from above
     }
 
     @Override
