--- before/net/minecraft/entity/monster/EntityGuardian.java
+++ after/net/minecraft/entity/monster/EntityGuardian.java
@@ -250,15 +250,15 @@
                 }
                 else
                 {
-                    this.clientSideTailAnimationSpeed = this.clientSideTailAnimationSpeed + (0.5F - this.clientSideTailAnimationSpeed) * 0.1F;
+                    this.clientSideTailAnimationSpeed += (0.5F - this.clientSideTailAnimationSpeed) * 0.1F;
                 }
             }
             else
             {
-                this.clientSideTailAnimationSpeed = this.clientSideTailAnimationSpeed + (0.125F - this.clientSideTailAnimationSpeed) * 0.2F;
+                this.clientSideTailAnimationSpeed += (0.125F - this.clientSideTailAnimationSpeed) * 0.2F;
             }
 
-            this.clientSideTailAnimation = this.clientSideTailAnimation + this.clientSideTailAnimationSpeed;
+            this.clientSideTailAnimation += this.clientSideTailAnimationSpeed;
             this.clientSideSpikesAnimationO = this.clientSideSpikesAnimation;
 
             if (!this.isInWater())
@@ -267,18 +267,18 @@
             }
             else if (this.isMoving())
             {
-                this.clientSideSpikesAnimation = this.clientSideSpikesAnimation + (0.0F - this.clientSideSpikesAnimation) * 0.25F;
+                this.clientSideSpikesAnimation += (0.0F - this.clientSideSpikesAnimation) * 0.25F;
             }
             else
             {
-                this.clientSideSpikesAnimation = this.clientSideSpikesAnimation + (1.0F - this.clientSideSpikesAnimation) * 0.06F;
+                this.clientSideSpikesAnimation += (1.0F - this.clientSideSpikesAnimation) * 0.06F;
             }
 
             if (this.isMoving() && this.isInWater())
             {
                 Vec3d vec3d = this.getLook(0.0F);
 
-                for (int i = 0; i < 2; i++)
+                for (int i = 0; i < 2; ++i)
                 {
                     this.world
                     .spawnParticle(
@@ -297,7 +297,7 @@
             {
                 if (this.clientSideAttackTime < this.getAttackDuration())
                 {
-                    this.clientSideAttackTime++;
+                    ++this.clientSideAttackTime;
                 }
 
                 EntityLivingBase entitylivingbase = this.getTargetedEntity();
@@ -343,8 +343,8 @@
         else if (this.onGround)
         {
             this.motionY += 0.5;
-            this.motionX = this.motionX + (double)((this.rand.nextFloat() * 2.0F - 1.0F) * 0.4F);
-            this.motionZ = this.motionZ + (double)((this.rand.nextFloat() * 2.0F - 1.0F) * 0.4F);
+            this.motionX += (double)((this.rand.nextFloat() * 2.0F - 1.0F) * 0.4F);
+            this.motionZ += (double)((this.rand.nextFloat() * 2.0F - 1.0F) * 0.4F);
             this.rotationYaw = this.rand.nextFloat() * 360.0F;
             this.onGround = false;
             this.isAirBorne = true;
@@ -510,7 +510,7 @@
                 }
                 else
                 {
-                    this.tickCounter++;
+                    ++this.tickCounter;
 
                     if (this.tickCounter == 0)
                     {
@@ -576,7 +576,7 @@
                     this.entityGuardian.motionZ += d4 * d6;
                     d4 = Math.sin((double)(this.entityGuardian.ticksExisted + this.entityGuardian.getEntityId()) * 0.75) * 0.05;
                     this.entityGuardian.motionY += d4 * (d6 + d5) * 0.25;
-                    this.entityGuardian.motionY = this.entityGuardian.motionY + (double)this.entityGuardian.getAIMoveSpeed() * d1 * 0.1;
+                    this.entityGuardian.motionY += (double)this.entityGuardian.getAIMoveSpeed() * d1 * 0.1;
                     EntityLookHelper entitylookhelper = this.entityGuardian.getLookHelper();
                     double d7 = this.entityGuardian.posX + d0 / d3 * 2.0;
                     double d8 = (double)this.entityGuardian.getEyeHeight() + this.entityGuardian.posY + d1 / d3;
