--- before/net/minecraft/entity/monster/EntityVex.java
+++ after/net/minecraft/entity/monster/EntityVex.java
@@ -35,6 +35,7 @@
 import net.minecraft.world.storage.loot.LootTableList;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import org.bukkit.event.entity.EntityTargetEvent;
 
 public class EntityVex extends EntityMob
 {
@@ -267,9 +268,14 @@
         @Override
         public boolean shouldExecute()
         {
-            return EntityVex.this.getAttackTarget() != null && !EntityVex.this.getMoveHelper().isUpdating() && EntityVex.this.rand.nextInt(7) == 0
-                   ? EntityVex.this.getDistanceSq(EntityVex.this.getAttackTarget()) > 4.0
-                   : false;
+            if (EntityVex.this.getAttackTarget() != null && !EntityVex.this.getMoveHelper().isUpdating() && EntityVex.this.rand.nextInt(7) == 0)
+            {
+                return EntityVex.this.getDistanceSq(EntityVex.this.getAttackTarget()) > 4.0;
+            }
+            else
+            {
+                return false;
+            }
         }
 
         @Override
@@ -338,7 +344,8 @@
         @Override
         public void startExecuting()
         {
-            EntityVex.this.setAttackTarget(EntityVex.this.owner.getAttackTarget());
+            // EntityVex.this.setAttackTarget(EntityVex.this.owner.getAttackTarget());
+            EntityVex.this.setGoalTarget(EntityVex.this.owner.getAttackTarget(), EntityTargetEvent.TargetReason.OWNER_ATTACKED_TARGET, true);
             super.startExecuting();
         }
     }
@@ -370,9 +377,9 @@
                 }
                 else
                 {
-                    EntityVex.this.motionX = EntityVex.this.motionX + d0 / d3 * 0.05 * this.speed;
-                    EntityVex.this.motionY = EntityVex.this.motionY + d1 / d3 * 0.05 * this.speed;
-                    EntityVex.this.motionZ = EntityVex.this.motionZ + d2 / d3 * 0.05 * this.speed;
+                    EntityVex.this.motionX += d0 / d3 * 0.05 * this.speed;
+                    EntityVex.this.motionY += d1 / d3 * 0.05 * this.speed;
+                    EntityVex.this.motionZ += d2 / d3 * 0.05 * this.speed;
 
                     if (EntityVex.this.getAttackTarget() == null)
                     {
@@ -421,7 +428,7 @@
                 blockpos = new BlockPos(EntityVex.this);
             }
 
-            for (int i = 0; i < 3; i++)
+            for (int i = 0; i < 3; ++i)
             {
                 BlockPos blockpos1 = blockpos.add(
                                          EntityVex.this.rand.nextInt(15) - 7, EntityVex.this.rand.nextInt(11) - 5, EntityVex.this.rand.nextInt(15) - 7
