--- before/net/minecraft/entity/monster/EntityPigZombie.java
+++ after/net/minecraft/entity/monster/EntityPigZombie.java
@@ -31,7 +31,7 @@
 {
     private static final UUID ATTACK_SPEED_BOOST_MODIFIER_UUID = UUID.fromString("49455A49-7EC5-45BA-B886-3B90B23A1718");
     private static final AttributeModifier ATTACK_SPEED_BOOST_MODIFIER = new AttributeModifier(ATTACK_SPEED_BOOST_MODIFIER_UUID, "Attacking speed boost", 0.05, 0).setSaved(false);
-    private int angerLevel;
+    public int angerLevel;
     private int randomSoundDelay;
     private UUID angerTargetUUID;
 
@@ -80,7 +80,7 @@
                 iattributeinstance.applyModifier(ATTACK_SPEED_BOOST_MODIFIER);
             }
 
-            this.angerLevel--;
+            --this.angerLevel;
         }
         else if (iattributeinstance.hasModifier(ATTACK_SPEED_BOOST_MODIFIER))
         {
