--- before/net/minecraft/entity/monster/EntityVindicator.java
+++ after/net/minecraft/entity/monster/EntityVindicator.java
@@ -161,11 +161,13 @@
         {
             return true;
         }
+        else if (entityIn instanceof EntityLivingBase && ((EntityLivingBase)entityIn).getCreatureAttribute() == EnumCreatureAttribute.ILLAGER)
+        {
+            return this.getTeam() == null && entityIn.getTeam() == null;
+        }
         else
         {
-            return entityIn instanceof EntityLivingBase && ((EntityLivingBase)entityIn).getCreatureAttribute() == EnumCreatureAttribute.ILLAGER
-                   ? this.getTeam() == null && entityIn.getTeam() == null
-                   : false;
+            return false;
         }
     }
 
