--- before/net/minecraft/client/resources/ResourcePackListEntry.java
+++ after/net/minecraft/client/resources/ResourcePackListEntry.java
@@ -137,7 +137,7 @@
         this.mc.fontRenderer.drawStringWithShadow(s, (float)(x + 32 + 2), (float)(y + 1), 16777215);
         List<String> list = this.mc.fontRenderer.listFormattedStringToWidth(s1, 157);
 
-        for (int l = 0; l < 2 && l < list.size(); l++)
+        for (int l = 0; l < 2 && l < list.size(); ++l)
         {
             this.mc.fontRenderer.drawStringWithShadow(list.get(l), (float)(x + 32 + 2), (float)(y + 12 + 10 * l), 8421504);
         }
