--- before/net/minecraft/client/renderer/tileentity/TileEntityEndPortalRenderer.java
+++ after/net/minecraft/client/renderer/tileentity/TileEntityEndPortalRenderer.java
@@ -38,7 +38,7 @@
         float f = this.getOffset();
         boolean flag = false;
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             GlStateManager.pushMatrix();
             float f1 = 2.0F / (float)(18 - j);
