--- before/net/minecraft/client/renderer/tileentity/TileEntitySignRenderer.java
+++ after/net/minecraft/client/renderer/tileentity/TileEntitySignRenderer.java
@@ -89,7 +89,7 @@
 
         if (p_192841_9_ < 0)
         {
-            for (int j = 0; j < p_192841_1_.field_145915_a.length; j++)
+            for (int j = 0; j < p_192841_1_.field_145915_a.length; ++j)
             {
                 if (p_192841_1_.field_145915_a[j] != null)
                 {
