--- before/net/minecraft/client/renderer/tileentity/TileEntitySignRenderer.java
+++ after/net/minecraft/client/renderer/tileentity/TileEntitySignRenderer.java
@@ -89,7 +89,7 @@
 
         if (destroyStage < 0)
         {
-            for (int j = 0; j < te.signText.length; j++)
+            for (int j = 0; j < te.signText.length; ++j)
             {
                 if (te.signText[j] != null)
                 {
