--- before/net/minecraft/client/renderer/tileentity/TileEntityBeaconRenderer.java
+++ after/net/minecraft/client/renderer/tileentity/TileEntityBeaconRenderer.java
@@ -49,7 +49,7 @@
             GlStateManager.disableFog();
             int i = 0;
 
-            for (int j = 0; j < beamSegments.size(); j++)
+            for (int j = 0; j < beamSegments.size(); ++j)
             {
                 TileEntityBeacon.BeamSegment tileentitybeacon$beamsegment = beamSegments.get(j);
                 renderBeamSegment(
