--- before/net/minecraft/client/multiplayer/ServerList.java
+++ after/net/minecraft/client/multiplayer/ServerList.java
@@ -39,7 +39,7 @@
 
             NBTTagList nbttaglist = nbttagcompound.func_150295_c("servers", 10);
 
-            for (int i = 0; i < nbttaglist.func_74745_c(); i++)
+            for (int i = 0; i < nbttaglist.func_74745_c(); ++i)
             {
                 this.field_78858_b.add(ServerData.func_78837_a(nbttaglist.func_150305_b(i)));
             }
@@ -109,7 +109,7 @@
         ServerList serverlist = new ServerList(Minecraft.func_71410_x());
         serverlist.func_78853_a();
 
-        for (int i = 0; i < serverlist.func_78856_c(); i++)
+        for (int i = 0; i < serverlist.func_78856_c(); ++i)
         {
             ServerData serverdata = serverlist.func_78850_a(i);
 
