--- before/net/minecraft/client/multiplayer/ServerList.java
+++ after/net/minecraft/client/multiplayer/ServerList.java
@@ -39,7 +39,7 @@
 
             NBTTagList nbttaglist = nbttagcompound.getTagList("servers", 10);
 
-            for (int i = 0; i < nbttaglist.tagCount(); i++)
+            for (int i = 0; i < nbttaglist.tagCount(); ++i)
             {
                 this.servers.add(ServerData.getServerDataFromNBTCompound(nbttaglist.getCompoundTagAt(i)));
             }
@@ -109,7 +109,7 @@
         ServerList serverlist = new ServerList(Minecraft.getMinecraft());
         serverlist.loadServerList();
 
-        for (int i = 0; i < serverlist.countServers(); i++)
+        for (int i = 0; i < serverlist.countServers(); ++i)
         {
             ServerData serverdata = serverlist.getServerData(i);
 
