--- before/net/minecraft/util/datafix/fixes/AddBedTileEntity.java
+++ after/net/minecraft/util/datafix/fixes/AddBedTileEntity.java
@@ -29,13 +29,13 @@
             NBTTagList nbttaglist = nbttagcompound.getTagList("TileEntities", 10);
             NBTTagList nbttaglist1 = nbttagcompound.getTagList("Sections", 10);
 
-            for (int l = 0; l < nbttaglist1.tagCount(); l++)
+            for (int l = 0; l < nbttaglist1.tagCount(); ++l)
             {
                 NBTTagCompound nbttagcompound1 = nbttaglist1.getCompoundTagAt(l);
                 int i1 = nbttagcompound1.getByte("Y");
                 byte[] abyte = nbttagcompound1.getByteArray("Blocks");
 
-                for (int j1 = 0; j1 < abyte.length; j1++)
+                for (int j1 = 0; j1 < abyte.length; ++j1)
                 {
                     if (416 == (abyte[j1] & 255) << 4)
                     {
