--- before/net/minecraft/util/datafix/fixes/BookPagesStrictJSON.java
+++ after/net/minecraft/util/datafix/fixes/BookPagesStrictJSON.java
@@ -29,7 +29,7 @@
             {
                 NBTTagList nbttaglist = nbttagcompound.getTagList("pages", 8);
 
-                for (int i = 0; i < nbttaglist.tagCount(); i++)
+                for (int i = 0; i < nbttaglist.tagCount(); ++i)
                 {
                     String s = nbttaglist.getStringTagAt(i);
                     ITextComponent itextcomponent = null;
