--- before/net/minecraft/client/shader/Shader.java
+++ after/net/minecraft/client/shader/Shader.java
@@ -73,7 +73,7 @@
         GlStateManager.viewport(0, 0, (int)f, (int)f1);
         this.manager.addSamplerTexture("DiffuseSampler", this.framebufferIn);
 
-        for (int i = 0; i < this.listAuxFramebuffers.size(); i++)
+        for (int i = 0; i < this.listAuxFramebuffers.size(); ++i)
         {
             this.manager.addSamplerTexture(this.listAuxNames.get(i), this.listAuxFramebuffers.get(i));
             this.manager
