--- before/net/minecraft/client/entity/AbstractClientPlayer.java
+++ after/net/minecraft/client/entity/AbstractClientPlayer.java
@@ -1,6 +1,7 @@
 package net.minecraft.client.entity;
 
 import com.mojang.authlib.GameProfile;
+import java.io.File;
 import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.network.NetworkPlayerInfo;
@@ -33,14 +34,12 @@
         super(worldIn, playerProfile);
     }
 
-    @Override
     public boolean isSpectator()
     {
         NetworkPlayerInfo networkplayerinfo = Minecraft.getMinecraft().getConnection().getPlayerInfo(this.getGameProfile().getId());
         return networkplayerinfo != null && networkplayerinfo.getGameType() == GameType.SPECTATOR;
     }
 
-    @Override
     public boolean isCreative()
     {
         NetworkPlayerInfo networkplayerinfo = Minecraft.getMinecraft().getConnection().getPlayerInfo(this.getGameProfile().getId());
@@ -101,12 +100,7 @@
 
         if (itextureobject == null)
         {
-            itextureobject = new ThreadDownloadImageData(
-                null,
-                String.format("http://skins.minecraft.net/MinecraftSkins/%s.png", StringUtils.stripControlCodes(username)),
-                DefaultPlayerSkin.getDefaultSkin(getOfflineUUID(username)),
-                new ImageBufferDownload()
-            );
+            itextureobject = new ThreadDownloadImageData((File)null, String.format("http://skins.minecraft.net/MinecraftSkins/%s.png", StringUtils.stripControlCodes(username)), DefaultPlayerSkin.getDefaultSkin(getOfflineUUID(username)), new ImageBufferDownload());
             texturemanager.loadTexture(resourceLocationIn, itextureobject);
         }
 
@@ -134,14 +128,14 @@
         }
 
         IAttributeInstance iattributeinstance = this.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED);
-        f = (float)((double)f * ((iattributeinstance.getAttributeValue() / (double)this.capabilities.getWalkSpeed() + 1.0) / 2.0));
+        f = (float)((double)f * ((iattributeinstance.getAttributeValue() / (double)this.capabilities.getWalkSpeed() + 1.0D) / 2.0D));
 
         if (this.capabilities.getWalkSpeed() == 0.0F || Float.isNaN(f) || Float.isInfinite(f))
         {
             f = 1.0F;
         }
 
-        if (this.isHandActive() && this.getActiveItemStack().getItem() == Items.BOW)
+        if (this.isHandActive() && this.getActiveItemStack().getItem() instanceof net.minecraft.item.ItemBow)
         {
             int i = this.getItemInUseMaxCount();
             float f1 = (float)i / 20.0F;
@@ -152,12 +146,12 @@
             }
             else
             {
-                f1 *= f1;
+                f1 = f1 * f1;
             }
 
             f *= 1.0F - f1 * 0.15F;
         }
 
-        return f;
+        return net.minecraftforge.client.ForgeHooksClient.getOffsetFOV(this, f);
     }
 }
