--- before/net/minecraft/client/gui/GuiClickableScrolledSelectionListProxy.java
+++ after/net/minecraft/client/gui/GuiClickableScrolledSelectionListProxy.java
@@ -98,7 +98,7 @@
     {
         int i = this.getSize();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             int k = insideTop + j * this.slotHeight + this.headerPadding;
             int l = this.slotHeight - 4;
