--- before/net/minecraft/client/gui/GuiLabel.java
+++ after/net/minecraft/client/gui/GuiLabel.java
@@ -71,7 +71,7 @@
             int i = this.y + this.height / 2 + this.border / 2;
             int j = i - this.labels.size() * 10 / 2;
 
-            for (int k = 0; k < this.labels.size(); k++)
+            for (int k = 0; k < this.labels.size(); ++k)
             {
                 if (this.centered)
                 {
