--- before/net/minecraft/client/gui/GuiMerchant.java
+++ after/net/minecraft/client/gui/GuiMerchant.java
@@ -81,7 +81,7 @@
 
         if (button == this.nextButton)
         {
-            this.selectedMerchantRecipe++;
+            ++this.selectedMerchantRecipe;
             MerchantRecipeList merchantrecipelist = this.merchant.getRecipes(this.mc.player);
 
             if (merchantrecipelist != null && this.selectedMerchantRecipe >= merchantrecipelist.size())
@@ -93,7 +93,7 @@
         }
         else if (button == this.previousButton)
         {
-            this.selectedMerchantRecipe--;
+            --this.selectedMerchantRecipe;
 
             if (this.selectedMerchantRecipe < 0)
             {
