--- before/net/minecraft/client/gui/GuiNewChat.java
+++ after/net/minecraft/client/gui/GuiNewChat.java
@@ -55,7 +55,7 @@
                 GlStateManager.scale(f1, f1, 1.0F);
                 int l = 0;
 
-                for (int i1 = 0; i1 + this.scrollPos < this.drawnChatLines.size() && i1 < i; i1++)
+                for (int i1 = 0; i1 + this.scrollPos < this.drawnChatLines.size() && i1 < i; ++i1)
                 {
                     ChatLine chatline = this.drawnChatLines.get(i1 + this.scrollPos);
 
@@ -78,7 +78,7 @@
                             }
 
                             l1 = (int)((float)l1 * f);
-                            l++;
+                            ++l;
 
                             if (l1 > 3)
                             {
@@ -183,7 +183,7 @@
         this.drawnChatLines.clear();
         this.resetScroll();
 
-        for (int i = this.chatLines.size() - 1; i >= 0; i--)
+        for (int i = this.chatLines.size() - 1; i >= 0; --i)
         {
             ChatLine chatline = this.chatLines.get(i);
             this.setChatLine(chatline.getChatComponent(), chatline.getChatLineID(), chatline.getUpdatedCounter(), true);
