--- before/net/minecraft/client/gui/GuiPlayerTabOverlay.java
+++ after/net/minecraft/client/gui/GuiPlayerTabOverlay.java
@@ -85,7 +85,7 @@
 
         for (j4 = 1; i4 > 20; i4 = (l3 + j4 - 1) / j4)
         {
-            j4++;
+            ++j4;
         }
 
         boolean flag = this.mc.isIntegratedServerRunning() || this.mc.getConnection().getNetworkManager().isEncrypted();
@@ -152,12 +152,12 @@
                 k1 += this.mc.fontRenderer.FONT_HEIGHT;
             }
 
-            k1++;
+            ++k1;
         }
 
         drawRect(width / 2 - l1 / 2 - 1, k1 - 1, width / 2 + l1 / 2 + 1, k1 + i4 * 9, Integer.MIN_VALUE);
 
-        for (int k4 = 0; k4 < l3; k4++)
+        for (int k4 = 0; k4 < l3; ++k4)
         {
             int l4 = k4 / i4;
             int i5 = k4 % i4;
@@ -325,12 +325,12 @@
 
                 if (f > 3.0F)
                 {
-                    for (int l = j; l < k; l++)
+                    for (int l = j; l < k; ++l)
                     {
                         this.drawTexturedModalRect((float)p_175247_4_ + (float)l * f, (float)p_175247_2_, flag ? 25 : 16, 0, 9, 9);
                     }
 
-                    for (int j1 = 0; j1 < j; j1++)
+                    for (int j1 = 0; j1 < j; ++j1)
                     {
                         this.drawTexturedModalRect((float)p_175247_4_ + (float)j1 * f, (float)p_175247_2_, flag ? 25 : 16, 0, 9, 9);
 
