--- before/net/minecraft/client/gui/GuiPageButtonList.java
+++ after/net/minecraft/client/gui/GuiPageButtonList.java
@@ -317,7 +317,7 @@
                         }
                         else
                         {
-                            k--;
+                            --k;
                         }
                     }
                     else if (k == this.field_178072_w.size() - 1)
@@ -326,7 +326,7 @@
                     }
                     else
                     {
-                        k++;
+                        ++k;
                     }
 
                     this.field_178075_A = this.field_178072_w.get(k);
@@ -337,7 +337,7 @@
 
                     if (l > this.field_148154_c)
                     {
-                        this.field_148169_q = this.field_148169_q + (float)(l - this.field_148154_c);
+                        this.field_148169_q += (float)(l - this.field_148154_c);
                     }
                     else if (i1 < this.field_148153_b)
                     {
@@ -368,7 +368,7 @@
                     }
                     else
                     {
-                        j++;
+                        ++j;
                     }
 
                     if (j == i)
