--- before/net/minecraft/client/gui/GuiSnooper.java
+++ after/net/minecraft/client/gui/GuiSnooper.java
@@ -52,7 +52,7 @@
 
         for (Entry<String, String> entry : (new TreeMap<String, String>(this.mc.getPlayerUsageSnooper().getCurrentStats())).entrySet())
         {
-            this.keys.add((flag ? "C " : "") + entry.getKey());
+            this.keys.add((flag ? "C " : "") + (String)entry.getKey());
             this.values.add(this.fontRenderer.trimStringToWidth(entry.getValue(), this.width - 220));
         }
 
@@ -60,7 +60,7 @@
         {
             for (Entry<String, String> entry1 : (new TreeMap<String, String>(this.mc.getIntegratedServer().getPlayerUsageSnooper().getCurrentStats())).entrySet())
             {
-                this.keys.add("S " + entry1.getKey());
+                this.keys.add("S " + (String)entry1.getKey());
                 this.values.add(this.fontRenderer.trimStringToWidth(entry1.getValue(), this.width - 220));
             }
         }
