--- before/net/minecraft/client/gui/GuiWinGame.java
+++ after/net/minecraft/client/gui/GuiWinGame.java
@@ -103,14 +103,17 @@
 
                     while ((s1 = bufferedreader.readLine()) != null)
                     {
-                        s1 = s1.replaceAll("PLAYERNAME", this.mc.getSession().getUsername());
+                        String s2;
+                        String s3;
 
-                        while (s1.contains(s))
+                        for (s1 = s1.replaceAll("PLAYERNAME", this.mc.getSession().getUsername());
+                                s1.contains(s);
+                                s1 = s2 + TextFormatting.WHITE + TextFormatting.OBFUSCATED + "XXXXXXXX".substring(0, random.nextInt(4) + 3) + s3
+                            )
                         {
                             int j = s1.indexOf(s);
-                            String s2 = s1.substring(0, j);
-                            String s3 = s1.substring(j + s.length());
-                            s1 = s2 + TextFormatting.WHITE + TextFormatting.OBFUSCATED + "XXXXXXXX".substring(0, random.nextInt(4) + 3) + s3;
+                            s2 = s1.substring(0, j);
+                            s3 = s1.substring(j + s.length());
                         }
 
                         this.lines.addAll(this.mc.fontRenderer.listFormattedStringToWidth(s1, 274));
@@ -119,7 +122,7 @@
 
                     inputstream.close();
 
-                    for (int k = 0; k < 8; k++)
+                    for (int k = 0; k < 8; ++k)
                     {
                         this.lines.add("");
                     }
@@ -219,7 +222,7 @@
         GlStateManager.disableAlpha();
         int l = k + 100;
 
-        for (int i1 = 0; i1 < this.lines.size(); i1++)
+        for (int i1 = 0; i1 < this.lines.size(); ++i1)
         {
             if (i1 == this.lines.size() - 1)
             {
