--- before/net/minecraft/client/gui/ScaledResolution.java
+++ after/net/minecraft/client/gui/ScaledResolution.java
@@ -29,12 +29,12 @@
 
         while (this.scaleFactor < i && this.scaledWidth / (this.scaleFactor + 1) >= 320 && this.scaledHeight / (this.scaleFactor + 1) >= 240)
         {
-            this.scaleFactor++;
+            ++this.scaleFactor;
         }
 
         if (flag && this.scaleFactor % 2 != 0 && this.scaleFactor != 1)
         {
-            this.scaleFactor--;
+            --this.scaleFactor;
         }
 
         this.scaledWidthD = (double)this.scaledWidth / (double)this.scaleFactor;
