--- before/net/minecraft/client/gui/ScaledResolution.java
+++ after/net/minecraft/client/gui/ScaledResolution.java
@@ -29,12 +29,12 @@
 
         while (this.field_78330_e < i && this.field_78333_a / (this.field_78330_e + 1) >= 320 && this.field_78331_b / (this.field_78330_e + 1) >= 240)
         {
-            this.field_78330_e++;
+            ++this.field_78330_e;
         }
 
         if (flag && this.field_78330_e % 2 != 0 && this.field_78330_e != 1)
         {
-            this.field_78330_e--;
+            --this.field_78330_e;
         }
 
         this.field_78332_c = (double)this.field_78333_a / (double)this.field_78330_e;
