--- before/net/minecraft/client/gui/ServerSelectionList.java
+++ after/net/minecraft/client/gui/ServerSelectionList.java
@@ -65,7 +65,7 @@
     {
         this.serverListInternet.clear();
 
-        for (int i = 0; i < p_148195_1_.countServers(); i++)
+        for (int i = 0; i < p_148195_1_.countServers(); ++i)
         {
             this.serverListInternet.add(new ServerListEntryNormal(this.owner, p_148195_1_.getServerData(i)));
         }
