--- before/net/minecraft/client/particle/ParticleFallingDust.java
+++ after/net/minecraft/client/particle/ParticleFallingDust.java
@@ -73,7 +73,7 @@
         }
 
         this.prevParticleAngle = this.particleAngle;
-        this.particleAngle = this.particleAngle + (float) Math.PI * this.rotSpeed * 2.0F;
+        this.particleAngle += (float) Math.PI * this.rotSpeed * 2.0F;
 
         if (this.onGround)
         {
