--- before/net/minecraft/client/particle/ParticleFirework.java
+++ after/net/minecraft/client/particle/ParticleFirework.java
@@ -238,7 +238,7 @@
                     {
                         this.particleMaxAge = this.fireworkExplosions.tagCount() * 2 - 1;
 
-                        for (int i = 0; i < this.fireworkExplosions.tagCount(); i++)
+                        for (int i = 0; i < this.fireworkExplosions.tagCount(); ++i)
                         {
                             NBTTagCompound nbttagcompound = this.fireworkExplosions.getCompoundTagAt(i);
 
@@ -281,7 +281,7 @@
                     }
                     else
                     {
-                        for (int i = 0; i < this.fireworkExplosions.tagCount(); i++)
+                        for (int i = 0; i < this.fireworkExplosions.tagCount(); ++i)
                         {
                             NBTTagCompound nbttagcompound = this.fireworkExplosions.getCompoundTagAt(i);
 
@@ -402,7 +402,7 @@
                     this.manager.addEffect(particlefirework$overlay);
                 }
 
-                this.fireworkAge++;
+                ++this.fireworkAge;
 
                 if (this.fireworkAge > this.particleMaxAge)
                 {
@@ -471,11 +471,11 @@
                 double d1 = this.posY;
                 double d2 = this.posZ;
 
-                for (int i = -size; i <= size; i++)
+                for (int i = -size; i <= size; ++i)
                 {
-                    for (int j = -size; j <= size; j++)
+                    for (int j = -size; j <= size; ++j)
                     {
-                        for (int k = -size; k <= size; k++)
+                        for (int k = -size; k <= size; ++k)
                         {
                             double d3 = (double)j + (this.rand.nextDouble() - this.rand.nextDouble()) * 0.5;
                             double d4 = (double)i + (this.rand.nextDouble() - this.rand.nextDouble()) * 0.5;
@@ -513,13 +513,13 @@
                 float f = this.rand.nextFloat() * (float) Math.PI;
                 double d2 = p_92038_8_ ? 0.034 : 0.34;
 
-                for (int i = 0; i < 3; i++)
+                for (int i = 0; i < 3; ++i)
                 {
                     double d3 = (double)f + (double)((float)i * (float) Math.PI) * d2;
                     double d4 = d0;
                     double d5 = d1;
 
-                    for (int j = 1; j < shape.length; j++)
+                    for (int j = 1; j < shape.length; ++j)
                     {
                         double d6 = shape[j][0];
                         double d7 = shape[j][1];
@@ -550,7 +550,7 @@
                 double d0 = this.rand.nextGaussian() * 0.05;
                 double d1 = this.rand.nextGaussian() * 0.05;
 
-                for (int i = 0; i < 70; i++)
+                for (int i = 0; i < 70; ++i)
                 {
                     double d2 = this.motionX * 0.5 + this.rand.nextGaussian() * 0.15 + d0;
                     double d3 = this.motionZ * 0.5 + this.rand.nextGaussian() * 0.15 + d1;
