--- before/net/minecraft/client/particle/ParticleCloud.java
+++ after/net/minecraft/client/particle/ParticleCloud.java
@@ -78,8 +78,8 @@
 
             if (this.posY > axisalignedbb.minY)
             {
-                this.posY = this.posY + (axisalignedbb.minY - this.posY) * 0.2;
-                this.motionY = this.motionY + (entityplayer.motionY - this.motionY) * 0.2;
+                this.posY += (axisalignedbb.minY - this.posY) * 0.2;
+                this.motionY += (entityplayer.motionY - this.motionY) * 0.2;
                 this.setPosition(this.posX, this.posY, this.posZ);
             }
         }
