--- before/net/minecraft/client/particle/ParticleSuspend.java
+++ after/net/minecraft/client/particle/ParticleSuspend.java
@@ -19,7 +19,7 @@
         this.particleBlue = 0.7F;
         this.setParticleTextureIndex(0);
         this.setSize(0.01F, 0.01F);
-        this.particleScale = this.particleScale * (this.rand.nextFloat() * 0.6F + 0.2F);
+        this.particleScale *= this.rand.nextFloat() * 0.6F + 0.2F;
         this.motionX = xSpeedIn * 0.0;
         this.motionY = ySpeedIn * 0.0;
         this.motionZ = zSpeedIn * 0.0;
