--- before/net/minecraft/client/particle/ParticleWaterWake.java
+++ after/net/minecraft/client/particle/ParticleWaterWake.java
@@ -33,7 +33,7 @@
         this.prevPosX = this.posX;
         this.prevPosY = this.posY;
         this.prevPosZ = this.posZ;
-        this.motionY = this.motionY - (double)this.particleGravity;
+        this.motionY -= (double)this.particleGravity;
         this.move(this.motionX, this.motionY, this.motionZ);
         this.motionX *= 0.98F;
         this.motionY *= 0.98F;
