--- before/net/minecraft/client/particle/ParticleLava.java
+++ after/net/minecraft/client/particle/ParticleLava.java
@@ -22,7 +22,7 @@
         this.particleRed = 1.0F;
         this.particleGreen = 1.0F;
         this.particleBlue = 1.0F;
-        this.particleScale = this.particleScale * (this.rand.nextFloat() * 2.0F + 0.2F);
+        this.particleScale *= this.rand.nextFloat() * 2.0F + 0.2F;
         this.lavaParticleScale = this.particleScale;
         this.particleMaxAge = (int)(16.0 / (Math.random() * 0.8 + 0.2));
         this.setParticleTextureIndex(49);
