--- before/net/minecraft/client/particle/ParticleBubble.java
+++ after/net/minecraft/client/particle/ParticleBubble.java
@@ -19,7 +19,7 @@
         this.particleBlue = 1.0F;
         this.setParticleTextureIndex(32);
         this.setSize(0.02F, 0.02F);
-        this.particleScale = this.particleScale * (this.rand.nextFloat() * 0.6F + 0.2F);
+        this.particleScale *= this.rand.nextFloat() * 0.6F + 0.2F;
         this.motionX = xSpeedIn * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
         this.motionY = ySpeedIn * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
         this.motionZ = zSpeedIn * 0.2F + (Math.random() * 2.0 - 1.0) * 0.02F;
