--- before/net/minecraft/client/particle/ParticleFlame.java
+++ after/net/minecraft/client/particle/ParticleFlame.java
@@ -18,9 +18,9 @@
         this.motionX = this.motionX * 0.01F + xSpeedIn;
         this.motionY = this.motionY * 0.01F + ySpeedIn;
         this.motionZ = this.motionZ * 0.01F + zSpeedIn;
-        this.posX = this.posX + (double)((this.rand.nextFloat() - this.rand.nextFloat()) * 0.05F);
-        this.posY = this.posY + (double)((this.rand.nextFloat() - this.rand.nextFloat()) * 0.05F);
-        this.posZ = this.posZ + (double)((this.rand.nextFloat() - this.rand.nextFloat()) * 0.05F);
+        this.posX += (double)((this.rand.nextFloat() - this.rand.nextFloat()) * 0.05F);
+        this.posY += (double)((this.rand.nextFloat() - this.rand.nextFloat()) * 0.05F);
+        this.posZ += (double)((this.rand.nextFloat() - this.rand.nextFloat()) * 0.05F);
         this.flameScale = this.particleScale;
         this.particleRed = 1.0F;
         this.particleGreen = 1.0F;
