--- before/net/minecraft/client/particle/ParticleSimpleAnimated.java
+++ after/net/minecraft/client/particle/ParticleSimpleAnimated.java
@@ -67,18 +67,18 @@
 
             if (this.fadingColor)
             {
-                this.particleRed = this.particleRed + (this.fadeTargetRed - this.particleRed) * 0.2F;
-                this.particleGreen = this.particleGreen + (this.fadeTargetGreen - this.particleGreen) * 0.2F;
-                this.particleBlue = this.particleBlue + (this.fadeTargetBlue - this.particleBlue) * 0.2F;
+                this.particleRed += (this.fadeTargetRed - this.particleRed) * 0.2F;
+                this.particleGreen += (this.fadeTargetGreen - this.particleGreen) * 0.2F;
+                this.particleBlue += (this.fadeTargetBlue - this.particleBlue) * 0.2F;
             }
         }
 
         this.setParticleTextureIndex(this.textureIdx + (this.numAgingFrames - 1 - this.particleAge * this.numAgingFrames / this.particleMaxAge));
-        this.motionY = this.motionY + (double)this.yAccel;
+        this.motionY += (double)this.yAccel;
         this.move(this.motionX, this.motionY, this.motionZ);
-        this.motionX = this.motionX * (double)this.baseAirFriction;
-        this.motionY = this.motionY * (double)this.baseAirFriction;
-        this.motionZ = this.motionZ * (double)this.baseAirFriction;
+        this.motionX *= (double)this.baseAirFriction;
+        this.motionY *= (double)this.baseAirFriction;
+        this.motionZ *= (double)this.baseAirFriction;
 
         if (this.onGround)
         {
