--- before/net/minecraft/client/particle/ParticleLava.java
+++ after/net/minecraft/client/particle/ParticleLava.java
@@ -22,7 +22,7 @@
         this.field_70552_h = 1.0F;
         this.field_70553_i = 1.0F;
         this.field_70551_j = 1.0F;
-        this.field_70544_f = this.field_70544_f * (this.field_187136_p.nextFloat() * 2.0F + 0.2F);
+        this.field_70544_f *= this.field_187136_p.nextFloat() * 2.0F + 0.2F;
         this.field_70586_a = this.field_70544_f;
         this.field_70547_e = (int)(16.0 / (Math.random() * 0.8 + 0.2));
         this.func_70536_a(49);
