--- before/net/minecraft/world/biome/BiomeVoidDecorator.java
+++ after/net/minecraft/world/biome/BiomeVoidDecorator.java
@@ -20,9 +20,9 @@
             BlockPos blockpos2 = new BlockPos(blockpos.getX() + 16, blockpos.getY() - 1, blockpos.getZ() + 16);
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(blockpos1);
 
-            for (int j = pos.getZ(); j < pos.getZ() + 16; j++)
+            for (int j = pos.getZ(); j < pos.getZ() + 16; ++j)
             {
-                for (int k = pos.getX(); k < pos.getX() + 16; k++)
+                for (int k = pos.getX(); k < pos.getX() + 16; ++k)
                 {
                     if (j >= blockpos1.getZ() && j <= blockpos2.getZ() && k >= blockpos1.getX() && k <= blockpos2.getX())
                     {
