--- before/net/minecraft/world/biome/BiomeCache.java
+++ after/net/minecraft/world/biome/BiomeCache.java
@@ -51,7 +51,7 @@
         {
             this.lastCleanupTime = i;
 
-            for (int k = 0; k < this.cache.size(); k++)
+            for (int k = 0; k < this.cache.size(); ++k)
             {
                 BiomeCache.Block biomecache$block = this.cache.get(k);
                 long l = i - biomecache$block.lastAccessTime;
