--- before/net/minecraft/client/settings/HotbarSnapshot.java
+++ after/net/minecraft/client/settings/HotbarSnapshot.java
@@ -17,7 +17,7 @@
     {
         this.ensureCapacity(field_192835_a);
 
-        for (int i = 0; i < field_192835_a; i++)
+        for (int i = 0; i < field_192835_a; ++i)
         {
             this.add(ItemStack.field_190927_a);
         }
@@ -27,7 +27,7 @@
     {
         NBTTagList nbttaglist = new NBTTagList();
 
-        for (int i = 0; i < field_192835_a; i++)
+        for (int i = 0; i < field_192835_a; ++i)
         {
             nbttaglist.func_74742_a(this.get(i).func_77955_b(new NBTTagCompound()));
         }
@@ -37,7 +37,7 @@
 
     public void func_192833_a(NBTTagList p_192833_1_)
     {
-        for (int i = 0; i < field_192835_a; i++)
+        for (int i = 0; i < field_192835_a; ++i)
         {
             this.set(i, new ItemStack(p_192833_1_.func_150305_b(i)));
         }
@@ -46,7 +46,7 @@
     @Override
     public boolean isEmpty()
     {
-        for (int i = 0; i < field_192835_a; i++)
+        for (int i = 0; i < field_192835_a; ++i)
         {
             if (!this.get(i).func_190926_b())
             {
