--- before/net/minecraft/client/settings/HotbarSnapshot.java
+++ after/net/minecraft/client/settings/HotbarSnapshot.java
@@ -17,7 +17,7 @@
     {
         this.ensureCapacity(HOTBAR_SIZE);
 
-        for (int i = 0; i < HOTBAR_SIZE; i++)
+        for (int i = 0; i < HOTBAR_SIZE; ++i)
         {
             this.add(ItemStack.EMPTY);
         }
@@ -27,7 +27,7 @@
     {
         NBTTagList nbttaglist = new NBTTagList();
 
-        for (int i = 0; i < HOTBAR_SIZE; i++)
+        for (int i = 0; i < HOTBAR_SIZE; ++i)
         {
             nbttaglist.appendTag(this.get(i).writeToNBT(new NBTTagCompound()));
         }
@@ -37,7 +37,7 @@
 
     public void fromTag(NBTTagList tag)
     {
-        for (int i = 0; i < HOTBAR_SIZE; i++)
+        for (int i = 0; i < HOTBAR_SIZE; ++i)
         {
             this.set(i, new ItemStack(tag.getCompoundTagAt(i)));
         }
@@ -46,7 +46,7 @@
     @Override
     public boolean isEmpty()
     {
-        for (int i = 0; i < HOTBAR_SIZE; i++)
+        for (int i = 0; i < HOTBAR_SIZE; ++i)
         {
             if (!this.get(i).isEmpty())
             {
