--- before/net/minecraft/advancements/AdvancementManager.java
+++ after/net/minecraft/advancements/AdvancementManager.java
@@ -9,6 +9,7 @@
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParseException;
 import java.io.BufferedReader;
+import java.io.Closeable;
 import java.io.File;
 import java.io.IOException;
 import java.io.Reader;
@@ -41,25 +42,18 @@
 public class AdvancementManager
 {
     private static final Logger LOGGER = LogManager.getLogger();
-    public static final Gson GSON = new GsonBuilder()
-        .registerTypeHierarchyAdapter(
-            Advancement.Builder.class,
-            new JsonDeserializer<Advancement.Builder>()
-    {
-        public Advancement.Builder deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_) throws JsonParseException
-        {
+    public static final Gson GSON = new GsonBuilder().registerTypeHierarchyAdapter(Advancement.Builder.class, new JsonDeserializer<Advancement.Builder>() {
+        public Advancement.Builder deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_) throws JsonParseException {
             JsonObject jsonobject = JsonUtils.getJsonObject(p_deserialize_1_, "advancement");
             return Advancement.Builder.deserialize(jsonobject, p_deserialize_3_);
         }
-    }
-    )
-    .registerTypeAdapter(AdvancementRewards.class, new AdvancementRewards.Deserializer())
-    .registerTypeHierarchyAdapter(ITextComponent.class, new ITextComponent.Serializer())
-    .registerTypeHierarchyAdapter(Style.class, new Style.Serializer())
-    .registerTypeAdapterFactory(new EnumTypeAdapterFactory())
-    .create();
-    private static final AdvancementList ADVANCEMENT_LIST = new AdvancementList();
-    private final File advancementsDir;
+    }).registerTypeAdapter(AdvancementRewards.class, new AdvancementRewards.Deserializer())
+        .registerTypeHierarchyAdapter(ITextComponent.class, new ITextComponent.Serializer())
+        .registerTypeHierarchyAdapter(Style.class, new Style.Serializer())
+        .registerTypeAdapterFactory(new EnumTypeAdapterFactory())
+        .create();
+    public static final AdvancementList ADVANCEMENT_LIST = new AdvancementList();
+    public final File advancementsDir;
     private boolean hasErrored;
 
     public AdvancementManager(@Nullable File advancementsDirIn)
@@ -74,6 +68,7 @@
         ADVANCEMENT_LIST.clear();
         Map<ResourceLocation, Advancement.Builder> map = this.loadCustomAdvancements();
         this.loadBuiltInAdvancements(map);
+        this.hasErrored |= net.minecraftforge.common.ForgeHooks.loadAdvancements(map);
         ADVANCEMENT_LIST.loadAdvancements(map);
 
         for (Advancement advancement : ADVANCEMENT_LIST.getRoots())
@@ -187,6 +182,11 @@
                         Path path2 = path.relativize(path1);
                         String s = FilenameUtils.removeExtension(path2.toString()).replaceAll("\\\\", "/");
                         ResourceLocation resourcelocation = new ResourceLocation("minecraft", s);
+                        // Spigot start
+                        if (org.spigotmc.SpigotConfig.disabledAdvancements != null && (org.spigotmc.SpigotConfig.disabledAdvancements.contains("*") || org.spigotmc.SpigotConfig.disabledAdvancements.contains(resourcelocation.toString()))) {
+                            continue;
+                        }
+                        // Spigot end
 
                         if (!map.containsKey(resourcelocation))
                         {
@@ -215,12 +215,18 @@
                         }
                     }
                 }
+
+                return;
             }
+
+            LOGGER.error("Couldn't find .mcassetsroot");
+            this.hasErrored = true;
         }
         catch (IOException | URISyntaxException urisyntaxexception)
         {
             LOGGER.error("Couldn't get a list of all built-in advancement files", (Throwable)urisyntaxexception);
             this.hasErrored = true;
+            return;
         }
         finally
         {
