--- before/net/minecraft/advancements/Advancement.java
+++ after/net/minecraft/advancements/Advancement.java
@@ -32,6 +32,7 @@
     private final String[][] requirements;
     private final Set<Advancement> children = Sets.newLinkedHashSet();
     private final ITextComponent displayText;
+    public final org.bukkit.advancement.Advancement bukkit = new org.bukkit.craftbukkit.v1_12_R1.advancement.CraftAdvancement(this);
 
     public Advancement(
         ResourceLocation id,
@@ -293,12 +294,12 @@
                     JsonArray jsonarray = JsonUtils.getJsonArray(json, "requirements", new JsonArray());
                     String[][] astring = new String[jsonarray.size()][];
 
-                    for (int i = 0; i < jsonarray.size(); i++)
+                    for (int i = 0; i < jsonarray.size(); ++i)
                     {
                         JsonArray jsonarray1 = JsonUtils.getJsonArray(jsonarray.get(i), "requirements[" + i + "]");
                         astring[i] = new String[jsonarray1.size()];
 
-                        for (int j = 0; j < jsonarray1.size(); j++)
+                        for (int j = 0; j < jsonarray1.size(); ++j)
                         {
                             astring[i][j] = JsonUtils.getString(jsonarray1.get(j), "requirements[" + i + "][" + j + "]");
                         }
@@ -363,11 +364,11 @@
                 Map<String, Criterion> map = Criterion.criteriaFromNetwork(buf);
                 String[][] astring = new String[buf.readVarInt()][];
 
-                for (int i = 0; i < astring.length; i++)
+                for (int i = 0; i < astring.length; ++i)
                 {
                     astring[i] = new String[buf.readVarInt()];
 
-                    for (int j = 0; j < astring[i].length; j++)
+                    for (int j = 0; j < astring[i].length; ++j)
                     {
                         astring[i][j] = buf.readString(32767);
                     }
