--- before/net/minecraft/server/management/DemoPlayerInteractionManager.java
+++ after/net/minecraft/server/management/DemoPlayerInteractionManager.java
@@ -26,7 +26,7 @@
     public void updateBlockRemoving()
     {
         super.updateBlockRemoving();
-        this.gameModeTicks++;
+        ++this.gameModeTicks;
         long i = this.world.getTotalWorldTime();
         long j = i / 24000L + 1L;
 
@@ -40,7 +40,7 @@
 
         if (this.demoTimeExpired)
         {
-            this.demoEndedReminder++;
+            ++this.demoEndedReminder;
         }
 
         if (i % 24000L == 500L)
