--- before/net/minecraft/server/management/PlayerProfileCache.java
+++ after/net/minecraft/server/management/PlayerProfileCache.java
@@ -47,7 +47,7 @@
     private static boolean onlineMode;
     private final Map<String, PlayerProfileCache.ProfileEntry> usernameToProfileEntryMap = Maps.newHashMap();
     private final Map<UUID, PlayerProfileCache.ProfileEntry> uuidToProfileEntryMap = Maps.newHashMap();
-    private final Deque<GameProfile> gameProfiles = Lists.newLinkedList();
+    private final Deque<GameProfile> gameProfiles = new java.util.concurrent.LinkedBlockingDeque<>();
     private final GameProfileRepository profileRepo;
     protected final Gson gson;
     private final File usercacheFile;
@@ -96,6 +96,7 @@
                 agameprofile[0] = null;
             }
         };
+        if (!catserver.server.CatServer.getConfig().disableUpdateGameProfile || isOnlineMode()) // CatServer
         profileRepoIn.findProfilesByNames(new String[] {name}, Agent.MINECRAFT, profilelookupcallback);
 
         if (!isOnlineMode() && agameprofile[0] == null)
@@ -123,7 +124,7 @@
         this.addEntry(gameProfile, null);
     }
 
-    private void addEntry(GameProfile gameProfile, Date expirationDate)
+    private synchronized void addEntry(GameProfile gameProfile, Date expirationDate) // Paper - synchronize
     {
         UUID uuid = gameProfile.getId();
 
@@ -138,9 +139,9 @@
         String s = gameProfile.getName().toLowerCase(Locale.ROOT);
         PlayerProfileCache.ProfileEntry playerprofilecache$profileentry = new PlayerProfileCache.ProfileEntry(gameProfile, expirationDate);
 
-        if (this.uuidToProfileEntryMap.containsKey(uuid))
-        {
-            PlayerProfileCache.ProfileEntry playerprofilecache$profileentry1 = this.uuidToProfileEntryMap.get(uuid);
+        // if (this.uuidToProfileEntryMap.containsKey(uuid)) { // Paper
+        ProfileEntry playerprofilecache$profileentry1 = this.uuidToProfileEntryMap.get(uuid);
+        if (playerprofilecache$profileentry1 != null) { // Paper
             this.usernameToProfileEntryMap.remove(playerprofilecache$profileentry1.getGameProfile().getName().toLowerCase(Locale.ROOT));
             this.gameProfiles.remove(gameProfile);
         }
@@ -148,11 +149,11 @@
         this.usernameToProfileEntryMap.put(gameProfile.getName().toLowerCase(Locale.ROOT), playerprofilecache$profileentry);
         this.uuidToProfileEntryMap.put(uuid, playerprofilecache$profileentry);
         this.gameProfiles.addFirst(gameProfile);
-        this.save();
+        if(!org.spigotmc.SpigotConfig.saveUserCacheOnStopOnly) this.save(); // Spigot - skip saving if disabled
     }
 
     @Nullable
-    public GameProfile getGameProfileForUsername(String username)
+    public synchronized GameProfile getGameProfileForUsername(String username) // Paper - synchronize
     {
         String s = username.toLowerCase(Locale.ROOT);
         PlayerProfileCache.ProfileEntry playerprofilecache$profileentry = this.usernameToProfileEntryMap.get(s);
@@ -182,18 +183,18 @@
             }
         }
 
-        this.save();
+        if(!org.spigotmc.SpigotConfig.saveUserCacheOnStopOnly) this.save(); // Spigot - skip saving if disabled
         return playerprofilecache$profileentry == null ? null : playerprofilecache$profileentry.getGameProfile();
     }
 
-    public String[] getUsernames()
+    public synchronized String[] getUsernames() // Paper - synchronize
     {
         List<String> list = Lists.newArrayList(this.usernameToProfileEntryMap.keySet());
         return list.toArray(new String[list.size()]);
     }
 
     @Nullable
-    public GameProfile getProfileByUUID(UUID uuid)
+    public synchronized GameProfile getProfileByUUID(UUID uuid) // Paper - synchronize
     {
         PlayerProfileCache.ProfileEntry playerprofilecache$profileentry = this.uuidToProfileEntryMap.get(uuid);
         return playerprofilecache$profileentry == null ? null : playerprofilecache$profileentry.getGameProfile();
@@ -248,9 +249,16 @@
         }
     }
 
+    // Paper start
     public void save()
     {
-        String s = this.gson.toJson(this.getEntriesWithLimit(1000));
+        save(true);
+    }
+    public void save(boolean asyncSave)
+    {
+        // Paper end
+        String s = this.gson.toJson(this.getEntriesWithLimit(org.spigotmc.SpigotConfig.userCacheCap)); // Spigot
+        Runnable save = () -> {
         BufferedWriter bufferedwriter = null;
 
         try
@@ -270,6 +278,15 @@
         {
             IOUtils.closeQuietly((Writer)bufferedwriter);
         }
+        // Paper start
+        };
+        if (asyncSave) {
+            catserver.server.CatServer.scheduleAsyncTask(save);
+        } else {
+            save.run();
+        }
+        // Paper end
+
     }
 
     private List<PlayerProfileCache.ProfileEntry> getEntriesWithLimit(int limitSize)
