--- before/net/minecraft/server/management/PreYggdrasilConverter.java
+++ after/net/minecraft/server/management/PreYggdrasilConverter.java
@@ -22,6 +22,8 @@
 import java.util.UUID;
 import javax.annotation.Nullable;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.dedicated.DedicatedServer;
 import net.minecraft.server.dedicated.PropertyManager;
@@ -39,25 +41,6 @@
     public static final File OLD_OPS_FILE = new File("ops.txt");
     public static final File OLD_WHITELIST_FILE = new File("white-list.txt");
 
-    @SideOnly(Side.SERVER)
-    static List<String> readFile(File inFile, Map<String, String[]> read) throws IOException
-    {
-        List<String> list = Files.readLines(inFile, StandardCharsets.UTF_8);
-
-        for (String s : list)
-        {
-            s = s.trim();
-
-            if (!s.startsWith("#") && s.length() >= 1)
-            {
-                String[] astring = s.split("\\|");
-                read.put(astring[0].toLowerCase(Locale.ROOT), astring);
-            }
-        }
-
-        return list;
-    }
-
     private static void lookupNames(MinecraftServer server, Collection<String> names, ProfileLookupCallback callback)
     {
         String[] astring = Iterators.toArray(Iterators.filter(names.iterator(), new Predicate<String>()
@@ -68,7 +51,7 @@
             }
         }), String.class);
 
-        if (server.isServerInOnlineMode())
+        if (server.isServerInOnlineMode() || org.spigotmc.SpigotConfig.bungee) // Spigot: bungee = online mode, for now.
         {
             server.getGameProfileRepository().findProfilesByNames(astring, Agent.MINECRAFT, callback);
         }
@@ -83,6 +66,64 @@
         }
     }
 
+    public static String convertMobOwnerIfNeeded(final MinecraftServer server, String username)
+    {
+        if (!StringUtils.isNullOrEmpty(username) && username.length() <= 16)
+        {
+            GameProfile gameprofile = server.getPlayerProfileCache().getGameProfileForUsername(username);
+
+            if (gameprofile != null && gameprofile.getId() != null)
+            {
+                return gameprofile.getId().toString();
+            }
+            else if (!server.isSinglePlayer() && server.isServerInOnlineMode())
+            {
+                final List<GameProfile> list = Lists.<GameProfile>newArrayList();
+                ProfileLookupCallback profilelookupcallback = new ProfileLookupCallback()
+                {
+                    public void onProfileLookupSucceeded(GameProfile p_onProfileLookupSucceeded_1_)
+                    {
+                        server.getPlayerProfileCache().addEntry(p_onProfileLookupSucceeded_1_);
+                        list.add(p_onProfileLookupSucceeded_1_);
+                    }
+                    public void onProfileLookupFailed(GameProfile p_onProfileLookupFailed_1_, Exception p_onProfileLookupFailed_2_)
+                    {
+                        PreYggdrasilConverter.LOGGER.warn("Could not lookup user whitelist entry for {}", p_onProfileLookupFailed_1_.getName(), p_onProfileLookupFailed_2_);
+                    }
+                };
+                lookupNames(server, Lists.newArrayList(username), profilelookupcallback);
+                return !list.isEmpty() && ((GameProfile)list.get(0)).getId() != null ? ((GameProfile)list.get(0)).getId().toString() : "";
+            }
+            else
+            {
+                return EntityPlayer.getUUID(new GameProfile((UUID)null, username)).toString();
+            }
+        }
+        else
+        {
+            return username;
+        }
+    }
+
+    @SideOnly(Side.SERVER)
+    static List<String> readFile(File inFile, Map<String, String[]> read) throws IOException
+    {
+        List<String> list = Files.readLines(inFile, StandardCharsets.UTF_8);
+
+        for (String s : list)
+        {
+            s = s.trim();
+
+            if (!s.startsWith("#") && s.length() >= 1)
+            {
+                String[] astring = s.split("\\|");
+                read.put(astring[0].toLowerCase(Locale.ROOT), astring);
+            }
+        }
+
+        return list;
+    }
+
     @SideOnly(Side.SERVER)
     public static boolean convertUserBanlist(final MinecraftServer server) throws IOException
     {
@@ -96,9 +137,10 @@
                 {
                     userlistbans.readSavedFile();
                 }
-                catch (FileNotFoundException filenotfoundexception)
+                // CraftBukkit start - FileNotFoundException -> IOException, don't print stacktrace
+                catch (IOException filenotfoundexception)
                 {
-                    LOGGER.warn("Could not load existing file {}", userlistbans.getSaveFile().getName(), filenotfoundexception);
+                    LOGGER.warn("Could not load existing file {}", userlistbans.getSaveFile().getName());
                 }
             }
 
@@ -112,7 +154,7 @@
                     public void onProfileLookupSucceeded(GameProfile p_onProfileLookupSucceeded_1_)
                     {
                         server.getPlayerProfileCache().addEntry(p_onProfileLookupSucceeded_1_);
-                        String[] astring = map.get(p_onProfileLookupSucceeded_1_.getName().toLowerCase(Locale.ROOT));
+                        String[] astring = (String[])map.get(p_onProfileLookupSucceeded_1_.getName().toLowerCase(Locale.ROOT));
 
                         if (astring == null)
                         {
@@ -177,9 +219,10 @@
                 {
                     userlistipbans.readSavedFile();
                 }
-                catch (FileNotFoundException filenotfoundexception)
+                // CraftBukkit start - FileNotFoundException -> IOException, don't print stacktrace
+                catch (IOException filenotfoundexception)
                 {
-                    LOGGER.warn("Could not load existing file {}", userlistipbans.getSaveFile().getName(), filenotfoundexception);
+                    LOGGER.warn("Could not load existing file {}", userlistipbans.getSaveFile().getName());
                 }
             }
 
@@ -190,7 +233,7 @@
 
                 for (String s : map.keySet())
                 {
-                    String[] astring = map.get(s);
+                    String[] astring = (String[])map.get(s);
                     Date date = astring.length > 1 ? parseDate(astring[1], null) : null;
                     String s1 = astring.length > 2 ? astring[2] : null;
                     Date date1 = astring.length > 3 ? parseDate(astring[3], null) : null;
@@ -227,9 +270,10 @@
                 {
                     userlistops.readSavedFile();
                 }
-                catch (FileNotFoundException filenotfoundexception)
+                // CraftBukkit start - FileNotFoundException -> IOException, don't print stacktrace
+                catch (IOException filenotfoundexception)
                 {
-                    LOGGER.warn("Could not load existing file {}", userlistops.getSaveFile().getName(), filenotfoundexception);
+                    LOGGER.warn("Could not load existing file {}", userlistops.getSaveFile().getName());
                 }
             }
 
@@ -293,9 +337,10 @@
                 {
                     userlistwhitelist.readSavedFile();
                 }
-                catch (FileNotFoundException filenotfoundexception)
+                // CraftBukkit start - FileNotFoundException -> IOException, don't print stacktrace
+                catch (IOException filenotfoundexception)
                 {
-                    LOGGER.warn("Could not load existing file {}", userlistwhitelist.getSaveFile().getName(), filenotfoundexception);
+                    LOGGER.warn("Could not load existing file {}", userlistwhitelist.getSaveFile().getName());
                 }
             }
 
@@ -318,9 +363,7 @@
 
                         if (!(p_onProfileLookupFailed_2_ instanceof ProfileNotFoundException))
                         {
-                            throw new PreYggdrasilConverter.ConversionError(
-                                "Could not request user " + p_onProfileLookupFailed_1_.getName() + " from backend systems", p_onProfileLookupFailed_2_
-                            );
+                            throw new ConversionError("Could not request user " + p_onProfileLookupFailed_1_.getName() + " from backend systems", p_onProfileLookupFailed_2_);
                         }
                     }
                 };
@@ -346,48 +389,6 @@
         }
     }
 
-    public static String convertMobOwnerIfNeeded(final MinecraftServer server, String username)
-    {
-        if (!StringUtils.isNullOrEmpty(username) && username.length() <= 16)
-        {
-            GameProfile gameprofile = server.getPlayerProfileCache().getGameProfileForUsername(username);
-
-            if (gameprofile != null && gameprofile.getId() != null)
-            {
-                return gameprofile.getId().toString();
-            }
-            else if (!server.isSinglePlayer() && server.isServerInOnlineMode())
-            {
-                final List<GameProfile> list = Lists.newArrayList();
-                ProfileLookupCallback profilelookupcallback = new ProfileLookupCallback()
-                {
-                    @Override
-                    public void onProfileLookupSucceeded(GameProfile p_onProfileLookupSucceeded_1_)
-                    {
-                        server.getPlayerProfileCache().addEntry(p_onProfileLookupSucceeded_1_);
-                        list.add(p_onProfileLookupSucceeded_1_);
-                    }
-                    @Override
-                    public void onProfileLookupFailed(GameProfile p_onProfileLookupFailed_1_, Exception p_onProfileLookupFailed_2_)
-                    {
-                        PreYggdrasilConverter.LOGGER
-                        .warn("Could not lookup user whitelist entry for {}", p_onProfileLookupFailed_1_.getName(), p_onProfileLookupFailed_2_);
-                    }
-                };
-                lookupNames(server, Lists.newArrayList(username), profilelookupcallback);
-                return !list.isEmpty() && list.get(0).getId() != null ? list.get(0).getId().toString() : "";
-            }
-            else
-            {
-                return EntityPlayer.getUUID(new GameProfile(null, username)).toString();
-            }
-        }
-        else
-        {
-            return username;
-        }
-    }
-
     @SideOnly(Side.SERVER)
     public static boolean convertSaveFiles(final DedicatedServer server, PropertyManager p_152723_1_)
     {
@@ -457,6 +458,29 @@
                     {
                         File file5 = new File(file1, p_152743_2_ + ".dat");
                         File file6 = new File(p_152743_1_, p_152743_3_ + ".dat");
+                        // CraftBukkit start - Use old file name to seed lastKnownName
+                        NBTTagCompound root = null;
+
+                        try {
+                            root = CompressedStreamTools.readCompressed(new java.io.FileInputStream(file1));
+                        } catch (Exception exception) {
+                            exception.printStackTrace();
+                        }
+
+                        if (root != null) {
+                            if (!root.hasKey("bukkit")) {
+                                root.setTag("bukkit", new NBTTagCompound());
+                            }
+                            NBTTagCompound data = root.getCompoundTag("bukkit");
+                            data.setString("lastKnownName", p_152743_2_);
+
+                            try {
+                                CompressedStreamTools.writeCompressed(root, new java.io.FileOutputStream(file2));
+                            } catch (Exception exception) {
+                                exception.printStackTrace();
+                            }
+                        }
+                        // CraftBukkit end
                         PreYggdrasilConverter.mkdir(p_152743_1_);
 
                         if (!file5.renameTo(file6))
@@ -611,7 +635,7 @@
     private static File getPlayersDirectory(PropertyManager properties)
     {
         String s = properties.getStringProperty("level-name", "world");
-        File file1 = new File(s);
+        File file1 = new File(MinecraftServer.getServerInst().server.getWorldContainer(), s); // CraftBukkit - Respect container setting
         return new File(file1, "players");
     }
 
