--- before/net/minecraft/server/management/UserList.java
+++ after/net/minecraft/server/management/UserList.java
@@ -157,12 +157,16 @@
 
     protected UserListEntry<K> createEntry(JsonObject entryData)
     {
-        return new UserListEntry<>(null, entryData);
+        return new UserListEntry<>((K)null, entryData);
     }
 
     protected Map<String, V> getValues()
     {
         return this.values;
+    }
+
+    public Collection<V> getValuesCB() {
+        return this.values.values();
     }
 
     public void writeChanges() throws IOException
