--- before/net/minecraft/client/util/JsonBlendingMode.java
+++ after/net/minecraft/client/util/JsonBlendingMode.java
@@ -110,9 +110,13 @@
             {
                 return false;
             }
+            else if (this.srcAlphaFactor != jsonblendingmode.srcAlphaFactor)
+            {
+                return false;
+            }
             else
             {
-                return this.srcAlphaFactor != jsonblendingmode.srcAlphaFactor ? false : this.srcColorFactor == jsonblendingmode.srcColorFactor;
+                return this.srcColorFactor == jsonblendingmode.srcColorFactor;
             }
         }
     }
