--- before/net/minecraft/client/util/SearchTree.java
+++ after/net/minecraft/client/util/SearchTree.java
@@ -119,7 +119,7 @@
                         i = Integer.compare(this.numbers.getInt(this.left), this.numbers.getInt(this.right));
                     }
 
-                    T t = i <= 0 ? this.left : this.right;
+                    T t = (T)(i <= 0 ? this.left : this.right);
 
                     if (i <= 0)
                     {
