--- before/net/minecraft/client/util/SuffixArray.java
+++ after/net/minecraft/client/util/SuffixArray.java
@@ -37,7 +37,7 @@
         this.list.add(p_194057_1_);
         this.wordStarts.add(this.chars.size());
 
-        for (int j = 0; j < p_194057_2_.length(); j++)
+        for (int j = 0; j < p_194057_2_.length(); ++j)
         {
             this.suffixToT.add(i);
             this.offsets.add(j);
@@ -87,7 +87,7 @@
             }
         };
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             aint[j] = this.chars.getInt(j);
         }
@@ -104,7 +104,7 @@
 
             Arrays.quickSort(0, i, intcomparator, swapper);
 
-            for (int l1 = 0; l1 < i; l1++)
+            for (int l1 = 0; l1 < i; ++l1)
             {
                 if (l1 > 0 && aint1[l1] == aint1[l1 - 1] && aint2[l1] == aint2[l1 - 1])
                 {
@@ -122,7 +122,7 @@
         this.suffixToT = new IntArrayList(intlist1.size());
         this.offsets = new IntArrayList(intlist.size());
 
-        for (int i1 = 0; i1 < i; i1++)
+        for (int i1 = 0; i1 < i; ++i1)
         {
             int j1 = aint3[i1];
             this.suffixToT.add(intlist1.getInt(j1));
@@ -137,7 +137,7 @@
 
     private void printArray()
     {
-        for (int i = 0; i < this.suffixToT.size(); i++)
+        for (int i = 0; i < this.suffixToT.size(); ++i)
         {
             LOGGER.debug("{} {}", i, this.getString(i));
         }
@@ -151,7 +151,7 @@
         int j = this.wordStarts.getInt(this.suffixToT.getInt(p_194059_1_));
         StringBuilder stringbuilder = new StringBuilder();
 
-        for (int k = 0; j + k < this.chars.size(); k++)
+        for (int k = 0; j + k < this.chars.size(); ++k)
         {
             if (k == i)
             {
@@ -176,7 +176,7 @@
         int i = this.wordStarts.getInt(this.suffixToT.getInt(p_194056_2_));
         int j = this.offsets.getInt(p_194056_2_);
 
-        for (int k = 0; k < p_194056_1_.length(); k++)
+        for (int k = 0; k < p_194056_1_.length(); ++k)
         {
             int l = this.chars.getInt(i + j + k);
 
@@ -256,7 +256,7 @@
             int k2 = j;
             IntSet intset = new IntOpenHashSet();
 
-            for (int k1 = i2; k1 < k2; k1++)
+            for (int k1 = i2; k1 < k2; ++k1)
             {
                 intset.add(this.suffixToT.getInt(k1));
             }
