--- before/net/minecraft/entity/EntityCreature.java
+++ after/net/minecraft/entity/EntityCreature.java
@@ -7,6 +7,7 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Vec3d;
 import net.minecraft.world.World;
+import org.bukkit.event.entity.EntityUnleashEvent;
 
 public abstract class EntityCreature extends EntityLiving
 {
@@ -44,7 +45,14 @@
 
     public boolean isWithinHomeDistanceFromPosition(BlockPos pos)
     {
-        return this.maximumHomeDistance == -1.0F ? true : this.homePosition.distanceSq(pos) < (double)(this.maximumHomeDistance * this.maximumHomeDistance);
+        if (this.maximumHomeDistance == -1.0F)
+        {
+            return true;
+        }
+        else
+        {
+            return this.homePosition.distanceSq(pos) < (double)(this.maximumHomeDistance * this.maximumHomeDistance);
+        }
     }
 
     public void setHomePosAndDistance(BlockPos pos, int distance)
@@ -88,6 +96,7 @@
             {
                 if (f > 10.0F)
                 {
+                    this.world.getServer().getPluginManager().callEvent(new EntityUnleashEvent(this.getBukkitEntity(), EntityUnleashEvent.UnleashReason.DISTANCE));
                     this.clearLeashed(true, true);
                 }
 
@@ -98,6 +107,7 @@
 
             if (f > 10.0F)
             {
+                this.world.getServer().getPluginManager().callEvent(new EntityUnleashEvent(this.getBukkitEntity(), EntityUnleashEvent.UnleashReason.DISTANCE));
                 this.clearLeashed(true, true);
                 this.tasks.disableControlFlag(1);
             }
@@ -106,9 +116,9 @@
                 double d0 = (entity.posX - this.posX) / (double)f;
                 double d1 = (entity.posY - this.posY) / (double)f;
                 double d2 = (entity.posZ - this.posZ) / (double)f;
-                this.motionX = this.motionX + d0 * Math.abs(d0) * 0.4;
-                this.motionY = this.motionY + d1 * Math.abs(d1) * 0.4;
-                this.motionZ = this.motionZ + d2 * Math.abs(d2) * 0.4;
+                this.motionX += d0 * Math.abs(d0) * 0.4;
+                this.motionY += d1 * Math.abs(d1) * 0.4;
+                this.motionZ += d2 * Math.abs(d2) * 0.4;
             }
             else
             {
