--- before/net/minecraft/network/play/client/CPacketUpdateSign.java
+++ after/net/minecraft/network/play/client/CPacketUpdateSign.java
@@ -34,7 +34,7 @@
         this.pos = buf.readBlockPos();
         this.lines = new String[4];
 
-        for (int i = 0; i < 4; i++)
+        for (int i = 0; i < 4; ++i)
         {
             this.lines[i] = buf.readString(384);
         }
@@ -45,7 +45,7 @@
     {
         buf.writeBlockPos(this.pos);
 
-        for (int i = 0; i < 4; i++)
+        for (int i = 0; i < 4; ++i)
         {
             buf.writeString(this.lines[i]);
         }
