--- before/net/minecraft/client/renderer/entity/layers/LayerDeadmau5Head.java
+++ after/net/minecraft/client/renderer/entity/layers/LayerDeadmau5Head.java
@@ -31,7 +31,7 @@
         {
             this.playerRenderer.bindTexture(entitylivingbaseIn.getLocationSkin());
 
-            for (int i = 0; i < 2; i++)
+            for (int i = 0; i < 2; ++i)
             {
                 float f = entitylivingbaseIn.prevRotationYaw
                           + (entitylivingbaseIn.rotationYaw - entitylivingbaseIn.prevRotationYaw) * partialTicks
