--- before/net/minecraft/client/renderer/entity/layers/LayerCustomHead.java
+++ after/net/minecraft/client/renderer/entity/layers/LayerCustomHead.java
@@ -97,7 +97,7 @@
 
                         if (!StringUtils.isBlank(s))
                         {
-                            gameprofile = TileEntitySkull.updateGameProfile(new GameProfile(null, s));
+                            gameprofile = TileEntitySkull.updateGameprofile(new GameProfile(null, s));
                             nbttagcompound.setTag("SkullOwner", NBTUtil.writeGameProfile(new NBTTagCompound(), gameprofile));
                         }
                     }
