--- before/net/minecraft/block/state/BlockPistonStructureHelper.java
+++ after/net/minecraft/block/state/BlockPistonStructureHelper.java
@@ -17,8 +17,8 @@
     private final BlockPos pistonPos;
     private final BlockPos blockToMove;
     private final EnumFacing moveDirection;
-    private final List<BlockPos> toMove = Lists.newArrayList();
-    private final List<BlockPos> toDestroy = Lists.newArrayList();
+    private final List<BlockPos> toMove = Lists.<BlockPos>newArrayList();
+    private final List<BlockPos> toDestroy = Lists.<BlockPos>newArrayList();
 
     public BlockPistonStructureHelper(World worldIn, BlockPos posIn, EnumFacing pistonFacing, boolean extending)
     {
@@ -61,11 +61,11 @@
         }
         else
         {
-            for (int i = 0; i < this.toMove.size(); i++)
+            for (int i = 0; i < this.toMove.size(); ++i)
             {
                 BlockPos blockpos = this.toMove.get(i);
 
-                if (this.world.getBlockState(blockpos).getBlock() == Blocks.SLIME_BLOCK && !this.addBranchingBlocks(blockpos))
+                if (this.world.getBlockState(blockpos).getBlock().isStickyBlock(this.world.getBlockState(blockpos)) && !this.addBranchingBlocks(blockpos))
                 {
                     return false;
                 }
@@ -80,7 +80,7 @@
         IBlockState iblockstate = this.world.getBlockState(origin);
         Block block = iblockstate.getBlock();
 
-        if (iblockstate.getMaterial() == Material.AIR)
+        if (iblockstate.getBlock().isAir(iblockstate, this.world, origin))
         {
             return true;
         }
@@ -106,20 +106,20 @@
             }
             else
             {
-                while (block == Blocks.SLIME_BLOCK)
+                while (block.isStickyBlock(iblockstate))
                 {
                     BlockPos blockpos = origin.offset(this.moveDirection.getOpposite(), i);
                     iblockstate = this.world.getBlockState(blockpos);
                     block = iblockstate.getBlock();
 
-                    if (iblockstate.getMaterial() == Material.AIR
-                            || !BlockPistonBase.canPush(iblockstate, this.world, blockpos, this.moveDirection, false, this.moveDirection.getOpposite())
-                            || blockpos.equals(this.pistonPos))
+                    if (iblockstate.getBlock().isAir(iblockstate, this.world, blockpos) || !BlockPistonBase.canPush(iblockstate, this.world, blockpos, this.moveDirection, false, this.moveDirection.getOpposite()) || blockpos.equals(this.pistonPos))
                     {
                         break;
                     }
 
-                    if (++i + this.toMove.size() > 12)
+                    ++i;
+
+                    if (i + this.toMove.size() > 12)
                     {
                         return false;
                     }
@@ -127,10 +127,10 @@
 
                 int i1 = 0;
 
-                for (int j = i - 1; j >= 0; j--)
+                for (int j = i - 1; j >= 0; --j)
                 {
                     this.toMove.add(origin.offset(this.moveDirection.getOpposite(), j));
-                    i1++;
+                    ++i1;
                 }
 
                 int j1 = 1;
@@ -144,11 +144,11 @@
                     {
                         this.reorderListAtCollision(i1, k);
 
-                        for (int l = 0; l <= k + i1; l++)
+                        for (int l = 0; l <= k + i1; ++l)
                         {
                             BlockPos blockpos2 = this.toMove.get(l);
 
-                            if (this.world.getBlockState(blockpos2).getBlock() == Blocks.SLIME_BLOCK && !this.addBranchingBlocks(blockpos2))
+                            if (this.world.getBlockState(blockpos2).getBlock().isStickyBlock(this.world.getBlockState(blockpos2)) && !this.addBranchingBlocks(blockpos2))
                             {
                                 return false;
                             }
@@ -159,13 +159,12 @@
 
                     iblockstate = this.world.getBlockState(blockpos1);
 
-                    if (iblockstate.getMaterial() == Material.AIR)
+                    if (iblockstate.getBlock().isAir(iblockstate, this.world, blockpos1))
                     {
                         return true;
                     }
 
-                    if (!BlockPistonBase.canPush(iblockstate, this.world, blockpos1, this.moveDirection, true, this.moveDirection)
-                            || blockpos1.equals(this.pistonPos))
+                    if (!BlockPistonBase.canPush(iblockstate, this.world, blockpos1, this.moveDirection, true, this.moveDirection) || blockpos1.equals(this.pistonPos))
                     {
                         return false;
                     }
@@ -182,8 +181,8 @@
                     }
 
                     this.toMove.add(blockpos1);
-                    i1++;
-                    j1++;
+                    ++i1;
+                    ++j1;
                 }
             }
         }
@@ -191,9 +190,9 @@
 
     private void reorderListAtCollision(int p_177255_1_, int p_177255_2_)
     {
-        List<BlockPos> list = Lists.newArrayList();
-        List<BlockPos> list1 = Lists.newArrayList();
-        List<BlockPos> list2 = Lists.newArrayList();
+        List<BlockPos> list = Lists.<BlockPos>newArrayList();
+        List<BlockPos> list1 = Lists.<BlockPos>newArrayList();
+        List<BlockPos> list2 = Lists.<BlockPos>newArrayList();
         list.addAll(this.toMove.subList(0, p_177255_2_));
         list1.addAll(this.toMove.subList(this.toMove.size() - p_177255_1_, this.toMove.size()));
         list2.addAll(this.toMove.subList(p_177255_2_, this.toMove.size() - p_177255_1_));
