--- before/net/minecraft/world/gen/ChunkGeneratorSettings.java
+++ after/net/minecraft/world/gen/ChunkGeneratorSettings.java
@@ -698,11 +698,13 @@
                     {
                         return false;
                     }
+                    else if (this.useWaterLakes != chunkgeneratorsettings$factory.useWaterLakes)
+                    {
+                        return false;
+                    }
                     else
                     {
-                        return this.useWaterLakes != chunkgeneratorsettings$factory.useWaterLakes
-                               ? false
-                               : this.waterLakeChance == chunkgeneratorsettings$factory.waterLakeChance;
+                        return this.waterLakeChance == chunkgeneratorsettings$factory.waterLakeChance;
                     }
                 }
                 else
