--- before/net/minecraft/world/gen/ChunkGeneratorDebug.java
+++ after/net/minecraft/world/gen/ChunkGeneratorDebug.java
@@ -33,9 +33,9 @@
     {
         ChunkPrimer chunkprimer = new ChunkPrimer();
 
-        for (int i = 0; i < 16; i++)
+        for (int i = 0; i < 16; ++i)
         {
-            for (int j = 0; j < 16; j++)
+            for (int j = 0; j < 16; ++j)
             {
                 int k = x * 16 + i;
                 int l = z * 16 + j;
@@ -54,7 +54,7 @@
         Biome[] abiome = this.world.getBiomeProvider().getBiomes(null, x * 16, z * 16, 16, 16);
         byte[] abyte = chunk.getBiomeArray();
 
-        for (int i1 = 0; i1 < abyte.length; i1++)
+        for (int i1 = 0; i1 < abyte.length; ++i1)
         {
             abyte[i1] = (byte)Biome.getIdForBiome(abiome[i1]);
         }
