--- before/net/minecraft/world/gen/NoiseGeneratorSimplex.java
+++ after/net/minecraft/world/gen/NoiseGeneratorSimplex.java
@@ -33,7 +33,7 @@
             this.p[i] = i++;
         }
 
-        for (int l = 0; l < 256; l++)
+        for (int l = 0; l < 256; ++l)
         {
             int j = seed.nextInt(256 - l) + l;
             int k = this.p[l];
@@ -143,11 +143,11 @@
     {
         int i = 0;
 
-        for (int j = 0; j < p_151606_7_; j++)
+        for (int j = 0; j < p_151606_7_; ++j)
         {
             double d0 = (p_151606_4_ + (double)j) * p_151606_10_ + this.yo;
 
-            for (int k = 0; k < p_151606_6_; k++)
+            for (int k = 0; k < p_151606_6_; ++k)
             {
                 double d1 = (p_151606_2_ + (double)k) * p_151606_8_ + this.xo;
                 double d5 = (d1 + d0) * F2;
@@ -220,7 +220,8 @@
                     d4 = d17 * d17 * dot(grad3[l2], d13, d14);
                 }
 
-                p_151606_1_[i++] += 70.0 * (d2 + d3 + d4) * p_151606_12_;
+                int i3 = i++;
+                p_151606_1_[i3] += 70.0 * (d2 + d3 + d4) * p_151606_12_;
             }
         }
     }
