--- before/net/minecraft/world/gen/NoiseGeneratorOctaves.java
+++ after/net/minecraft/world/gen/NoiseGeneratorOctaves.java
@@ -13,7 +13,7 @@
         this.field_76306_b = p_i2111_2_;
         this.field_76307_a = new NoiseGeneratorImproved[p_i2111_2_];
 
-        for (int i = 0; i < p_i2111_2_; i++)
+        for (int i = 0; i < p_i2111_2_; ++i)
         {
             this.field_76307_a[i] = new NoiseGeneratorImproved(p_i2111_1_);
         }
@@ -38,7 +38,7 @@
         }
         else
         {
-            for (int i = 0; i < p_76304_1_.length; i++)
+            for (int i = 0; i < p_76304_1_.length; ++i)
             {
                 p_76304_1_[i] = 0.0;
             }
@@ -46,7 +46,7 @@
 
         double d3 = 1.0;
 
-        for (int j = 0; j < this.field_76306_b; j++)
+        for (int j = 0; j < this.field_76306_b; ++j)
         {
             double d0 = (double)p_76304_2_ * d3 * p_76304_8_;
             double d1 = (double)p_76304_3_ * d3 * p_76304_10_;
