--- before/net/minecraft/world/gen/NoiseGeneratorPerlin.java
+++ after/net/minecraft/world/gen/NoiseGeneratorPerlin.java
@@ -12,7 +12,7 @@
         this.levels = levelsIn;
         this.noiseLevels = new NoiseGeneratorSimplex[levelsIn];
 
-        for (int i = 0; i < levelsIn; i++)
+        for (int i = 0; i < levelsIn; ++i)
         {
             this.noiseLevels[i] = new NoiseGeneratorSimplex(seed);
         }
@@ -23,7 +23,7 @@
         double d0 = 0.0;
         double d1 = 1.0;
 
-        for (int i = 0; i < this.levels; i++)
+        for (int i = 0; i < this.levels; ++i)
         {
             d0 += this.noiseLevels[i].getValue(p_151601_1_ * d1, p_151601_3_ * d1) / d1;
             d1 /= 2.0;
@@ -60,7 +60,7 @@
     {
         if (p_151600_1_ != null && p_151600_1_.length >= p_151600_6_ * p_151600_7_)
         {
-            for (int i = 0; i < p_151600_1_.length; i++)
+            for (int i = 0; i < p_151600_1_.length; ++i)
             {
                 p_151600_1_[i] = 0.0;
             }
@@ -73,7 +73,7 @@
         double d1 = 1.0;
         double d0 = 1.0;
 
-        for (int j = 0; j < this.levels; j++)
+        for (int j = 0; j < this.levels; ++j)
         {
             this.noiseLevels[j]
             .add(p_151600_1_, p_151600_2_, p_151600_4_, p_151600_6_, p_151600_7_, p_151600_8_ * d0 * d1, p_151600_10_ * d0 * d1, 0.55 / d1);
