--- before/net/minecraft/command/server/CommandSaveAll.java
+++ after/net/minecraft/command/server/CommandSaveAll.java
@@ -38,7 +38,7 @@
 
         try
         {
-            for (int i = 0; i < p_184881_1_.field_71305_c.length; i++)
+            for (int i = 0; i < p_184881_1_.field_71305_c.length; ++i)
             {
                 if (p_184881_1_.field_71305_c[i] != null)
                 {
@@ -54,7 +54,7 @@
             {
                 p_184881_2_.func_145747_a(new TextComponentTranslation("commands.save.flushStart"));
 
-                for (int j = 0; j < p_184881_1_.field_71305_c.length; j++)
+                for (int j = 0; j < p_184881_1_.field_71305_c.length; ++j)
                 {
                     if (p_184881_1_.field_71305_c[j] != null)
                     {
