--- before/net/minecraft/command/server/CommandScoreboard.java
+++ after/net/minecraft/command/server/CommandScoreboard.java
@@ -293,7 +293,7 @@
     {
         int i = -1;
 
-        for (int j = 0; j < args.length; j++)
+        for (int j = 0; j < args.length; ++j)
         {
             if (this.isUsernameIndex(args, j) && "*".equals(args[j]))
             {
@@ -1075,7 +1075,7 @@
             else
             {
                 int i = args[startIndex].equals("*") ? Integer.MIN_VALUE : parseInt(args[startIndex]);
-                startIndex++;
+                ++startIndex;
                 int j = startIndex < args.length && !args[startIndex].equals("*") ? parseInt(args[startIndex], i) : Integer.MAX_VALUE;
                 Score score = scoreboard.getOrCreateScore(s, scoreobjective);
 
@@ -1491,11 +1491,22 @@
     {
         if (!"players".equalsIgnoreCase(args[0]))
         {
-            return "teams".equalsIgnoreCase(args[0]) ? index == 2 : false;
+            if ("teams".equalsIgnoreCase(args[0]))
+            {
+                return index == 2;
+            }
+            else
+            {
+                return false;
+            }
+        }
+        else if (args.length > 1 && "operation".equalsIgnoreCase(args[1]))
+        {
+            return index == 2 || index == 5;
         }
         else
         {
-            return args.length > 1 && "operation".equalsIgnoreCase(args[1]) ? index == 2 || index == 5 : index == 2;
+            return index == 2;
         }
     }
 }
