--- before/net/minecraft/command/server/CommandSaveAll.java
+++ after/net/minecraft/command/server/CommandSaveAll.java
@@ -38,7 +38,7 @@
 
         try
         {
-            for (int i = 0; i < server.worlds.length; i++)
+            for (int i = 0; i < server.worlds.length; ++i)
             {
                 if (server.worlds[i] != null)
                 {
@@ -54,7 +54,7 @@
             {
                 sender.sendMessage(new TextComponentTranslation("commands.save.flushStart"));
 
-                for (int j = 0; j < server.worlds.length; j++)
+                for (int j = 0; j < server.worlds.length; ++j)
                 {
                     if (server.worlds[j] != null)
                     {
