--- before/net/minecraft/client/renderer/GlStateManager.java
+++ after/net/minecraft/client/renderer/GlStateManager.java
@@ -874,7 +874,7 @@
 
     static
     {
-        for (int i = 0; i < 8; i++)
+        for (int i = 0; i < 8; ++i)
         {
             field_179159_c[i] = new GlStateManager.BooleanState(16384 + i);
         }
@@ -892,7 +892,7 @@
         field_179161_n = new GlStateManager.BooleanState(2977);
         field_179174_p = new GlStateManager.TextureState[8];
 
-        for (int j = 0; j < 8; j++)
+        for (int j = 0; j < 8; ++j)
         {
             field_179174_p[j] = new GlStateManager.TextureState();
         }
@@ -1183,7 +1183,7 @@
                 GlStateManager.func_179140_f();
                 GlStateManager.func_187424_a(2899, RenderHelper.func_74521_a(0.2F, 0.2F, 0.2F, 1.0F));
 
-                for (int i = 0; i < 8; i++)
+                for (int i = 0; i < 8; ++i)
                 {
                     GlStateManager.func_179122_b(i);
                     GlStateManager.func_187438_a(16384 + i, 4608, RenderHelper.func_74521_a(0.0F, 0.0F, 0.0F, 1.0F));
