--- before/net/minecraft/client/renderer/GlStateManager.java
+++ after/net/minecraft/client/renderer/GlStateManager.java
@@ -874,7 +874,7 @@
 
     static
     {
-        for (int i = 0; i < 8; i++)
+        for (int i = 0; i < 8; ++i)
         {
             lightState[i] = new GlStateManager.BooleanState(16384 + i);
         }
@@ -892,7 +892,7 @@
         normalizeState = new GlStateManager.BooleanState(2977);
         textureState = new GlStateManager.TextureState[8];
 
-        for (int j = 0; j < 8; j++)
+        for (int j = 0; j < 8; ++j)
         {
             textureState[j] = new GlStateManager.TextureState();
         }
@@ -1183,7 +1183,7 @@
                 GlStateManager.disableLighting();
                 GlStateManager.glLightModel(2899, RenderHelper.setColorBuffer(0.2F, 0.2F, 0.2F, 1.0F));
 
-                for (int i = 0; i < 8; i++)
+                for (int i = 0; i < 8; ++i)
                 {
                     GlStateManager.disableLight(i);
                     GlStateManager.glLight(16384 + i, 4608, RenderHelper.setColorBuffer(0.0F, 0.0F, 0.0F, 1.0F));
