--- before/net/minecraft/client/renderer/ImageBufferDownload.java
+++ after/net/minecraft/client/renderer/ImageBufferDownload.java
@@ -72,9 +72,9 @@
 
     private void func_189559_a(int p_189559_1_, int p_189559_2_, int p_189559_3_, int p_189559_4_)
     {
-        for (int i = p_189559_1_; i < p_189559_3_; i++)
+        for (int i = p_189559_1_; i < p_189559_3_; ++i)
         {
-            for (int j = p_189559_2_; j < p_189559_4_; j++)
+            for (int j = p_189559_2_; j < p_189559_4_; ++j)
             {
                 int k = this.field_78438_a[i + j * this.field_78436_b];
 
@@ -85,22 +85,22 @@
             }
         }
 
-        for (int l = p_189559_1_; l < p_189559_3_; l++)
+        for (int l = p_189559_1_; l < p_189559_3_; ++l)
         {
-            for (int i1 = p_189559_2_; i1 < p_189559_4_; i1++)
+            for (int i1 = p_189559_2_; i1 < p_189559_4_; ++i1)
             {
-                this.field_78438_a[l + i1 * this.field_78436_b] = this.field_78438_a[l + i1 * this.field_78436_b] & 16777215;
+                this.field_78438_a[l + i1 * this.field_78436_b] &= 16777215;
             }
         }
     }
 
     private void func_78433_b(int p_78433_1_, int p_78433_2_, int p_78433_3_, int p_78433_4_)
     {
-        for (int i = p_78433_1_; i < p_78433_3_; i++)
+        for (int i = p_78433_1_; i < p_78433_3_; ++i)
         {
-            for (int j = p_78433_2_; j < p_78433_4_; j++)
+            for (int j = p_78433_2_; j < p_78433_4_; ++j)
             {
-                this.field_78438_a[i + j * this.field_78436_b] = this.field_78438_a[i + j * this.field_78436_b] | 0xFF000000;
+                this.field_78438_a[i + j * this.field_78436_b] |= -16777216;
             }
         }
     }
