--- before/net/minecraft/client/renderer/ImageBufferDownload.java
+++ after/net/minecraft/client/renderer/ImageBufferDownload.java
@@ -72,9 +72,9 @@
 
     private void setAreaTransparent(int x, int y, int width, int height)
     {
-        for (int i = x; i < width; i++)
+        for (int i = x; i < width; ++i)
         {
-            for (int j = y; j < height; j++)
+            for (int j = y; j < height; ++j)
             {
                 int k = this.imageData[i + j * this.imageWidth];
 
@@ -85,22 +85,22 @@
             }
         }
 
-        for (int l = x; l < width; l++)
+        for (int l = x; l < width; ++l)
         {
-            for (int i1 = y; i1 < height; i1++)
+            for (int i1 = y; i1 < height; ++i1)
             {
-                this.imageData[l + i1 * this.imageWidth] = this.imageData[l + i1 * this.imageWidth] & 16777215;
+                this.imageData[l + i1 * this.imageWidth] &= 16777215;
             }
         }
     }
 
     private void setAreaOpaque(int x, int y, int width, int height)
     {
-        for (int i = x; i < width; i++)
+        for (int i = x; i < width; ++i)
         {
-            for (int j = y; j < height; j++)
+            for (int j = y; j < height; ++j)
             {
-                this.imageData[i + j * this.imageWidth] = this.imageData[i + j * this.imageWidth] | 0xFF000000;
+                this.imageData[i + j * this.imageWidth] |= -16777216;
             }
         }
     }
