--- before/net/minecraft/network/play/server/SPacketExplosion.java
+++ after/net/minecraft/network/play/server/SPacketExplosion.java
@@ -55,7 +55,7 @@
         int k = (int)this.field_149156_b;
         int l = (int)this.field_149157_c;
 
-        for (int i1 = 0; i1 < i; i1++)
+        for (int i1 = 0; i1 < i; ++i1)
         {
             int j1 = p_148837_1_.readByte() + j;
             int k1 = p_148837_1_.readByte() + k;
