--- before/net/minecraft/network/play/server/SPacketMaps.java
+++ after/net/minecraft/network/play/server/SPacketMaps.java
@@ -48,9 +48,9 @@
         this.rows = rowsIn;
         this.mapDataBytes = new byte[columnsIn * rowsIn];
 
-        for (int i = 0; i < columnsIn; i++)
+        for (int i = 0; i < columnsIn; ++i)
         {
-            for (int j = 0; j < rowsIn; j++)
+            for (int j = 0; j < rowsIn; ++j)
             {
                 this.mapDataBytes[i + j * columnsIn] = p_i46937_5_[minXIn + i + (minZIn + j) * 128];
             }
@@ -65,7 +65,7 @@
         this.trackingPosition = buf.readBoolean();
         this.icons = new MapDecoration[buf.readVarInt()];
 
-        for (int i = 0; i < this.icons.length; i++)
+        for (int i = 0; i < this.icons.length; ++i)
         {
             short short1 = (short)buf.readByte();
             this.icons[i] = new MapDecoration(
@@ -128,15 +128,15 @@
         mapdataIn.trackingPosition = this.trackingPosition;
         mapdataIn.mapDecorations.clear();
 
-        for (int i = 0; i < this.icons.length; i++)
+        for (int i = 0; i < this.icons.length; ++i)
         {
             MapDecoration mapdecoration = this.icons[i];
             mapdataIn.mapDecorations.put("icon-" + i, mapdecoration);
         }
 
-        for (int j = 0; j < this.columns; j++)
+        for (int j = 0; j < this.columns; ++j)
         {
-            for (int k = 0; k < this.rows; k++)
+            for (int k = 0; k < this.rows; ++k)
             {
                 mapdataIn.colors[this.minX + j + (this.minZ + k) * 128] = this.mapDataBytes[j + k * this.columns];
             }
