--- before/net/minecraft/network/play/server/SPacketSetPassengers.java
+++ after/net/minecraft/network/play/server/SPacketSetPassengers.java
@@ -24,7 +24,7 @@
         List<Entity> list = entityIn.getPassengers();
         this.passengerIds = new int[list.size()];
 
-        for (int i = 0; i < list.size(); i++)
+        for (int i = 0; i < list.size(); ++i)
         {
             this.passengerIds[i] = list.get(i).getEntityId();
         }
