--- before/net/minecraft/network/play/server/SPacketAdvancementInfo.java
+++ after/net/minecraft/network/play/server/SPacketAdvancementInfo.java
@@ -57,7 +57,7 @@
         this.progressUpdates = Maps.newHashMap();
         int i = buf.readVarInt();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             ResourceLocation resourcelocation = buf.readResourceLocation();
             Advancement.Builder advancement$builder = Advancement.Builder.readFrom(buf);
@@ -66,7 +66,7 @@
 
         i = buf.readVarInt();
 
-        for (int k = 0; k < i; k++)
+        for (int k = 0; k < i; ++k)
         {
             ResourceLocation resourcelocation1 = buf.readResourceLocation();
             this.advancementsToRemove.add(resourcelocation1);
@@ -74,7 +74,7 @@
 
         i = buf.readVarInt();
 
-        for (int l = 0; l < i; l++)
+        for (int l = 0; l < i; ++l)
         {
             ResourceLocation resourcelocation2 = buf.readResourceLocation();
             this.progressUpdates.put(resourcelocation2, AdvancementProgress.fromNetwork(buf));
