--- before/net/minecraft/network/play/server/SPacketStatistics.java
+++ after/net/minecraft/network/play/server/SPacketStatistics.java
@@ -36,7 +36,7 @@
         int i = buf.readVarInt();
         this.statisticMap = Maps.newHashMap();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             StatBase statbase = StatList.getOneShotStat(buf.readString(32767));
             int k = buf.readVarInt();
