--- before/net/minecraft/network/play/server/SPacketChunkData.java
+++ after/net/minecraft/network/play/server/SPacketChunkData.java
@@ -74,7 +74,7 @@
             int j = buf.readVarInt();
             this.tileEntityTags = Lists.newArrayList();
 
-            for (int k = 0; k < j; k++)
+            for (int k = 0; k < j; ++k)
             {
                 this.tileEntityTags.add(buf.readCompoundTag());
             }
@@ -122,7 +122,7 @@
         ExtendedBlockStorage[] aextendedblockstorage = chunkIn.getBlockStorageArray();
         int j = 0;
 
-        for (int k = aextendedblockstorage.length; j < k; j++)
+        for (int k = aextendedblockstorage.length; j < k; ++j)
         {
             ExtendedBlockStorage extendedblockstorage = aextendedblockstorage[j];
 
@@ -153,7 +153,7 @@
         ExtendedBlockStorage[] aextendedblockstorage = chunkIn.getBlockStorageArray();
         int j = 0;
 
-        for (int k = aextendedblockstorage.length; j < k; j++)
+        for (int k = aextendedblockstorage.length; j < k; ++j)
         {
             ExtendedBlockStorage extendedblockstorage = aextendedblockstorage[j];
 
