--- before/net/minecraft/network/play/server/SPacketPlayerListItem.java
+++ after/net/minecraft/network/play/server/SPacketPlayerListItem.java
@@ -61,7 +61,7 @@
         this.action = buf.readEnumValue(SPacketPlayerListItem.Action.class);
         int i = buf.readVarInt();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             GameProfile gameprofile = null;
             int k = 0;
@@ -75,7 +75,7 @@
                     int l = buf.readVarInt();
                     int i1 = 0;
 
-                    for (; i1 < l; i1++)
+                    for (; i1 < l; ++i1)
                     {
                         String s = buf.readString(32767);
                         String s1 = buf.readString(32767);
