--- before/net/minecraft/network/play/server/SPacketRecipeBook.java
+++ after/net/minecraft/network/play/server/SPacketRecipeBook.java
@@ -46,7 +46,7 @@
         int i = buf.readVarInt();
         this.recipes = Lists.newArrayList();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             this.recipes.add(CraftingManager.getRecipeById(buf.readVarInt()));
         }
@@ -56,7 +56,7 @@
             i = buf.readVarInt();
             this.displayedRecipes = Lists.newArrayList();
 
-            for (int k = 0; k < i; k++)
+            for (int k = 0; k < i; ++k)
             {
                 this.displayedRecipes.add(CraftingManager.getRecipeById(buf.readVarInt()));
             }
