--- before/net/minecraft/network/play/server/SPacketMultiBlockChange.java
+++ after/net/minecraft/network/play/server/SPacketMultiBlockChange.java
@@ -26,7 +26,7 @@
         this.chunkPos = new ChunkPos(p_i46959_3_.x, p_i46959_3_.z);
         this.changedBlocks = new SPacketMultiBlockChange.BlockUpdateData[p_i46959_1_];
 
-        for (int i = 0; i < this.changedBlocks.length; i++)
+        for (int i = 0; i < this.changedBlocks.length; ++i)
         {
             this.changedBlocks[i] = new SPacketMultiBlockChange.BlockUpdateData(p_i46959_2_[i], p_i46959_3_);
         }
@@ -38,7 +38,7 @@
         this.chunkPos = new ChunkPos(buf.readInt(), buf.readInt());
         this.changedBlocks = new SPacketMultiBlockChange.BlockUpdateData[buf.readVarInt()];
 
-        for (int i = 0; i < this.changedBlocks.length; i++)
+        for (int i = 0; i < this.changedBlocks.length; ++i)
         {
             this.changedBlocks[i] = new SPacketMultiBlockChange.BlockUpdateData(
                 buf.readShort(), Block.BLOCK_STATE_IDS.getByValue(buf.readVarInt())
