--- before/net/minecraft/network/play/server/SPacketEntityProperties.java
+++ after/net/minecraft/network/play/server/SPacketEntityProperties.java
@@ -43,14 +43,14 @@
         this.entityId = buf.readVarInt();
         int i = buf.readInt();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             String s = buf.readString(64);
             double d0 = buf.readDouble();
             List<AttributeModifier> list = Lists.newArrayList();
             int k = buf.readVarInt();
 
-            for (int l = 0; l < k; l++)
+            for (int l = 0; l < k; ++l)
             {
                 UUID uuid = buf.readUniqueId();
                 list.add(new AttributeModifier(uuid, "Unknown synced attribute modifier", buf.readDouble(), buf.readByte()));
