--- before/net/minecraft/network/play/server/SPacketPlayerListItem.java
+++ after/net/minecraft/network/play/server/SPacketPlayerListItem.java
@@ -61,7 +61,7 @@
         this.field_179770_a = p_148837_1_.func_179257_a(SPacketPlayerListItem.Action.class);
         int i = p_148837_1_.func_150792_a();
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             GameProfile gameprofile = null;
             int k = 0;
@@ -75,7 +75,7 @@
                     int l = p_148837_1_.func_150792_a();
                     int i1 = 0;
 
-                    for (; i1 < l; i1++)
+                    for (; i1 < l; ++i1)
                     {
                         String s = p_148837_1_.func_150789_c(32767);
                         String s1 = p_148837_1_.func_150789_c(32767);
