--- before/net/minecraft/network/play/server/SPacketWindowItems.java
+++ after/net/minecraft/network/play/server/SPacketWindowItems.java
@@ -24,7 +24,7 @@
         this.windowId = p_i47317_1_;
         this.itemStacks = NonNullList.withSize(p_i47317_2_.size(), ItemStack.EMPTY);
 
-        for (int i = 0; i < this.itemStacks.size(); i++)
+        for (int i = 0; i < this.itemStacks.size(); ++i)
         {
             ItemStack itemstack = p_i47317_2_.get(i);
             this.itemStacks.set(i, itemstack.copy());
@@ -38,7 +38,7 @@
         int i = buf.readShort();
         this.itemStacks = NonNullList.withSize(i, ItemStack.EMPTY);
 
-        for (int j = 0; j < i; j++)
+        for (int j = 0; j < i; ++j)
         {
             this.itemStacks.set(j, buf.readItemStack());
         }
