--- before/net/minecraft/client/network/ServerPinger.java
+++ after/net/minecraft/client/network/ServerPinger.java
@@ -50,30 +50,26 @@
 {
     private static final Splitter PING_RESPONSE_SPLITTER = Splitter.on('\u0000').limit(6);
     private static final Logger LOGGER = LogManager.getLogger();
-    private final List<NetworkManager> pingDestinations = Collections.synchronizedList(Lists.newArrayList());
+    private final List<NetworkManager> pingDestinations = Collections.<NetworkManager>synchronizedList(Lists.newArrayList());
 
     public void ping(final ServerData server) throws UnknownHostException
     {
         ServerAddress serveraddress = ServerAddress.fromString(server.serverIP);
-        final NetworkManager networkmanager = NetworkManager.createNetworkManagerAndConnect(
-                                                  InetAddress.getByName(serveraddress.getIP()), serveraddress.getPort(), false
-                                              );
+        final NetworkManager networkmanager = NetworkManager.createNetworkManagerAndConnect(InetAddress.getByName(serveraddress.getIP()), serveraddress.getPort(), false);
         this.pingDestinations.add(networkmanager);
         server.serverMOTD = I18n.format("multiplayer.status.pinging");
         server.pingToServer = -1L;
         server.playerList = null;
-        networkmanager.setNetHandler(
-            new INetHandlerStatusClient()
+        networkmanager.setNetHandler(new INetHandlerStatusClient()
         {
             private boolean successful;
             private boolean receivedStatus;
             private long pingSentAt;
-            @Override
             public void handleServerInfo(SPacketServerInfo packetIn)
             {
                 if (this.receivedStatus)
                 {
-                    networkmanager.closeChannel(new TextComponentTranslation("multiplayer.status.unrequested"));
+                    networkmanager.closeChannel(new TextComponentTranslation("multiplayer.status.unrequested", new Object[0]));
                 }
                 else
                 {
@@ -102,14 +98,7 @@
 
                     if (serverstatusresponse.getPlayers() != null)
                     {
-                        server.populationInfo = TextFormatting.GRAY
-                                                    + ""
-                                                    + serverstatusresponse.getPlayers().getOnlinePlayerCount()
-                                                    + ""
-                                                    + TextFormatting.DARK_GRAY
-                                                    + "/"
-                                                    + TextFormatting.GRAY
-                                                    + serverstatusresponse.getPlayers().getMaxPlayers();
+                        server.populationInfo = TextFormatting.GRAY + "" + serverstatusresponse.getPlayers().getOnlinePlayerCount() + "" + TextFormatting.DARK_GRAY + "/" + TextFormatting.GRAY + serverstatusresponse.getPlayers().getMaxPlayers();
 
                         if (ArrayUtils.isNotEmpty(serverstatusresponse.getPlayers().getPlayers()))
                         {
@@ -132,12 +121,7 @@
                                     stringbuilder.append("\n");
                                 }
 
-                                stringbuilder.append(
-                                    I18n.format(
-                                        "multiplayer.status.and_more",
-                                        serverstatusresponse.getPlayers().getOnlinePlayerCount() - serverstatusresponse.getPlayers().getPlayers().length
-                                    )
-                                );
+                                stringbuilder.append(I18n.format("multiplayer.status.and_more", serverstatusresponse.getPlayers().getOnlinePlayerCount() - serverstatusresponse.getPlayers().getPlayers().length));
                             }
 
                             server.playerList = stringbuilder.toString();
@@ -163,15 +147,15 @@
                     }
                     else
                     {
-                        server.setBase64EncodedIconData(null);
+                        server.setBase64EncodedIconData((String)null);
                     }
 
+                    net.minecraftforge.fml.client.FMLClientHandler.instance().bindServerListData(server, serverstatusresponse);
                     this.pingSentAt = Minecraft.getSystemTime();
                     networkmanager.sendPacket(new CPacketPing(this.pingSentAt));
                     this.successful = true;
                 }
             }
-            @Override
             public void handlePong(SPacketPong packetIn)
             {
                 long i = this.pingSentAt;
@@ -179,7 +163,6 @@
                 server.pingToServer = j - i;
                 networkmanager.closeChannel(new TextComponentString("Finished"));
             }
-            @Override
             public void onDisconnect(ITextComponent reason)
             {
                 if (!this.successful)
@@ -190,12 +173,11 @@
                     ServerPinger.this.tryCompatibilityPing(server);
                 }
             }
-        }
-        );
+        });
 
         try
         {
-            networkmanager.sendPacket(new C00Handshake(serveraddress.getIP(), serveraddress.getPort(), EnumConnectionState.STATUS));
+            networkmanager.sendPacket(new C00Handshake(serveraddress.getIP(), serveraddress.getPort(), EnumConnectionState.STATUS, true));
             networkmanager.sendPacket(new CPacketServerQuery());
         }
         catch (Throwable throwable)
@@ -207,22 +189,21 @@
     private void tryCompatibilityPing(final ServerData server)
     {
         final ServerAddress serveraddress = ServerAddress.fromString(server.serverIP);
-        new Bootstrap().group(NetworkManager.CLIENT_NIO_EVENTLOOP.getValue()).handler(new ChannelInitializer<Channel>()
+        ((Bootstrap)((Bootstrap)((Bootstrap)(new Bootstrap()).group(NetworkManager.CLIENT_NIO_EVENTLOOP.getValue())).handler(new ChannelInitializer<Channel>()
         {
-            @Override
             protected void initChannel(Channel p_initChannel_1_) throws Exception
             {
                 try
                 {
-                    p_initChannel_1_.config().setOption(ChannelOption.TCP_NODELAY, true);
+                    p_initChannel_1_.config().setOption(ChannelOption.TCP_NODELAY, Boolean.valueOf(true));
                 }
-                catch (ChannelException channelexception)
+                catch (ChannelException var3)
                 {
+                    ;
                 }
 
                 p_initChannel_1_.pipeline().addLast(new SimpleChannelInboundHandler<ByteBuf>()
                 {
-                    @Override
                     public void channelActive(ChannelHandlerContext p_channelActive_1_) throws Exception
                     {
                         super.channelActive(p_channelActive_1_);
@@ -266,7 +247,7 @@
                         if (short1 == 255)
                         {
                             String s = new String(p_channelRead0_2_.readBytes(p_channelRead0_2_.readShort() * 2).array(), StandardCharsets.UTF_16BE);
-                            String[] astring = Iterables.toArray(ServerPinger.PING_RESPONSE_SPLITTER.split(s), String.class);
+                            String[] astring = (String[])Iterables.toArray(ServerPinger.PING_RESPONSE_SPLITTER.split(s), String.class);
 
                             if ("\u00a71".equals(astring[0]))
                             {
@@ -284,14 +265,13 @@
 
                         p_channelRead0_1_.close();
                     }
-                    @Override
                     public void exceptionCaught(ChannelHandlerContext p_exceptionCaught_1_, Throwable p_exceptionCaught_2_) throws Exception
                     {
                         p_exceptionCaught_1_.close();
                     }
                 });
             }
-        }).channel(NioSocketChannel.class).connect(serveraddress.getIP(), serveraddress.getPort());
+        })).channel(NioSocketChannel.class)).connect(serveraddress.getIP(), serveraddress.getPort());
     }
 
     public void pingPendingNetworks()
@@ -330,7 +310,7 @@
                 if (networkmanager.isChannelOpen())
                 {
                     iterator.remove();
-                    networkmanager.closeChannel(new TextComponentTranslation("multiplayer.status.cancelled"));
+                    networkmanager.closeChannel(new TextComponentTranslation("multiplayer.status.cancelled", new Object[0]));
                 }
             }
         }
