--- before/net/minecraft/block/properties/PropertyHelper.java
+++ after/net/minecraft/block/properties/PropertyHelper.java
@@ -44,7 +44,7 @@
         }
         else
         {
-            PropertyHelper<?> propertyhelper = (PropertyHelper<?>)p_equals_1_;
+            PropertyHelper<?> propertyhelper = (PropertyHelper)p_equals_1_;
             return this.valueClass.equals(propertyhelper.valueClass) && this.name.equals(propertyhelper.name);
         }
     }
