--- before/net/minecraft/block/properties/PropertyEnum.java
+++ after/net/minecraft/block/properties/PropertyEnum.java
@@ -60,7 +60,7 @@
         }
         else if (p_equals_1_ instanceof PropertyEnum && super.equals(p_equals_1_))
         {
-            PropertyEnum<?> propertyenum = (PropertyEnum<?>)p_equals_1_;
+            PropertyEnum<?> propertyenum = (PropertyEnum)p_equals_1_;
             return this.allowedValues.equals(propertyenum.allowedValues) && this.nameToValue.equals(propertyenum.nameToValue);
         }
         else
