--- before/net/minecraft/block/properties/PropertyHelper.java
+++ after/net/minecraft/block/properties/PropertyHelper.java
@@ -44,7 +44,7 @@
         }
         else
         {
-            PropertyHelper<?> propertyhelper = (PropertyHelper<?>)p_equals_1_;
+            PropertyHelper<?> propertyhelper = (PropertyHelper)p_equals_1_;
             return this.field_177704_a.equals(propertyhelper.field_177704_a) && this.field_177703_b.equals(propertyhelper.field_177703_b);
         }
     }
