--- before/net/minecraft/util/math/AxisAlignedBB.java
+++ after/net/minecraft/util/math/AxisAlignedBB.java
@@ -98,11 +98,13 @@
             {
                 return false;
             }
+            else if (Double.compare(axisalignedbb.maxY, this.maxY) != 0)
+            {
+                return false;
+            }
             else
             {
-                return Double.compare(axisalignedbb.maxY, this.maxY) != 0
-                       ? false
-                       : Double.compare(axisalignedbb.maxZ, this.maxZ) == 0;
+                return Double.compare(axisalignedbb.maxZ, this.maxZ) == 0;
             }
         }
     }
@@ -412,11 +414,13 @@
         {
             return false;
         }
+        else if (vec.y <= this.minY || vec.y >= this.maxY)
+        {
+            return false;
+        }
         else
         {
-            return vec.y <= this.minY || vec.y >= this.maxY
-                   ? false
-                   : !(vec.z <= this.minZ) && !(vec.z >= this.maxZ);
+            return !(vec.z <= this.minZ) && !(vec.z >= this.maxZ);
         }
     }
 
