--- before/net/minecraft/util/math/BlockPos.java
+++ after/net/minecraft/util/math/BlockPos.java
@@ -21,7 +21,7 @@
     private static final int NUM_X_BITS = 1 + MathHelper.log2(MathHelper.smallestEncompassingPowerOfTwo(30000000));
     private static final int NUM_Z_BITS = NUM_X_BITS;
     private static final int NUM_Y_BITS = 64 - NUM_X_BITS - NUM_Z_BITS;
-    private static final int Y_SHIFT = 0 + NUM_Z_BITS;
+    private static final int Y_SHIFT = NUM_Z_BITS;
     private static final int X_SHIFT = Y_SHIFT + NUM_Y_BITS;
     private static final long X_MASK = (1L << NUM_X_BITS) - 1L;
     private static final long Y_MASK = (1L << NUM_Y_BITS) - 1L;
@@ -154,18 +154,12 @@
 
     public BlockPos rotate(Rotation rotationIn)
     {
-        switch (rotationIn)
-        {
-            case NONE:
-            default:
-                return this;
-            case CLOCKWISE_90:
-                return new BlockPos(-this.getZ(), this.getY(), this.getX());
-            case CLOCKWISE_180:
-                return new BlockPos(-this.getX(), this.getY(), -this.getZ());
-            case COUNTERCLOCKWISE_90:
-                return new BlockPos(this.getZ(), this.getY(), -this.getX());
-        }
+        return switch (rotationIn) {
+            case CLOCKWISE_90 -> new BlockPos(-this.getZ(), this.getY(), this.getX());
+            case CLOCKWISE_180 -> new BlockPos(-this.getX(), this.getY(), -this.getZ());
+            case COUNTERCLOCKWISE_90 -> new BlockPos(this.getZ(), this.getY(), -this.getX());
+            default -> this;
+        };
     }
 
     public BlockPos crossProduct(Vec3i vec)
@@ -181,7 +175,7 @@
     {
         return ((long)this.getX() & X_MASK) << X_SHIFT
                | ((long)this.getY() & Y_MASK) << Y_SHIFT
-               | ((long)this.getZ() & Z_MASK) << 0;
+               | ((long) this.getZ() & Z_MASK);
     }
 
     public static BlockPos fromLong(long serialized)
@@ -237,18 +231,18 @@
                         {
                             if (this.lastPosX < x2)
                             {
-                                this.lastPosX++;
+                                ++this.lastPosX;
                             }
                             else if (this.lastPosY < y2)
                             {
                                 this.lastPosX = x1;
-                                this.lastPosY++;
+                                ++this.lastPosY;
                             }
                             else if (this.lastPosZ < z2)
                             {
                                 this.lastPosX = x1;
                                 this.lastPosY = y1;
-                                this.lastPosZ++;
+                                ++this.lastPosZ;
                             }
 
                             return new BlockPos(this.lastPosX, this.lastPosY, this.lastPosZ);
@@ -305,18 +299,18 @@
                         {
                             if (this.pos.x < x2)
                             {
-                                this.pos.x++;
+                                ++this.pos.x;
                             }
                             else if (this.pos.y < y2)
                             {
                                 this.pos.x = x1;
-                                this.pos.y++;
+                                ++this.pos.y;
                             }
                             else if (this.pos.z < z2)
                             {
                                 this.pos.x = x1;
                                 this.pos.y = y1;
-                                this.pos.z++;
+                                ++this.pos.z;
                             }
 
                             return this.pos;
