--- before/net/minecraft/util/math/Cartesian.java
+++ after/net/minecraft/util/math/Cartesian.java
@@ -89,7 +89,7 @@
                         this.iterables = iterables;
                         this.iterators = Cartesian.createArray(Iterator.class, this.iterables.length);
 
-                        for (int i = 0; i < this.iterables.length; i++)
+                        for (int i = 0; i < this.iterables.length; ++i)
                         {
                             this.iterators[i] = iterables[i].iterator();
                         }
@@ -126,7 +126,7 @@
                         {
                             if (this.index >= this.iterators.length)
                             {
-                                for (this.index = this.iterators.length - 1; this.index >= 0; this.index--)
+                                for (this.index = this.iterators.length - 1; this.index >= 0; --this.index)
                                 {
                                     Iterator <? extends T > iterator = this.iterators[this.index];
 
@@ -166,8 +166,8 @@
                         {
                             while (this.index < this.iterators.length)
                             {
-                                this.results[this.index] = (T)this.iterators[this.index].next();
-                                this.index++;
+                                this.results[this.index] = this.iterators[this.index].next();
+                                ++this.index;
                             }
 
                             return (T[])((Object[])this.results.clone());
