--- before/net/minecraft/util/math/Vec3i.java
+++ after/net/minecraft/util/math/Vec3i.java
@@ -30,21 +30,23 @@
         {
             return true;
         }
-        else if (!(p_equals_1_ instanceof Vec3i))
+        else if (!(p_equals_1_ instanceof Vec3i vec3i))
         {
             return false;
         }
         else
         {
-            Vec3i vec3i = (Vec3i)p_equals_1_;
-
             if (this.getX() != vec3i.getX())
             {
                 return false;
             }
+            else if (this.getY() != vec3i.getY())
+            {
+                return false;
+            }
             else
             {
-                return this.getY() != vec3i.getY() ? false : this.getZ() == vec3i.getZ();
+                return this.getZ() == vec3i.getZ();
             }
         }
     }
@@ -95,31 +97,31 @@
 
     public double getDistance(int xIn, int yIn, int zIn)
     {
-        double d0 = (double)(this.getX() - xIn);
-        double d1 = (double)(this.getY() - yIn);
-        double d2 = (double)(this.getZ() - zIn);
+        double d0 = this.getX() - xIn;
+        double d1 = this.getY() - yIn;
+        double d2 = this.getZ() - zIn;
         return Math.sqrt(d0 * d0 + d1 * d1 + d2 * d2);
     }
 
     public double distanceSq(double toX, double toY, double toZ)
     {
-        double d0 = (double)this.getX() - toX;
-        double d1 = (double)this.getY() - toY;
-        double d2 = (double)this.getZ() - toZ;
+        double d0 = (double) this.getX() - toX;
+        double d1 = (double) this.getY() - toY;
+        double d2 = (double) this.getZ() - toZ;
         return d0 * d0 + d1 * d1 + d2 * d2;
     }
 
     public double distanceSqToCenter(double xIn, double yIn, double zIn)
     {
-        double d0 = (double)this.getX() + 0.5 - xIn;
-        double d1 = (double)this.getY() + 0.5 - yIn;
-        double d2 = (double)this.getZ() + 0.5 - zIn;
+        double d0 = (double) this.getX() + 0.5 - xIn;
+        double d1 = (double) this.getY() + 0.5 - yIn;
+        double d2 = (double) this.getZ() + 0.5 - zIn;
         return d0 * d0 + d1 * d1 + d2 * d2;
     }
 
     public double distanceSq(Vec3i to)
     {
-        return this.distanceSq((double)to.getX(), (double)to.getY(), (double)to.getZ());
+        return this.distanceSq(to.getX(), to.getY(), to.getZ());
     }
 
     @Override
