--- before/net/minecraft/entity/passive/AbstractChestHorse.java
+++ after/net/minecraft/entity/passive/AbstractChestHorse.java
@@ -78,7 +78,7 @@
     @Override
     public void onDeath(DamageSource cause)
     {
-        super.onDeath(cause);
+        // super.onDeath(cause); // CraftBukkit - moved down
 
         if (this.hasChest())
         {
@@ -87,8 +87,10 @@
                 this.dropItem(Item.getItemFromBlock(Blocks.CHEST), 1);
             }
 
-            this.setChested(false);
+            // this.setChested(false); // CraftBukkit - moved down
         }
+        super.onDeath(cause);
+        this.setChested(false);
     }
 
     public static void registerFixesAbstractChestHorse(DataFixer fixer, Class<?> entityClass)
@@ -107,7 +109,7 @@
         {
             NBTTagList nbttaglist = new NBTTagList();
 
-            for (int i = 2; i < this.horseChest.getSizeInventory(); i++)
+            for (int i = 2; i < this.horseChest.getSizeInventory(); ++i)
             {
                 ItemStack itemstack = this.horseChest.getStackInSlot(i);
 
@@ -135,7 +137,7 @@
             NBTTagList nbttaglist = compound.getTagList("Items", 10);
             this.initHorseChest();
 
-            for (int i = 0; i < nbttaglist.tagCount(); i++)
+            for (int i = 0; i < nbttaglist.tagCount(); ++i)
             {
                 NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
                 int j = nbttagcompound.getByte("Slot") & 255;
