--- before/net/minecraft/entity/passive/EntityDonkey.java
+++ after/net/minecraft/entity/passive/EntityDonkey.java
@@ -57,11 +57,13 @@
         {
             return false;
         }
+        else if (!(otherAnimal instanceof EntityDonkey) && !(otherAnimal instanceof EntityHorse))
+        {
+            return false;
+        }
         else
         {
-            return !(otherAnimal instanceof EntityDonkey) && !(otherAnimal instanceof EntityHorse)
-                   ? false
-                   : this.canMate() && ((AbstractHorse)otherAnimal).canMate();
+            return this.canMate() && ((AbstractHorse)otherAnimal).canMate();
         }
     }
 
