--- before/net/minecraft/entity/passive/EntityAnimal.java
+++ after/net/minecraft/entity/passive/EntityAnimal.java
@@ -25,6 +25,8 @@
     private int inLove;
     private UUID playerInLove;
 
+    public ItemStack breedItem; // CraftBukkit - Add breedItem variable
+
     public EntityAnimal(World worldIn)
     {
         super(worldIn);
@@ -53,7 +55,7 @@
 
         if (this.inLove > 0)
         {
-            this.inLove--;
+            --this.inLove;
 
             if (this.inLove % 10 == 0)
             {
@@ -74,7 +76,8 @@
         }
     }
 
-    @Override
+    /* CraftBukkit start
+    // Function disabled as it has no special function anymore after setSitting is disabled.
     public boolean attackEntityFrom(DamageSource source, float amount)
     {
         if (this.isEntityInvulnerable(source))
@@ -87,6 +90,7 @@
             return super.attackEntityFrom(source, amount);
         }
     }
+    // CraftBukkit end */
 
     @Override
     public float getBlockPathWeight(BlockPos pos)
@@ -140,8 +144,7 @@
         return 120;
     }
 
-    @Override
-    protected boolean canDespawn()
+    public boolean canDespawn()
     {
         return false;
     }
@@ -197,6 +200,7 @@
         if (player != null)
         {
             this.playerInLove = player.getUniqueID();
+            this.breedItem = player.inventory.getCurrentItem();
         }
 
         this.world.setEntityState(this, (byte)18);
@@ -232,9 +236,13 @@
         {
             return false;
         }
+        else if (otherAnimal.getClass() != this.getClass())
+        {
+            return false;
+        }
         else
         {
-            return otherAnimal.getClass() != this.getClass() ? false : this.isInLove() && otherAnimal.isInLove();
+            return this.isInLove() && otherAnimal.isInLove();
         }
     }
 
@@ -244,7 +252,7 @@
     {
         if (id == 18)
         {
-            for (int i = 0; i < 7; i++)
+            for (int i = 0; i < 7; ++i)
             {
                 double d0 = this.rand.nextGaussian() * 0.02;
                 double d1 = this.rand.nextGaussian() * 0.02;
