--- before/net/minecraft/entity/passive/EntityChicken.java
+++ after/net/minecraft/entity/passive/EntityChicken.java
@@ -82,6 +82,9 @@
     @Override
     public void onLivingUpdate()
     {
+        if (this.isChickenJockey()) {
+            this.persistenceRequired = !this.canDespawn();
+        }
         super.onLivingUpdate();
         this.oFlap = this.wingRotation;
         this.oFlapSpeed = this.destPos;
@@ -100,12 +103,14 @@
             this.motionY *= 0.6;
         }
 
-        this.wingRotation = this.wingRotation + this.wingRotDelta * 2.0F;
+        this.wingRotation += this.wingRotDelta * 2.0F;
 
         if (!this.world.isRemote && !this.isChild() && !this.isChickenJockey() && --this.timeUntilNextEgg <= 0)
         {
             this.playSound(SoundEvents.ENTITY_CHICKEN_EGG, 1.0F, (this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F);
+            this.forceDrops = true;
             this.dropItem(Items.EGG, 1);
+            this.forceDrops = false;
             this.timeUntilNextEgg = this.rand.nextInt(6000) + 6000;
         }
     }
@@ -189,7 +194,7 @@
     }
 
     @Override
-    protected boolean canDespawn()
+    public boolean canDespawn()
     {
         return this.isChickenJockey() && !this.isBeingRidden();
     }
