--- before/net/minecraft/entity/passive/EntityBat.java
+++ after/net/minecraft/entity/passive/EntityBat.java
@@ -6,6 +6,7 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.SoundEvents;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.network.datasync.DataParameter;
@@ -179,9 +180,9 @@
             double d0 = (double)this.field_82237_a.func_177958_n() + 0.5 - this.field_70165_t;
             double d1 = (double)this.field_82237_a.func_177956_o() + 0.1 - this.field_70163_u;
             double d2 = (double)this.field_82237_a.func_177952_p() + 0.5 - this.field_70161_v;
-            this.field_70159_w = this.field_70159_w + (Math.signum(d0) * 0.5 - this.field_70159_w) * 0.1F;
-            this.field_70181_x = this.field_70181_x + (Math.signum(d1) * 0.7F - this.field_70181_x) * 0.1F;
-            this.field_70179_y = this.field_70179_y + (Math.signum(d2) * 0.5 - this.field_70179_y) * 0.1F;
+            this.field_70159_w += (Math.signum(d0) * 0.5 - this.field_70159_w) * 0.1F;
+            this.field_70181_x += (Math.signum(d1) * 0.7F - this.field_70181_x) * 0.1F;
+            this.field_70179_y += (Math.signum(d2) * 0.5 - this.field_70179_y) * 0.1F;
             float f = (float)(MathHelper.func_181159_b(this.field_70179_y, this.field_70159_w) * 180.0F / (float)Math.PI) - 90.0F;
             float f1 = MathHelper.func_76142_g(f - this.field_70177_z);
             this.field_191988_bg = 0.5F;
