--- before/net/minecraft/server/gui/PlayerListComponent.java
+++ after/net/minecraft/server/gui/PlayerListComponent.java
@@ -26,7 +26,7 @@
         {
             Vector<String> vector = new Vector<>();
 
-            for (int i = 0; i < this.server.getPlayerList().getPlayers().size(); i++)
+            for (int i = 0; i < this.server.getPlayerList().getPlayers().size(); ++i)
             {
                 vector.add(this.server.getPlayerList().getPlayers().get(i).getName());
             }
