--- before/net/minecraft/inventory/ContainerMerchant.java
+++ after/net/minecraft/inventory/ContainerMerchant.java
@@ -5,6 +5,7 @@
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.World;
+import org.bukkit.craftbukkit.v1_12_R1.inventory.CraftInventoryView;
 
 public class ContainerMerchant extends Container
 {
@@ -12,6 +13,9 @@
     private final InventoryMerchant merchantInventory;
     private final World world;
 
+    private CraftInventoryView bukkitEntity = null;
+    private InventoryPlayer player;
+
     public ContainerMerchant(InventoryPlayer playerInventory, IMerchant merchant, World worldIn)
     {
         this.merchant = merchant;
@@ -20,16 +24,17 @@
         this.addSlotToContainer(new Slot(this.merchantInventory, 0, 36, 53));
         this.addSlotToContainer(new Slot(this.merchantInventory, 1, 62, 53));
         this.addSlotToContainer(new SlotMerchantResult(playerInventory.player, merchant, this.merchantInventory, 2, 120, 53));
+        this.player = playerInventory;
 
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < 3; ++i)
         {
-            for (int j = 0; j < 9; j++)
+            for (int j = 0; j < 9; ++j)
             {
                 this.addSlotToContainer(new Slot(playerInventory, j + i * 9 + 9, 8 + j * 18, 84 + i * 18));
             }
         }
 
-        for (int k = 0; k < 9; k++)
+        for (int k = 0; k < 9; ++k)
         {
             this.addSlotToContainer(new Slot(playerInventory, k, 8 + k * 18, 142));
         }
@@ -140,5 +145,13 @@
                 playerIn.dropItem(itemstack, false);
             }
         }
+    }
+
+    @Override
+    public CraftInventoryView getBukkitView() {
+        if (bukkitEntity == null) {
+            bukkitEntity = new CraftInventoryView(this.player.player.getBukkitEntity(), new org.bukkit.craftbukkit.v1_12_R1.inventory.CraftInventoryMerchant(merchantInventory), this);
+        }
+        return bukkitEntity;
     }
 }
