--- before/net/minecraft/inventory/InventoryHelper.java
+++ after/net/minecraft/inventory/InventoryHelper.java
@@ -23,7 +23,7 @@
 
     private static void dropInventoryItems(World worldIn, double x, double y, double z, IInventory inventory)
     {
-        for (int i = 0; i < inventory.getSizeInventory(); i++)
+        for (int i = 0; i < inventory.getSizeInventory(); ++i)
         {
             ItemStack itemstack = inventory.getStackInSlot(i);
 
