--- before/net/minecraft/inventory/ItemStackHelper.java
+++ after/net/minecraft/inventory/ItemStackHelper.java
@@ -29,7 +29,7 @@
     {
         NBTTagList nbttaglist = new NBTTagList();
 
-        for (int i = 0; i < list.size(); i++)
+        for (int i = 0; i < list.size(); ++i)
         {
             ItemStack itemstack = list.get(i);
 
@@ -54,7 +54,7 @@
     {
         NBTTagList nbttaglist = tag.getTagList("Items", 10);
 
-        for (int i = 0; i < nbttaglist.tagCount(); i++)
+        for (int i = 0; i < nbttaglist.tagCount(); ++i)
         {
             NBTTagCompound nbttagcompound = nbttaglist.getCompoundTagAt(i);
             int j = nbttagcompound.getByte("Slot") & 255;
