--- before/net/minecraft/inventory/SlotMerchantResult.java
+++ after/net/minecraft/inventory/SlotMerchantResult.java
@@ -32,7 +32,7 @@
     {
         if (this.getHasStack())
         {
-            this.removeCount = this.removeCount + Math.min(amount, this.getStack().getCount());
+            this.removeCount += Math.min(amount, this.getStack().getCount());
         }
 
         return super.decrStackSize(amount);
