--- before/net/minecraft/entity/ai/EntityAIDefendVillage.java
+++ after/net/minecraft/entity/ai/EntityAIDefendVillage.java
@@ -53,7 +53,7 @@
     @Override
     public void startExecuting()
     {
-        this.irongolem.setAttackTarget(this.villageAgressorTarget);
+        this.irongolem.setGoalTarget(this.villageAgressorTarget, org.bukkit.event.entity.EntityTargetEvent.TargetReason.DEFEND_VILLAGE, true);
         super.startExecuting();
     }
 }
