--- before/net/minecraft/entity/ai/EntityAIMoveToBlock.java
+++ after/net/minecraft/entity/ai/EntityAIMoveToBlock.java
@@ -28,7 +28,7 @@
     {
         if (this.runDelay > 0)
         {
-            this.runDelay--;
+            --this.runDelay;
             return false;
         }
         else
@@ -67,7 +67,7 @@
         if (this.creature.getDistanceSqToCenter(this.destinationBlock.up()) > 1.0)
         {
             this.isAboveDestination = false;
-            this.timeoutCounter++;
+            ++this.timeoutCounter;
 
             if (this.timeoutCounter % 40 == 0)
             {
@@ -84,7 +84,7 @@
         else
         {
             this.isAboveDestination = true;
-            this.timeoutCounter--;
+            --this.timeoutCounter;
         }
     }
 
@@ -101,7 +101,7 @@
 
         for (int k = 0; k <= 1; k = k > 0 ? -k : 1 - k)
         {
-            for (int l = 0; l < i; l++)
+            for (int l = 0; l < i; ++l)
             {
                 for (int i1 = 0; i1 <= l; i1 = i1 > 0 ? -i1 : 1 - i1)
                 {
