--- before/net/minecraft/entity/ai/EntityAIFindEntityNearest.java
+++ after/net/minecraft/entity/ai/EntityAIFindEntityNearest.java
@@ -94,16 +94,22 @@
         else
         {
             double d0 = this.getFollowRange();
-            return this.mob.getDistanceSq(entitylivingbase) > d0 * d0
-                   ? false
-                   : !(entitylivingbase instanceof EntityPlayerMP) || !((EntityPlayerMP)entitylivingbase).interactionManager.isCreative();
+
+            if (this.mob.getDistanceSq(entitylivingbase) > d0 * d0)
+            {
+                return false;
+            }
+            else
+            {
+                return !(entitylivingbase instanceof EntityPlayerMP) || !((EntityPlayerMP)entitylivingbase).interactionManager.isCreative();
+            }
         }
     }
 
     @Override
     public void startExecuting()
     {
-        this.mob.setAttackTarget(this.target);
+        this.mob.setGoalTarget(this.target, org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_ENTITY, true);
         super.startExecuting();
     }
 
