--- before/net/minecraft/entity/ai/EntityAISkeletonRiders.java
+++ after/net/minecraft/entity/ai/EntityAISkeletonRiders.java
@@ -1,6 +1,7 @@
 package net.minecraft.entity.ai;
 
 import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.effect.EntityLightningBolt;
 import net.minecraft.entity.monster.EntitySkeleton;
 import net.minecraft.entity.passive.AbstractHorse;
@@ -11,6 +12,8 @@
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.DifficultyInstance;
 
+import javax.annotation.Nullable;
+
 public class EntityAISkeletonRiders extends EntityAIBase
 {
     private final EntitySkeletonHorse horse;
@@ -43,17 +46,19 @@
             )
         );
         EntitySkeleton entityskeleton = this.createSkeleton(difficultyinstance, this.horse);
-        entityskeleton.startRiding(this.horse);
+        if (entityskeleton != null) entityskeleton.startRiding(this.horse);
 
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < 3; ++i)
         {
             AbstractHorse abstracthorse = this.createHorse(difficultyinstance);
+            if (abstracthorse == null) continue; // CraftBukkit
             EntitySkeleton entityskeleton1 = this.createSkeleton(difficultyinstance, abstracthorse);
-            entityskeleton1.startRiding(abstracthorse);
-            abstracthorse.addVelocity(this.horse.getRNG().nextGaussian() * 0.5, 0.0, this.horse.getRNG().nextGaussian() * 0.5);
+            if (entityskeleton1 != null) entityskeleton1.startRiding(abstracthorse);
+            abstracthorse.addVelocity(this.horse.getRNG().nextGaussian() * 0.5D, 0.0D, this.horse.getRNG().nextGaussian() * 0.5D);
         }
     }
 
+    @Nullable
     private AbstractHorse createHorse(DifficultyInstance p_188515_1_)
     {
         EntitySkeletonHorse entityskeletonhorse = new EntitySkeletonHorse(this.horse.world);
@@ -63,10 +68,11 @@
         entityskeletonhorse.enablePersistence();
         entityskeletonhorse.setHorseTamed(true);
         entityskeletonhorse.setGrowingAge(0);
-        entityskeletonhorse.world.spawnEntity(entityskeletonhorse);
+        if (!entityskeletonhorse.world.addEntity(entityskeletonhorse, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.TRAP)) return null; // CraftBukkit
         return entityskeletonhorse;
     }
 
+    @Nullable
     private EntitySkeleton createSkeleton(DifficultyInstance p_188514_1_, AbstractHorse p_188514_2_)
     {
         EntitySkeleton entityskeleton = new EntitySkeleton(p_188514_2_.world);
@@ -98,7 +104,7 @@
                 false
             )
         );
-        entityskeleton.world.spawnEntity(entityskeleton);
+        if (!entityskeleton.world.addEntity(entityskeleton, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.JOCKEY)) return null; // CraftBukkit
         return entityskeleton;
     }
 }
