--- before/net/minecraft/entity/ai/EntityAIMoveToBlock.java
+++ after/net/minecraft/entity/ai/EntityAIMoveToBlock.java
@@ -28,7 +28,7 @@
     {
         if (this.field_179496_a > 0)
         {
-            this.field_179496_a--;
+            --this.field_179496_a;
             return false;
         }
         else
@@ -67,7 +67,7 @@
         if (this.field_179495_c.func_174831_c(this.field_179494_b.func_177984_a()) > 1.0)
         {
             this.field_179491_g = false;
-            this.field_179493_e++;
+            ++this.field_179493_e;
 
             if (this.field_179493_e % 40 == 0)
             {
@@ -84,7 +84,7 @@
         else
         {
             this.field_179491_g = true;
-            this.field_179493_e--;
+            --this.field_179493_e;
         }
     }
 
@@ -101,7 +101,7 @@
 
         for (int k = 0; k <= 1; k = k > 0 ? -k : 1 - k)
         {
-            for (int l = 0; l < i; l++)
+            for (int l = 0; l < i; ++l)
             {
                 for (int i1 = 0; i1 <= l; i1 = i1 > 0 ? -i1 : 1 - i1)
                 {
