--- before/net/minecraft/entity/ai/EntityAIFindEntityNearest.java
+++ after/net/minecraft/entity/ai/EntityAIFindEntityNearest.java
@@ -94,16 +94,22 @@
         else
         {
             double d0 = this.func_179438_f();
-            return this.field_179442_b.func_70068_e(entitylivingbase) > d0 * d0
-                   ? false
-                   : !(entitylivingbase instanceof EntityPlayerMP) || !((EntityPlayerMP)entitylivingbase).field_71134_c.func_73083_d();
+
+            if (this.field_179442_b.func_70068_e(entitylivingbase) > d0 * d0)
+            {
+                return false;
+            }
+            else
+            {
+                return !(entitylivingbase instanceof EntityPlayerMP) || !((EntityPlayerMP)entitylivingbase).field_71134_c.func_73083_d();
+            }
         }
     }
 
     @Override
     public void func_75249_e()
     {
-        this.field_179442_b.func_70624_b(this.field_179441_e);
+        this.field_179442_b.setGoalTarget(this.field_179441_e, org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_ENTITY, true);
         super.func_75249_e();
     }
 
