--- before/net/minecraft/entity/ai/EntityAITradePlayer.java
+++ after/net/minecraft/entity/ai/EntityAITradePlayer.java
@@ -40,9 +40,13 @@
             {
                 return false;
             }
+            else if (this.field_75276_a.func_70068_e(entityplayer) > 16.0)
+            {
+                return false;
+            }
             else
             {
-                return this.field_75276_a.func_70068_e(entityplayer) > 16.0 ? false : entityplayer.field_71070_bA != null;
+                return entityplayer.field_71070_bA != null;
             }
         }
     }
