--- before/net/minecraft/entity/ai/EntityAITarget.java
+++ after/net/minecraft/entity/ai/EntityAITarget.java
@@ -13,6 +13,7 @@
 import net.minecraft.scoreboard.Team;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
+import org.bukkit.event.entity.EntityTargetEvent;
 
 public abstract class EntityAITarget extends EntityAIBase
 {
@@ -92,7 +93,7 @@
                     }
                     else
                     {
-                        this.taskOwner.setAttackTarget(entitylivingbase);
+                        this.taskOwner.setGoalTarget(entitylivingbase, EntityTargetEvent.TargetReason.CLOSEST_ENTITY, true);
                         return true;
                     }
                 }
@@ -117,7 +118,7 @@
     @Override
     public void resetTask()
     {
-        this.taskOwner.setAttackTarget(null);
+        this.taskOwner.setGoalTarget((EntityLivingBase)null, EntityTargetEvent.TargetReason.FORGOT_TARGET, true);
         this.target = null;
     }
 
@@ -135,7 +136,7 @@
         {
             return false;
         }
-        else if (!attacker.canAttackClass((Class <? extends EntityLivingBase >)target.getClass()))
+        else if (!attacker.canAttackClass(target.getClass()))
         {
             return false;
         }
