--- before/net/minecraft/entity/ai/EntityAIOcelotAttack.java
+++ after/net/minecraft/entity/ai/EntityAIOcelotAttack.java
@@ -41,9 +41,13 @@
         {
             return false;
         }
+        else if (this.entity.getDistanceSq(this.target) > 225.0)
+        {
+            return false;
+        }
         else
         {
-            return this.entity.getDistanceSq(this.target) > 225.0 ? false : !this.entity.getNavigator().noPath() || this.shouldExecute();
+            return !this.entity.getNavigator().noPath() || this.shouldExecute();
         }
     }
 
