--- before/net/minecraft/entity/ai/EntityAIFollowParent.java
+++ after/net/minecraft/entity/ai/EntityAIFollowParent.java
@@ -27,7 +27,7 @@
         {
             List<EntityAnimal> list = this.childAnimal
                                       .world
-                                      .getEntitiesWithinAABB((Class <? extends EntityAnimal >)this.childAnimal.getClass(), this.childAnimal.getEntityBoundingBox().grow(8.0, 4.0, 8.0));
+                                      .getEntitiesWithinAABB(this.childAnimal.getClass(), this.childAnimal.getEntityBoundingBox().grow(8.0, 4.0, 8.0));
             EntityAnimal entityanimal = null;
             double d0 = Double.MAX_VALUE;
 
