--- before/net/minecraft/entity/ai/EntityAIVillagerInteract.java
+++ after/net/minecraft/entity/ai/EntityAIVillagerInteract.java
@@ -41,13 +41,13 @@
 
         if (this.interactionDelay > 0)
         {
-            this.interactionDelay--;
+            --this.interactionDelay;
 
             if (this.interactionDelay == 0)
             {
                 InventoryBasic inventorybasic = this.villager.getVillagerInventory();
 
-                for (int i = 0; i < inventorybasic.getSizeInventory(); i++)
+                for (int i = 0; i < inventorybasic.getSizeInventory(); ++i)
                 {
                     ItemStack itemstack = inventorybasic.getStackInSlot(i);
                     ItemStack itemstack1 = ItemStack.EMPTY;
