--- before/net/minecraft/entity/ai/EntityAISit.java
+++ after/net/minecraft/entity/ai/EntityAISit.java
@@ -19,7 +19,8 @@
     {
         if (!this.tameable.isTamed())
         {
-            return false;
+            // return false;
+            return this.isSitting && this.tameable.getAttackTarget() == null; // CraftBukkit - Allow sitting for wild animals
         }
         else if (this.tameable.isInWater())
         {
