--- before/net/minecraft/entity/ai/EntityAILlamaFollowCaravan.java
+++ after/net/minecraft/entity/ai/EntityAILlamaFollowCaravan.java
@@ -24,7 +24,7 @@
         {
             List<EntityLlama> list = this.llama
                                      .world
-                                     .getEntitiesWithinAABB((Class <? extends EntityLlama >)this.llama.getClass(), this.llama.getEntityBoundingBox().grow(9.0, 4.0, 9.0));
+                                     .getEntitiesWithinAABB(this.llama.getClass(), this.llama.getEntityBoundingBox().grow(9.0, 4.0, 9.0));
             EntityLlama entityllama = null;
             double d0 = Double.MAX_VALUE;
 
@@ -107,7 +107,7 @@
 
             if (this.distCheckCounter > 0)
             {
-                this.distCheckCounter--;
+                --this.distCheckCounter;
             }
 
             return true;
