--- before/net/minecraft/entity/ai/EntityAIFleeSun.java
+++ after/net/minecraft/entity/ai/EntityAIFleeSun.java
@@ -81,7 +81,7 @@
         Random random = this.creature.getRNG();
         BlockPos blockpos = new BlockPos(this.creature.posX, this.creature.getEntityBoundingBox().minY, this.creature.posZ);
 
-        for (int i = 0; i < 10; i++)
+        for (int i = 0; i < 10; ++i)
         {
             BlockPos blockpos1 = blockpos.add(random.nextInt(20) - 10, random.nextInt(6) - 3, random.nextInt(20) - 10);
 
