--- before/net/minecraft/entity/ai/EntityAITradePlayer.java
+++ after/net/minecraft/entity/ai/EntityAITradePlayer.java
@@ -40,9 +40,13 @@
             {
                 return false;
             }
+            else if (this.villager.getDistanceSq(entityplayer) > 16.0)
+            {
+                return false;
+            }
             else
             {
-                return this.villager.getDistanceSq(entityplayer) > 16.0 ? false : entityplayer.openContainer != null;
+                return entityplayer.openContainer != null;
             }
         }
     }
