--- before/net/minecraft/entity/ai/EntityAIWatchClosest.java
+++ after/net/minecraft/entity/ai/EntityAIWatchClosest.java
@@ -81,9 +81,13 @@
         {
             return false;
         }
+        else if (this.entity.getDistanceSq(this.closestEntity) > (double)(this.maxDistance * this.maxDistance))
+        {
+            return false;
+        }
         else
         {
-            return this.entity.getDistanceSq(this.closestEntity) > (double)(this.maxDistance * this.maxDistance) ? false : this.lookTime > 0;
+            return this.lookTime > 0;
         }
     }
 
@@ -111,6 +115,6 @@
             (float)this.entity.getHorizontalFaceSpeed(),
             (float)this.entity.getVerticalFaceSpeed()
         );
-        this.lookTime--;
+        --this.lookTime;
     }
 }
