--- before/net/minecraft/entity/ai/EntityAIBeg.java
+++ after/net/minecraft/entity/ai/EntityAIBeg.java
@@ -37,11 +37,13 @@
         {
             return false;
         }
+        else if (this.field_75387_a.func_70068_e(this.field_75385_b) > (double)(this.field_75383_d * this.field_75383_d))
+        {
+            return false;
+        }
         else
         {
-            return this.field_75387_a.func_70068_e(this.field_75385_b) > (double)(this.field_75383_d * this.field_75383_d)
-                   ? false
-                   : this.field_75384_e > 0 && this.func_75382_a(this.field_75385_b);
+            return this.field_75384_e > 0 && this.func_75382_a(this.field_75385_b);
         }
     }
 
@@ -71,7 +73,7 @@
             10.0F,
             (float)this.field_75387_a.func_70646_bf()
         );
-        this.field_75384_e--;
+        --this.field_75384_e;
     }
 
     private boolean func_75382_a(EntityPlayer p_75382_1_)
