--- before/net/minecraft/entity/ai/EntityAIDoorInteract.java
+++ after/net/minecraft/entity/ai/EntityAIDoorInteract.java
@@ -43,7 +43,7 @@
 
             if (path != null && !path.isFinished() && pathnavigateground.getEnterDoors())
             {
-                for (int i = 0; i < Math.min(path.getCurrentPathIndex() + 2, path.getCurrentPathLength()); i++)
+                for (int i = 0; i < Math.min(path.getCurrentPathIndex() + 2, path.getCurrentPathLength()); ++i)
                 {
                     PathPoint pathpoint = path.getPathPointFromIndex(i);
                     this.doorPosition = new BlockPos(pathpoint.x, pathpoint.y + 1, pathpoint.z);
