--- before/net/minecraft/entity/ai/EntityAIBeg.java
+++ after/net/minecraft/entity/ai/EntityAIBeg.java
@@ -37,11 +37,13 @@
         {
             return false;
         }
+        else if (this.wolf.getDistanceSq(this.player) > (double)(this.minPlayerDistance * this.minPlayerDistance))
+        {
+            return false;
+        }
         else
         {
-            return this.wolf.getDistanceSq(this.player) > (double)(this.minPlayerDistance * this.minPlayerDistance)
-                   ? false
-                   : this.timeoutCounter > 0 && this.hasTemptationItemInHand(this.player);
+            return this.timeoutCounter > 0 && this.hasTemptationItemInHand(this.player);
         }
     }
 
@@ -71,7 +73,7 @@
             10.0F,
             (float)this.wolf.getVerticalFaceSpeed()
         );
-        this.timeoutCounter--;
+        --this.timeoutCounter;
     }
 
     private boolean hasTemptationItemInHand(EntityPlayer player)
