--- before/net/minecraft/entity/ai/EntityAIFindEntityNearestPlayer.java
+++ after/net/minecraft/entity/ai/EntityAIFindEntityNearestPlayer.java
@@ -124,9 +124,15 @@
             else
             {
                 double d0 = this.maxTargetRange();
-                return this.entityLiving.getDistanceSq(entitylivingbase) > d0 * d0
-                       ? false
-                       : !(entitylivingbase instanceof EntityPlayerMP) || !((EntityPlayerMP)entitylivingbase).interactionManager.isCreative();
+
+                if (this.entityLiving.getDistanceSq(entitylivingbase) > d0 * d0)
+                {
+                    return false;
+                }
+                else
+                {
+                    return !(entitylivingbase instanceof EntityPlayerMP) || !((EntityPlayerMP)entitylivingbase).interactionManager.isCreative();
+                }
             }
         }
     }
@@ -134,7 +140,7 @@
     @Override
     public void startExecuting()
     {
-        this.entityLiving.setAttackTarget(this.entityTarget);
+        this.entityLiving.setGoalTarget(this.entityTarget, org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_PLAYER, true);
         super.startExecuting();
     }
 
