--- before/net/minecraft/entity/ai/EntityAIRestrictOpenDoor.java
+++ after/net/minecraft/entity/ai/EntityAIRestrictOpenDoor.java
@@ -40,7 +40,15 @@
             else
             {
                 this.frontDoor = village.getNearestDoor(blockpos);
-                return this.frontDoor == null ? false : (double)this.frontDoor.getDistanceToInsideBlockSq(blockpos) < 2.25;
+
+                if (this.frontDoor == null)
+                {
+                    return false;
+                }
+                else
+                {
+                    return (double)this.frontDoor.getDistanceToInsideBlockSq(blockpos) < 2.25;
+                }
             }
         }
     }
@@ -48,9 +56,14 @@
     @Override
     public boolean shouldContinueExecuting()
     {
-        return this.entity.world.isDaytime()
-               ? false
-               : !this.frontDoor.getIsDetachedFromVillageFlag() && this.frontDoor.isInsideSide(new BlockPos(this.entity));
+        if (this.entity.world.isDaytime())
+        {
+            return false;
+        }
+        else
+        {
+            return !this.frontDoor.getIsDetachedFromVillageFlag() && this.frontDoor.isInsideSide(new BlockPos(this.entity));
+        }
     }
 
     @Override
