--- before/net/minecraft/entity/ai/attributes/ModifiableAttributeInstance.java
+++ after/net/minecraft/entity/ai/attributes/ModifiableAttributeInstance.java
@@ -28,7 +28,7 @@
         this.genericAttribute = genericAttributeIn;
         this.baseValue = genericAttributeIn.getDefaultValue();
 
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < 3; ++i)
         {
             this.mapByOperation.put(i, Sets.newHashSet());
         }
@@ -67,7 +67,7 @@
     {
         Set<AttributeModifier> set = Sets.newHashSet();
 
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < 3; ++i)
         {
             set.addAll(this.getModifiersByOperation(i));
         }
@@ -121,7 +121,7 @@
     @Override
     public void removeModifier(AttributeModifier modifier)
     {
-        for (int i = 0; i < 3; i++)
+        for (int i = 0; i < 3; ++i)
         {
             Set<AttributeModifier> set = this.mapByOperation.get(i);
             set.remove(modifier);
