--- before/net/minecraft/entity/ai/attributes/RangedAttribute.java
+++ after/net/minecraft/entity/ai/attributes/RangedAttribute.java
@@ -6,7 +6,7 @@
 public class RangedAttribute extends BaseAttribute
 {
     private final double minimumValue;
-    private final double maximumValue;
+    public double maximumValue;  // Spigot
     private String description;
 
     public RangedAttribute(@Nullable IAttribute parentIn, String unlocalizedNameIn, double defaultValue, double minimumValueIn, double maximumValueIn)
@@ -43,6 +43,9 @@
     @Override
     public double clampValue(double value)
     {
-        return MathHelper.clamp(value, this.minimumValue, this.maximumValue);
+        if (value != value) return getDefaultValue(); // CraftBukkit
+
+        value = MathHelper.clamp(value, this.minimumValue, this.maximumValue);
+        return value;
     }
 }
